{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/Bio/pairwise2.py:278: BiopythonDeprecationWarning: Bio.pairwise2 has been deprecated, and we intend to remove it in a future release of Biopython. As an alternative, please consider using Bio.Align.PairwiseAligner as a replacement, and contact the Biopython developers if you still need the Bio.pairwise2 module.\n",
      "  warnings.warn(\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/Bio/Data/SCOPData.py:18: BiopythonDeprecationWarning: The 'Bio.Data.SCOPData' module will be deprecated in a future release of Biopython in favor of 'Bio.Data.PDBData.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "from stapep.structure import Structure\n",
    "\n",
    "# Initialize a sequence and output directory\n",
    "# 序列规则:\n",
    "#   使用氨基酸单字母缩写用\n",
    "#   单字母氨基酸之间可以连续写，也可以用'-'分隔，非单字母氨基酸之间必须用'-'分隔。例如: Ac-BATPR8RRR-Aib-LBRR3FKRLQ-NH2\n",
    "#   N端可以使用'Ac-'开头, 添加Acetyl group\n",
    "#   C端可以使用'-NH2'结尾, 添加Amide group\n",
    "#   Aib是2-Aminoisobutyric acid\n",
    "#   B是Nle的缩写，Nle是Norleucine\n",
    "\n",
    "seq = 'Ac-BATP-R8-RRR-Aib-BLBR-R3-FKRLQ'\n",
    "st = Structure(verbose=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from stapep.structure import Structure\n",
    "\n",
    "# De novo 3D structure prediction using ESMFOLD\n",
    "st = Structure(verbose=True)\n",
    "st.de_novo_3d_structure(seq, output_pdb='data/denovo.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/3dmoljs_load.v0": "<div id=\"3dmolviewer_17026094571162472\"  style=\"position: relative; width: 640px; height: 480px;\">\n        <p id=\"3dmolwarning_17026094571162472\" style=\"background-color:#ffcccc;color:black\">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>\n        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>\n        </div>\n<script>\n\nvar loadScriptAsync = function(uri){\n  return new Promise((resolve, reject) => {\n    //this is to ignore the existence of requirejs amd\n    var savedexports, savedmodule;\n    if (typeof exports !== 'undefined') savedexports = exports;\n    else exports = {}\n    if (typeof module !== 'undefined') savedmodule = module;\n    else module = {}\n\n    var tag = document.createElement('script');\n    tag.src = uri;\n    tag.async = true;\n    tag.onload = () => {\n        exports = savedexports;\n        module = savedmodule;\n        resolve();\n    };\n  var firstScriptTag = document.getElementsByTagName('script')[0];\n  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n});\n};\n\nif(typeof $3Dmolpromise === 'undefined') {\n$3Dmolpromise = null;\n  $3Dmolpromise = loadScriptAsync('https://cdnjs.cloudflare.com/ajax/libs/3Dmol/2.0.4/3Dmol-min.js');\n}\n\nvar viewer_17026094571162472 = null;\nvar warn = document.getElementById(\"3dmolwarning_17026094571162472\");\nif(warn) {\n    warn.parentNode.removeChild(warn);\n}\n$3Dmolpromise.then(function() {\nviewer_17026094571162472 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_17026094571162472\"),{backgroundColor:\"white\"});\n$3Dmol.download(\"data/template.pdb\", viewer_17026094571162472, {}, function() {\nviewer_17026094571162472.zoomTo();\n\tviewer_17026094571162472.setStyle({\"model\": 0},{\"cartoon\": {\"color\": \"gray\"}});\n\tviewer_17026094571162472.zoomTo();\nviewer_17026094571162472.render();\n})\n});\n</script>",
      "text/html": [
       "<div id=\"3dmolviewer_17026094571162472\"  style=\"position: relative; width: 640px; height: 480px;\">\n",
       "        <p id=\"3dmolwarning_17026094571162472\" style=\"background-color:#ffcccc;color:black\">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>\n",
       "        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>\n",
       "        </div>\n",
       "<script>\n",
       "\n",
       "var loadScriptAsync = function(uri){\n",
       "  return new Promise((resolve, reject) => {\n",
       "    //this is to ignore the existence of requirejs amd\n",
       "    var savedexports, savedmodule;\n",
       "    if (typeof exports !== 'undefined') savedexports = exports;\n",
       "    else exports = {}\n",
       "    if (typeof module !== 'undefined') savedmodule = module;\n",
       "    else module = {}\n",
       "\n",
       "    var tag = document.createElement('script');\n",
       "    tag.src = uri;\n",
       "    tag.async = true;\n",
       "    tag.onload = () => {\n",
       "        exports = savedexports;\n",
       "        module = savedmodule;\n",
       "        resolve();\n",
       "    };\n",
       "  var firstScriptTag = document.getElementsByTagName('script')[0];\n",
       "  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n",
       "});\n",
       "};\n",
       "\n",
       "if(typeof $3Dmolpromise === 'undefined') {\n",
       "$3Dmolpromise = null;\n",
       "  $3Dmolpromise = loadScriptAsync('https://cdnjs.cloudflare.com/ajax/libs/3Dmol/2.0.4/3Dmol-min.js');\n",
       "}\n",
       "\n",
       "var viewer_17026094571162472 = null;\n",
       "var warn = document.getElementById(\"3dmolwarning_17026094571162472\");\n",
       "if(warn) {\n",
       "    warn.parentNode.removeChild(warn);\n",
       "}\n",
       "$3Dmolpromise.then(function() {\n",
       "viewer_17026094571162472 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_17026094571162472\"),{backgroundColor:\"white\"});\n",
       "$3Dmol.download(\"data/template.pdb\", viewer_17026094571162472, {}, function() {\n",
       "viewer_17026094571162472.zoomTo();\n",
       "\tviewer_17026094571162472.setStyle({\"model\": 0},{\"cartoon\": {\"color\": \"gray\"}});\n",
       "\tviewer_17026094571162472.zoomTo();\n",
       "viewer_17026094571162472.render();\n",
       "})\n",
       "});\n",
       "</script>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<py3Dmol.view at 0x7fe63074b220>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Install py3Dmol: \n",
    "# mamba install -c conda-forge py3dmol\n",
    "# mamba install -c conda-forge jupyterlab\n",
    "# mamba install -c conda-forge nodejs\n",
    "# jupyter labextension install jupyterlab_3dmol\n",
    "\n",
    "import py3Dmol\n",
    " \n",
    "p = py3Dmol.view(query='data/template.pdb')\n",
    "p.setStyle({'model': 0}, {'cartoon': {'color':'gray'}}) \n",
    "p.zoomTo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Some weights of EsmForProteinFolding were not initialized from the model checkpoint at facebook/esmfold_v1 and are newly initialized: ['esm.contact_head.regression.weight', 'esm.contact_head.regression.bias']\n",
      "You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/prep to search path.\n",
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/lib to search path.\n",
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/parm to search path.\n",
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/cmd to search path.\n",
      "-f: Source tleap.in.\n",
      "\n",
      "Welcome to LEaP!\n",
      "(no leaprc in search path)\n",
      "Sourcing: ./tleap.in\n",
      "----- Source: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/cmd/leaprc.protein.ff14SB\n",
      "----- Source of /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/cmd/leaprc.protein.ff14SB done\n",
      "Log file: ./leap.log\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/parm/parm10.dat\n",
      "Reading title:\n",
      "PARM99 + frcmod.ff99SB + frcmod.parmbsc0 + OL3 for RNA\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/parm/frcmod.ff14SB\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "ff14SB protein backbone and sidechain parameters\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/lib/amino12.lib\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/lib/aminoct12.lib\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/lib/aminont12.lib\n",
      "----- Source: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/cmd/leaprc.water.tip3p\n",
      "----- Source of /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/cmd/leaprc.water.tip3p done\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/lib/atomic_ions.lib\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/lib/solvents.lib\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/parm/frcmod.tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "This is the additional/replacement parameter set for TIP3P water\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/parm/frcmod.ions1lm_126_tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Li/Merz ion parameters of monovalent ions for TIP3P water model (12-6 normal usage set)\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/parm/frcmod.ionsjc_tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Monovalent ion parameters for Ewald and TIP3P water from Joung & Cheatham JPCB (2008)\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/dat/leap/parm/frcmod.ions234lm_126_tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Li/Merz ion parameters of divalent to tetravalent ions for TIP3P water model (12-6 normal usage set)\n",
      "Using ArgH and AspGluO modified Bondi2 radii\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR3/pr3.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR3/frcmod.pr3\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR8/pr8.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR8/frcmod.pr8\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/NLE/nle.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/NLE/frcmod.nle\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/AIB/aib.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/AIB/frcmod.aib\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading PDB file: ./model.pdb\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stap/bin/teLeap: Fatal Error!\n",
      "\tNo atoms!\n",
      "\n",
      "Exiting LEaP: Errors = 1; Warnings = 0; Notes = 0.\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "/tmp/88325384-9f67-44b4-988e-c4db4999ab96/pep_vac.prmtop does not exist",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 5\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[38;5;66;03m# homology modeling\u001b[39;00m\n\u001b[1;32m      4\u001b[0m st \u001b[38;5;241m=\u001b[39m Structure(verbose\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m----> 5\u001b[0m \u001b[43mst\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mde_novo_3d_structure\u001b[49m\u001b[43m(\u001b[49m\u001b[43mseq\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moutput_pdb\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mdata/denovo.pdb\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/structure.py:134\u001b[0m, in \u001b[0;36mStructure.de_novo_3d_structure\u001b[0;34m(self, seq, output_pdb)\u001b[0m\n\u001b[1;32m    132\u001b[0m pp \u001b[38;5;241m=\u001b[39m PrepareProt(seq, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtmp_dir, method\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124malphafold\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    133\u001b[0m pp\u001b[38;5;241m.\u001b[39m_gen_prmtop_and_inpcrd_file()\n\u001b[0;32m--> 134\u001b[0m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_short_time_simulation\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    135\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_get_opt_structure(seq, output_pdb)\n\u001b[1;32m    136\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msave_tmp_dir:\n",
      "File \u001b[0;32m~/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/structure.py:76\u001b[0m, in \u001b[0;36mStructure._short_time_simulation\u001b[0;34m(self, nsteps)\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_short_time_simulation\u001b[39m(\u001b[38;5;28mself\u001b[39m, nsteps: \u001b[38;5;28mint\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m100000\u001b[39m):\n\u001b[1;32m     75\u001b[0m     sim \u001b[38;5;241m=\u001b[39m Simulation(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mtmp_dir)\n\u001b[0;32m---> 76\u001b[0m     \u001b[43msim\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msetup\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mtype\u001b[39;49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mimplicit\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     77\u001b[0m \u001b[43m              \u001b[49m\u001b[43msolvent\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msolvent\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     78\u001b[0m \u001b[43m              \u001b[49m\u001b[43mtemperature\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m300\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     79\u001b[0m \u001b[43m              \u001b[49m\u001b[43mfriction\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     80\u001b[0m \u001b[43m              \u001b[49m\u001b[43mtimestep\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m2\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     81\u001b[0m \u001b[43m              \u001b[49m\u001b[43minterval\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;241;43m10\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m     82\u001b[0m \u001b[43m              \u001b[49m\u001b[43mnsteps\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mnsteps\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     83\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mverbose:\n\u001b[1;32m     84\u001b[0m         logging\u001b[38;5;241m.\u001b[39minfo(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mRunning short time simulation for \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mnsteps\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m steps\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/molecular_dynamics.py:417\u001b[0m, in \u001b[0;36mSimulation.setup\u001b[0;34m(self, type, solvent, temperature, friction, timestep, interval, nsteps)\u001b[0m\n\u001b[1;32m    415\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mprmtop \u001b[38;5;241m=\u001b[39m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mwork_path, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpep_vac.prmtop\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    416\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39minpcrd \u001b[38;5;241m=\u001b[39m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mwork_path, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpep_vac.inpcrd\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m--> 417\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpep_solv \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_load_file\u001b[49m\n\u001b[1;32m    418\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m solvent \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mwater\u001b[39m\u001b[38;5;124m'\u001b[39m:\n\u001b[1;32m    419\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msystem \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mpep_solv\u001b[38;5;241m.\u001b[39mcreateSystem(implicitSolvent\u001b[38;5;241m=\u001b[39mapp\u001b[38;5;241m.\u001b[39mOBC2, \n\u001b[1;32m    420\u001b[0m                         soluteDielectric\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1.0\u001b[39m,\n\u001b[1;32m    421\u001b[0m                         solventDielectric\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m78.5\u001b[39m)\n",
      "File \u001b[0;32m~/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/molecular_dynamics.py:479\u001b[0m, in \u001b[0;36mSimulation._load_file\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    477\u001b[0m \u001b[38;5;129m@property\u001b[39m\n\u001b[1;32m    478\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_load_file\u001b[39m(\u001b[38;5;28mself\u001b[39m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m pmd\u001b[38;5;241m.\u001b[39mStructure:\n\u001b[0;32m--> 479\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mpmd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mload_file\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mprmtop\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43minpcrd\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstructure\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Simulation/miniconda3/envs/stap/lib/python3.9/site-packages/parmed/formats/registry.py:166\u001b[0m, in \u001b[0;36mload_file\u001b[0;34m(filename, *args, **kwargs)\u001b[0m\n\u001b[1;32m    164\u001b[0m         \u001b[38;5;28;01massert\u001b[39;00m f\n\u001b[1;32m    165\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mexists(filename):\n\u001b[0;32m--> 166\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mFileNotFoundError\u001b[39;00m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m does not exist\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m%\u001b[39m filename)\n\u001b[1;32m    167\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m os\u001b[38;5;241m.\u001b[39maccess(filename, os\u001b[38;5;241m.\u001b[39mR_OK):\n\u001b[1;32m    168\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mFileNotFoundError\u001b[39;00m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m does not have read permissions set\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m%\u001b[39m filename)\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: /tmp/88325384-9f67-44b4-988e-c4db4999ab96/pep_vac.prmtop does not exist"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "                         MODELLER 10.4, 2022/10/28, r12463\n",
      "\n",
      "     PROTEIN STRUCTURE MODELLING BY SATISFACTION OF SPATIAL RESTRAINTS\n",
      "\n",
      "\n",
      "                     Copyright(c) 1989-2022 Andrej Sali\n",
      "                            All Rights Reserved\n",
      "\n",
      "                             Written by A. Sali\n",
      "                               with help from\n",
      "              B. Webb, M.S. Madhusudhan, M-Y. Shen, G.Q. Dong,\n",
      "          M.A. Marti-Renom, N. Eswar, F. Alber, M. Topf, B. Oliva,\n",
      "             A. Fiser, R. Sanchez, B. Yerkovich, A. Badretdinov,\n",
      "                     F. Melo, J.P. Overington, E. Feyfant\n",
      "                 University of California, San Francisco, USA\n",
      "                    Rockefeller University, New York, USA\n",
      "                      Harvard University, Cambridge, USA\n",
      "                   Imperial Cancer Research Fund, London, UK\n",
      "              Birkbeck College, University of London, London, UK\n",
      "\n",
      "\n",
      "Kind, OS, HostName, Kernel, Processor: 4, Linux M5 5.4.0-104-generic x86_64\n",
      "Date and time of compilation         : 2022/10/28 20:54:52\n",
      "MODELLER executable type             : x86_64-intel8\n",
      "Job starting time (YY/MM/DD HH:MM:SS): 2023/09/18 08:41:07\n",
      "\n",
      "read_pd_459W> Residue type  ACE not recognized. 'AutoModel' model building\n",
      "              will treat this residue as a rigid body.\n",
      "              To use real parameters, add the residue type to ${LIB}/restyp.lib,\n",
      "              its topology to ${LIB}/top_*.lib, and suitable forcefield\n",
      "              parameters to ${LIB}/par.lib.\n",
      "read_pd_459W> Residue type  NLE not recognized. 'AutoModel' model building\n",
      "              will treat this residue as a rigid body.\n",
      "              To use real parameters, add the residue type to ${LIB}/restyp.lib,\n",
      "              its topology to ${LIB}/top_*.lib, and suitable forcefield\n",
      "              parameters to ${LIB}/par.lib.\n",
      "read_pd_459W> Residue type  AIB not recognized. 'AutoModel' model building\n",
      "              will treat this residue as a rigid body.\n",
      "              To use real parameters, add the residue type to ${LIB}/restyp.lib,\n",
      "              its topology to ${LIB}/top_*.lib, and suitable forcefield\n",
      "              parameters to ${LIB}/par.lib.\n",
      "fndatmi_285W> Only       19 residues out of       20 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "mkapsa__637W> No residue topology library is in memory.\n",
      "              Better radii would be used if topology.read() is called first.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  C -->  CT2\n",
      "              This message is written only for the first such atom.\n",
      "fndatmi_285W> Only       19 residues out of       20 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "fndatmi_285W> Only       19 residues out of       20 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "\n",
      "Pairwise dynamic programming alignment (ALIGN2D):\n",
      "  Residue-residue metric   : $(LIB)/as1.sim.mat\n",
      "  Diagonal                 :          100\n",
      "  Overhang                 :            0\n",
      "  Maximal gap length       :       999999\n",
      "  Local alignment          :            F\n",
      "  MATRIX_OFFSET (local aln):       0.0000\n",
      "  FIX_OFFSETS              :      0.0    -1.0    -2.0    -3.0    -4.0\n",
      "  N_SUBOPT                 :            0\n",
      "  SUBOPT_OFFSET            :       0.0000\n",
      "  Alignment block          :            1\n",
      "  Gap introduction penalty :    -100.0000\n",
      "  Gap extension penalty    :       0.0000\n",
      "  Gap diagonal penalty     :       0.0000\n",
      "  Structure gap penalties  :    3.500   3.500   3.500   0.200   4.000   6.500   2.000   0.000\n",
      "  Break-break bonus        :   10000.0000\n",
      "  Length of alignment      :           21\n",
      "  Score                    :   12743.4443\n",
      "automodel__W> The class 'automodel' is deprecated; use 'AutoModel' instead\n",
      "fndatmi_285W> Only       19 residues out of       20 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "fndatmi_285W> Only       19 residues out of       20 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "fndatmi_285W> Only       19 residues out of       20 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  C -->  CT2\n",
      "              This message is written only for the first such atom.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:     1123      997\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :       19\n",
      "Number of all, selected real atoms                :      150     150\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :      997     997\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):      254\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :          26.0292\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :     151       0      0   0.002   0.002     0.21980       1.000\n",
      " 2 Bond angle potential               :     201       0      0   1.300   1.300      6.6069       1.000\n",
      " 3 Stereochemical cosine torsion poten:      85       0      0  42.177  42.177      22.330       1.000\n",
      " 4 Stereochemical improper torsion pot:      54       0      0   0.841   0.841     0.72093       1.000\n",
      " 5 Soft-sphere overlap restraints     :     254       0      0   0.000   0.000     0.51150E-02   1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :      92       0      0   0.138   0.138      1.2882       1.000\n",
      "10 Distance restraints 2 (N-O)        :     144       0      0   0.223   0.223      2.0271       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:      18       0      0   2.205   2.205      1.0321       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:      12       0      0  50.098  50.098     0.26503       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:      11       0      0  65.843  65.843      2.0188       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:       8       0      0  70.485  70.485      3.5095       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:       6       0      0 100.242 100.242      4.0546       1.000\n",
      "18 Disulfide distance restraints      :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "19 Disulfide angle restraints         :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "20 Disulfide dihedral angle restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :     176       0      0   0.314   0.314      1.5991       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:      17       0      0   8.550   8.550     -20.448       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :      22       0      0   0.914   0.914     0.80075       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: target.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:     907.8442\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    0    0    4   16   11   13   19   19   28   15\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "target.B99990001.pdb            26.02916\n",
      "\n",
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/prep to search path.\n",
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/lib to search path.\n",
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/parm to search path.\n",
      "-I: Adding /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/cmd to search path.\n",
      "-f: Source tleap.in.\n",
      "\n",
      "Welcome to LEaP!\n",
      "(no leaprc in search path)\n",
      "Sourcing: ./tleap.in\n",
      "----- Source: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/cmd/leaprc.protein.ff14SB\n",
      "----- Source of /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/cmd/leaprc.protein.ff14SB done\n",
      "Log file: ./leap.log\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/parm/parm10.dat\n",
      "Reading title:\n",
      "PARM99 + frcmod.ff99SB + frcmod.parmbsc0 + OL3 for RNA\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/parm/frcmod.ff14SB\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "ff14SB protein backbone and sidechain parameters\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/lib/amino12.lib\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/lib/aminoct12.lib\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/lib/aminont12.lib\n",
      "----- Source: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/cmd/leaprc.water.tip3p\n",
      "----- Source of /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/cmd/leaprc.water.tip3p done\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/lib/atomic_ions.lib\n",
      "Loading library: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/lib/solvents.lib\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/parm/frcmod.tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "This is the additional/replacement parameter set for TIP3P water\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/parm/frcmod.ions1lm_126_tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Li/Merz ion parameters of monovalent ions for TIP3P water model (12-6 normal usage set)\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/parm/frcmod.ionsjc_tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Monovalent ion parameters for Ewald and TIP3P water from Joung & Cheatham JPCB (2008)\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/dat/leap/parm/frcmod.ions234lm_126_tip3p\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Li/Merz ion parameters of divalent to tetravalent ions for TIP3P water model (12-6 normal usage set)\n",
      "Using ArgH and AspGluO modified Bondi2 radii\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR3/pr3.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR3/frcmod.pr3\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR8/pr8.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/PR8/frcmod.pr8\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/NLE/nle.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/NLE/frcmod.nle\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading Prep file: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/AIB/aib.prepin\n",
      "Loading parameters: /home/dahuilangda/Simulation/miniconda3/envs/stapep/lib/python3.9/site-packages/stapep-0.1-py3.9.egg/stapep/templates/AIB/frcmod.aib\n",
      "Reading force field modification type file (frcmod)\n",
      "Reading title:\n",
      "Remark line goes here\n",
      "Loading PDB file: ./model.pdb\n",
      "  Added missing heavy atom: .R<ACE 0>.A<CH3 2>\n",
      "  Added missing heavy atom: .R<ACE 0>.A<O 6>\n",
      "  Added missing heavy atom: .R<NLE 1>.A<C4 4>\n",
      "  Added missing heavy atom: .R<NLE 1>.A<C3 5>\n",
      "  Added missing heavy atom: .R<NLE 1>.A<C2 6>\n",
      "  Added missing heavy atom: .R<NLE 1>.A<C1 7>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C8 4>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C10 24>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C7 5>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C6 6>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C5 7>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C4 8>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C3 9>\n",
      "  Added missing heavy atom: .R<PR8 5>.A<C2 10>\n",
      "  Added missing heavy atom: .R<AIB 9>.A<C1 4>\n",
      "  Added missing heavy atom: .R<AIB 9>.A<C3 8>\n",
      "  Added missing heavy atom: .R<NLE 10>.A<C4 4>\n",
      "  Added missing heavy atom: .R<NLE 10>.A<C3 5>\n",
      "  Added missing heavy atom: .R<NLE 10>.A<C2 6>\n",
      "  Added missing heavy atom: .R<NLE 10>.A<C1 7>\n",
      "  Added missing heavy atom: .R<NLE 12>.A<C4 4>\n",
      "  Added missing heavy atom: .R<NLE 12>.A<C3 5>\n",
      "  Added missing heavy atom: .R<NLE 12>.A<C2 6>\n",
      "  Added missing heavy atom: .R<NLE 12>.A<C1 7>\n",
      "  Added missing heavy atom: .R<PR3 14>.A<C5 4>\n",
      "  Added missing heavy atom: .R<PR3 14>.A<C3 8>\n",
      "  Added missing heavy atom: .R<PR3 14>.A<C2 9>\n",
      "  total atoms in file: 145\n",
      "  Leap added 230 missing atoms according to residue templates:\n",
      "       27 Heavy\n",
      "       203 H / lone pairs\n",
      "Checking Unit.\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stapep/bin/teLeap: Warning!\n",
      "There is a bond of 18.617 angstroms between C2 and C2 atoms:\n",
      "-------  .R<PR8 5>.A<C2 10> and .R<PR3 14>.A<C2 9>\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stapep/bin/teLeap: Warning!\n",
      "The unperturbed charge of the unit (4.999990) is not zero.\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stapep/bin/teLeap: Note.\n",
      "Ignoring the warnings from Unit Checking.\n",
      "\n",
      "Building topology.\n",
      "Building atom parameters.\n",
      "Building bond parameters.\n",
      "Building angle parameters.\n",
      "Building proper torsion parameters.\n",
      "Building improper torsion parameters.\n",
      "old PREP-specified impropers:\n",
      " <NLE 1>:  CA   +M   C    O   \n",
      " <PR8 5>:  CA   +M   C    O   \n",
      " <AIB 9>:  CA   +M   C    O   \n",
      " <NLE 10>:  CA   +M   C    O   \n",
      " <NLE 12>:  CA   +M   C    O   \n",
      " <PR3 14>:  CA   +M   C    O   \n",
      " total 69 improper torsions applied\n",
      " 6 improper torsions in old prep form\n",
      "Building H-Bond parameters.\n",
      "Incorporating Non-Bonded adjustments.\n",
      "Not Marking per-residue atom chain types.\n",
      "Marking per-residue atom chain types.\n",
      "  (Residues lacking connect0/connect1 - \n",
      "   these don't have chain types marked:\n",
      "\n",
      "\tres\ttotal affected\n",
      "\n",
      "\tCGLN\t1\n",
      "  )\n",
      " (no restraints)\n",
      "  Solute vdw bounding box:              36.283 17.842 27.687\n",
      "  Total bounding box for atom centers:  56.283 37.842 47.687\n",
      "  Solvent unit box:                     18.774 18.774 18.774\n",
      "  Total vdw box size:                   58.950 41.086 50.124 angstroms.\n",
      "  Volume: 121400.146 A^3 \n",
      "  Total mass 54626.466 amu,  Density 0.747 g/cc\n",
      "  Added 2897 residues.\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stapep/bin/teLeap: Warning!\n",
      "addIons: 1st Ion & target unit have charges of the same sign:\n",
      "     unit charge = 4.99999; ion1 charge = 1;\n",
      "     can't neutralize.\n",
      "5 Cl- ions required to neutralize.\n",
      "Adding 5 counter ions to \"pep\" using 1A grid\n",
      "Grid extends from solute vdw + 2.51  to  8.42\n",
      "Resolution:      1.00 Angstrom.\n",
      "Solvent present: replacing closest with ion\n",
      "\t when steric overlaps occur\n",
      "Calculating grid charges\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:parmed.structure:Adding bonds...\n",
      "INFO:parmed.structure:Adding angles...\n",
      "INFO:parmed.structure:Adding dihedrals...\n",
      "INFO:parmed.structure:Adding Ryckaert-Bellemans torsions...\n",
      "INFO:parmed.structure:Adding Urey-Bradleys...\n",
      "INFO:parmed.structure:Adding improper torsions...\n",
      "INFO:parmed.structure:Adding CMAP torsions...\n",
      "INFO:parmed.structure:Adding trigonal angle terms...\n",
      "INFO:parmed.structure:Adding out-of-plane bends...\n",
      "INFO:parmed.structure:Adding pi-torsions...\n",
      "INFO:parmed.structure:Adding stretch-bends...\n",
      "INFO:parmed.structure:Adding torsion-torsions...\n",
      "INFO:parmed.structure:Adding Nonbonded force...\n",
      "INFO:parmed.structure:Adding GB force...\n",
      "INFO:root:Running short time simulation for 100000 steps\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(Replacing solvent molecule)\n",
      "Placed Cl- in pep at (-1.49, 1.98, -11.50).\n",
      "(Replacing solvent molecule)\n",
      "Placed Cl- in pep at (-0.21, 8.28, 1.49).\n",
      "(Replacing solvent molecule)\n",
      "Placed Cl- in pep at (11.39, -8.47, 0.90).\n",
      "(Replacing solvent molecule)\n",
      "Placed Cl- in pep at (-7.27, 2.81, 6.37).\n",
      "(Replacing solvent molecule)\n",
      "Placed Cl- in pep at (-2.96, -8.14, 8.63).\n",
      "\n",
      "Done adding ions.\n",
      "Checking Unit.\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stapep/bin/teLeap: Warning!\n",
      "There is a bond of 18.617 angstroms between C2 and C2 atoms:\n",
      "-------  .R<PR8 5>.A<C2 10> and .R<PR3 14>.A<C2 9>\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stapep/bin/teLeap: Note.\n",
      "Ignoring the warning from Unit Checking.\n",
      "\n",
      "Building topology.\n",
      "Building atom parameters.\n",
      "Building bond parameters.\n",
      "Building angle parameters.\n",
      "Building proper torsion parameters.\n",
      "Building improper torsion parameters.\n",
      "old PREP-specified impropers:\n",
      " <NLE 1>:  CA   +M   C    O   \n",
      " <PR8 5>:  CA   +M   C    O   \n",
      " <AIB 9>:  CA   +M   C    O   \n",
      " <NLE 10>:  CA   +M   C    O   \n",
      " <NLE 12>:  CA   +M   C    O   \n",
      " <PR3 14>:  CA   +M   C    O   \n",
      " total 69 improper torsions applied\n",
      " 6 improper torsions in old prep form\n",
      "Building H-Bond parameters.\n",
      "Incorporating Non-Bonded adjustments.\n",
      "Not Marking per-residue atom chain types.\n",
      "Marking per-residue atom chain types.\n",
      "  (Residues lacking connect0/connect1 - \n",
      "   these don't have chain types marked:\n",
      "\n",
      "\tres\ttotal affected\n",
      "\n",
      "\tCGLN\t1\n",
      "\tWAT\t2892\n",
      "  )\n",
      " (no restraints)\n",
      "Writing pdb file: pep_solv.pdb\n",
      "   printing CRYST1 record to PDB file with box info\n",
      "\n",
      "/home/dahuilangda/Simulation/miniconda3/envs/stapep/bin/teLeap: Warning!\n",
      " Converting C-terminal residue name to PDB format: CGLN -> GLN\n",
      "\tQuit\n",
      "\n",
      "Exiting LEaP: Errors = 0; Warnings = 5; Notes = 2.\n",
      "Minimizing...\n",
      "Running...\n",
      "Done! Time elapsed: 32.56246280670166 s\n",
      "/tmp/44003a39-df5a-4eaf-b7e8-2fb96a1f6f42/traj.dcd /tmp/44003a39-df5a-4eaf-b7e8-2fb96a1f6f42/pep_vac.prmtop\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'data/homology_model.pdb'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from stapep.structure import Structure, AlignStructure\n",
    "\n",
    "# homology modeling\n",
    "st = Structure(verbose=True)\n",
    "st.generate_3d_structure_from_template(seq=seq, \n",
    "                                       output_pdb='data/homology_model.pdb', \n",
    "                                       template_pdb='data/template.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/3dmoljs_load.v0": "<div id=\"3dmolviewer_16950270124179935\"  style=\"position: relative; width: 640px; height: 480px\">\n        <p id=\"3dmolwarning_16950270124179935\" style=\"background-color:#ffcccc;color:black\">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>\n        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>\n        </div>\n<script>\n\nvar loadScriptAsync = function(uri){\n  return new Promise((resolve, reject) => {\n    //this is to ignore the existence of requirejs amd\n    var savedexports, savedmodule;\n    if (typeof exports !== 'undefined') savedexports = exports;\n    else exports = {}\n    if (typeof module !== 'undefined') savedmodule = module;\n    else module = {}\n\n    var tag = document.createElement('script');\n    tag.src = uri;\n    tag.async = true;\n    tag.onload = () => {\n        exports = savedexports;\n        module = savedmodule;\n        resolve();\n    };\n  var firstScriptTag = document.getElementsByTagName('script')[0];\n  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n});\n};\n\nif(typeof $3Dmolpromise === 'undefined') {\n$3Dmolpromise = null;\n  $3Dmolpromise = loadScriptAsync('https://cdnjs.cloudflare.com/ajax/libs/3Dmol/2.0.1/3Dmol-min.js');\n}\n\nvar viewer_16950270124179935 = null;\nvar warn = document.getElementById(\"3dmolwarning_16950270124179935\");\nif(warn) {\n    warn.parentNode.removeChild(warn);\n}\n$3Dmolpromise.then(function() {\nviewer_16950270124179935 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_16950270124179935\"),{backgroundColor:\"white\"});\nviewer_16950270124179935.zoomTo();\n\tviewer_16950270124179935.addModel(\"ATOM      1  H1  ACE     1      21.501   0.153  -0.049  1.00  0.00           H  \\nATOM      2  CH3 ACE     1      21.104  -0.394   0.815  1.00  0.00           C  \\nATOM      3  H2  ACE     1      20.853  -1.414   0.526  1.00  0.00           H  \\nATOM      4  H3  ACE     1      20.160   0.074   1.123  1.00  0.00           H  \\nATOM      5  C   ACE     1      22.222  -0.247   1.853  1.00  0.00           C  \\nATOM      6  O   ACE     1      22.760  -1.215   2.324  1.00  0.00           O  \\nATOM      7  N   NLE     2      22.539   0.971   2.160  1.00  0.00           N  \\nATOM      8  H13 NLE     2      22.010   1.683   1.683  1.00  0.00           H  \\nATOM      9  CA  NLE     2      23.564   1.463   3.103  1.00  0.00           C  \\nATOM     10  C4  NLE     2      24.601   2.320   2.381  1.00  0.00           C  \\nATOM     11  C3  NLE     2      25.411   1.553   1.269  1.00  0.00           C  \\nATOM     12  C2  NLE     2      26.402   2.452   0.455  1.00  0.00           C  \\nATOM     13  C1  NLE     2      27.755   2.263   1.089  1.00  0.00           C  \\nATOM     14  H1  NLE     2      27.969   1.186   1.126  1.00  0.00           H  \\nATOM     15  H2  NLE     2      27.714   2.666   2.040  1.00  0.00           H  \\nATOM     16  H3  NLE     2      28.517   2.803   0.438  1.00  0.00           H  \\nATOM     17  H4  NLE     2      26.607   2.074  -0.498  1.00  0.00           H  \\nATOM     18  H5  NLE     2      26.292   3.519   0.444  1.00  0.00           H  \\nATOM     19  H6  NLE     2      25.934   0.721   1.645  1.00  0.00           H  \\nATOM     20  H7  NLE     2      24.748   1.005   0.662  1.00  0.00           H  \\nATOM     21  H8  NLE     2      24.040   3.129   1.912  1.00  0.00           H  \\nATOM     22  H9  NLE     2      25.343   2.598   3.229  1.00  0.00           H  \\nATOM     23  H10 NLE     2      24.080   0.540   3.335  1.00  0.00           H  \\nATOM     24  C   NLE     2      23.015   2.009   4.421  1.00  0.00           C  \\nATOM     25  O   NLE     2      23.377   3.095   4.864  1.00  0.00           O  \\nATOM     26  N   ALA     3      21.947   1.331   4.839  1.00  0.00           N  \\nATOM     27  H   ALA     3      21.726   0.427   4.400  1.00  0.00           H  \\nATOM     28  CA  ALA     3      21.127   1.828   5.909  1.00  0.00           C  \\nATOM     29  HA  ALA     3      20.946   2.873   5.691  1.00  0.00           H  \\nATOM     30  CB  ALA     3      19.763   1.016   5.818  1.00  0.00           C  \\nATOM     31  HB1 ALA     3      19.904  -0.068   5.758  1.00  0.00           H  \\nATOM     32  HB2 ALA     3      19.034   1.106   6.716  1.00  0.00           H  \\nATOM     33  HB3 ALA     3      19.276   1.398   4.886  1.00  0.00           H  \\nATOM     34  C   ALA     3      21.730   1.855   7.340  1.00  0.00           C  \\nATOM     35  O   ALA     3      22.759   1.298   7.567  1.00  0.00           O  \\nATOM     36  N   THR     4      21.138   2.596   8.288  1.00  0.00           N  \\nATOM     37  H   THR     4      20.269   3.012   8.003  1.00  0.00           H  \\nATOM     38  CA  THR     4      21.573   2.754   9.672  1.00  0.00           C  \\nATOM     39  HA  THR     4      22.671   2.717   9.788  1.00  0.00           H  \\nATOM     40  CB  THR     4      21.110   4.144  10.110  1.00  0.00           C  \\nATOM     41  HB  THR     4      21.397   4.280  11.191  1.00  0.00           H  \\nATOM     42  CG2 THR     4      21.801   5.306   9.382  1.00  0.00           C  \\nATOM     43 HG21 THR     4      21.539   5.305   8.357  1.00  0.00           H  \\nATOM     44 HG22 THR     4      21.372   6.219   9.852  1.00  0.00           H  \\nATOM     45 HG23 THR     4      22.898   5.134   9.436  1.00  0.00           H  \\nATOM     46  OG1 THR     4      19.786   4.361  10.087  1.00  0.00           O  \\nATOM     47  HG1 THR     4      19.724   5.267  10.036  1.00  0.00           H  \\nATOM     48  C   THR     4      21.029   1.610  10.527  1.00  0.00           C  \\nATOM     49  O   THR     4      19.894   1.109  10.244  1.00  0.00           O  \\nATOM     50  N   PRO     5      21.705   1.123  11.580  1.00  0.00           N  \\nATOM     51  CD  PRO     5      23.086   1.438  11.860  1.00  0.00           C  \\nATOM     52  HD2 PRO     5      23.014   2.515  12.268  1.00  0.00           H  \\nATOM     53  HD3 PRO     5      23.700   1.509  10.939  1.00  0.00           H  \\nATOM     54  CG  PRO     5      23.598   0.447  12.977  1.00  0.00           C  \\nATOM     55  HG2 PRO     5      24.127   1.045  13.804  1.00  0.00           H  \\nATOM     56  HG3 PRO     5      24.215  -0.360  12.621  1.00  0.00           H  \\nATOM     57  CB  PRO     5      22.269  -0.056  13.545  1.00  0.00           C  \\nATOM     58  HB2 PRO     5      21.919   0.707  14.340  1.00  0.00           H  \\nATOM     59  HB3 PRO     5      22.355  -1.047  13.811  1.00  0.00           H  \\nATOM     60  CA  PRO     5      21.266  -0.007  12.394  1.00  0.00           C  \\nATOM     61  HA  PRO     5      21.355  -0.968  11.912  1.00  0.00           H  \\nATOM     62  C   PRO     5      19.811   0.086  12.967  1.00  0.00           C  \\nATOM     63  O   PRO     5      19.137  -0.914  12.973  1.00  0.00           O  \\nATOM     64  N   PR8     6      19.409   1.284  13.298  1.00  0.00           N  \\nATOM     65  H19 PR8     6      20.082   2.012  13.260  1.00  0.00           H  \\nATOM     66  CA  PR8     6      18.032   1.740  13.505  1.00  0.00           C  \\nATOM     67  C8  PR8     6      17.617   1.287  14.907  1.00  0.00           C  \\nATOM     68  C7  PR8     6      16.356   1.819  15.622  1.00  0.00           C  \\nATOM     69  C6  PR8     6      16.294   1.282  17.077  1.00  0.00           C  \\nATOM     70  C5  PR8     6      14.979   1.439  17.834  1.00  0.00           C  \\nATOM     71  C4  PR8     6      14.161   2.695  17.680  1.00  0.00           C  \\nATOM     72  C3  PR8     6      12.851   2.796  18.438  1.00  0.00           C  \\nATOM     73  C2  PR8     6      11.885   1.845  17.835  1.00  0.00           C  \\nATOM     74  H3  PR8     6      11.952   0.748  17.987  1.00  0.00           H  \\nATOM     75  H4  PR8     6      12.959   2.540  19.471  1.00  0.00           H  \\nATOM     76  H5  PR8     6      12.513   3.854  18.290  1.00  0.00           H  \\nATOM     77  H6  PR8     6      14.762   3.543  18.005  1.00  0.00           H  \\nATOM     78  H7  PR8     6      14.028   2.813  16.537  1.00  0.00           H  \\nATOM     79  H8  PR8     6      14.343   0.530  17.665  1.00  0.00           H  \\nATOM     80  H9  PR8     6      15.174   1.316  18.854  1.00  0.00           H  \\nATOM     81  H10 PR8     6      17.068   1.784  17.598  1.00  0.00           H  \\nATOM     82  H11 PR8     6      16.592   0.243  17.072  1.00  0.00           H  \\nATOM     83  H12 PR8     6      16.339   2.878  15.551  1.00  0.00           H  \\nATOM     84  H13 PR8     6      15.496   1.424  15.111  1.00  0.00           H  \\nATOM     85  H14 PR8     6      17.705   0.216  14.939  1.00  0.00           H  \\nATOM     86  H15 PR8     6      18.336   1.568  15.619  1.00  0.00           H  \\nATOM     87  C10 PR8     6      17.970   3.280  13.431  1.00  0.00           C  \\nATOM     88  H16 PR8     6      18.548   3.703  14.218  1.00  0.00           H  \\nATOM     89  H17 PR8     6      16.994   3.697  13.583  1.00  0.00           H  \\nATOM     90  H18 PR8     6      18.341   3.607  12.465  1.00  0.00           H  \\nATOM     91  C   PR8     6      17.035   1.168  12.464  1.00  0.00           C  \\nATOM     92  O   PR8     6      15.937   0.656  12.783  1.00  0.00           O  \\nATOM     93  N   ARG     7      17.346   1.196  11.162  1.00  0.00           N  \\nATOM     94  H   ARG     7      18.242   1.494  10.929  1.00  0.00           H  \\nATOM     95  CA  ARG     7      16.398   0.839  10.056  1.00  0.00           C  \\nATOM     96  HA  ARG     7      15.344   0.948  10.307  1.00  0.00           H  \\nATOM     97  CB  ARG     7      16.738   1.717   8.888  1.00  0.00           C  \\nATOM     98  HB2 ARG     7      17.847   1.751   8.748  1.00  0.00           H  \\nATOM     99  HB3 ARG     7      16.299   1.343   7.859  1.00  0.00           H  \\nATOM    100  CG  ARG     7      16.447   3.208   9.031  1.00  0.00           C  \\nATOM    101  HG2 ARG     7      17.145   3.590   9.792  1.00  0.00           H  \\nATOM    102  HG3 ARG     7      16.746   3.651   8.019  1.00  0.00           H  \\nATOM    103  CD  ARG     7      14.987   3.550   9.323  1.00  0.00           C  \\nATOM    104  HD2 ARG     7      14.754   3.218  10.312  1.00  0.00           H  \\nATOM    105  HD3 ARG     7      14.838   4.648   9.307  1.00  0.00           H  \\nATOM    106  NE  ARG     7      14.119   3.068   8.228  1.00  0.00           N  \\nATOM    107  HE  ARG     7      14.522   3.144   7.300  1.00  0.00           H  \\nATOM    108  CZ  ARG     7      12.965   2.516   8.270  1.00  0.00           C  \\nATOM    109  NH1 ARG     7      12.436   2.157   9.392  1.00  0.00           N  \\nATOM    110 HH11 ARG     7      12.910   2.516  10.257  1.00  0.00           H  \\nATOM    111 HH12 ARG     7      11.475   1.792   9.317  1.00  0.00           H  \\nATOM    112  NH2 ARG     7      12.346   2.166   7.177  1.00  0.00           N  \\nATOM    113 HH21 ARG     7      12.680   2.479   6.233  1.00  0.00           H  \\nATOM    114 HH22 ARG     7      11.449   1.666   7.270  1.00  0.00           H  \\nATOM    115  C   ARG     7      16.519  -0.593   9.635  1.00  0.00           C  \\nATOM    116  O   ARG     7      15.531  -1.146   9.096  1.00  0.00           O  \\nATOM    117  N   ARG     8      17.631  -1.317   9.775  1.00  0.00           N  \\nATOM    118  H   ARG     8      18.392  -0.772  10.116  1.00  0.00           H  \\nATOM    119  CA  ARG     8      17.670  -2.815   9.744  1.00  0.00           C  \\nATOM    120  HA  ARG     8      17.129  -3.134   8.859  1.00  0.00           H  \\nATOM    121  CB  ARG     8      19.139  -3.246   9.584  1.00  0.00           C  \\nATOM    122  HB2 ARG     8      19.683  -3.029  10.517  1.00  0.00           H  \\nATOM    123  HB3 ARG     8      19.178  -4.265   9.412  1.00  0.00           H  \\nATOM    124  CG  ARG     8      19.940  -2.610   8.399  1.00  0.00           C  \\nATOM    125  HG2 ARG     8      19.380  -2.728   7.524  1.00  0.00           H  \\nATOM    126  HG3 ARG     8      20.216  -1.540   8.559  1.00  0.00           H  \\nATOM    127  CD  ARG     8      21.268  -3.351   8.255  1.00  0.00           C  \\nATOM    128  HD2 ARG     8      21.758  -3.456   9.130  1.00  0.00           H  \\nATOM    129  HD3 ARG     8      21.007  -4.437   7.948  1.00  0.00           H  \\nATOM    130  NE  ARG     8      22.105  -2.670   7.242  1.00  0.00           N  \\nATOM    131  HE  ARG     8      22.658  -1.893   7.618  1.00  0.00           H  \\nATOM    132  CZ  ARG     8      21.931  -2.602   5.881  1.00  0.00           C  \\nATOM    133  NH1 ARG     8      21.127  -3.296   5.113  1.00  0.00           N  \\nATOM    134 HH11 ARG     8      20.746  -4.163   5.485  1.00  0.00           H  \\nATOM    135 HH12 ARG     8      21.114  -3.200   4.072  1.00  0.00           H  \\nATOM    136  NH2 ARG     8      22.666  -1.667   5.249  1.00  0.00           N  \\nATOM    137 HH21 ARG     8      23.311  -1.076   5.775  1.00  0.00           H  \\nATOM    138 HH22 ARG     8      22.906  -1.834   4.263  1.00  0.00           H  \\nATOM    139  C   ARG     8      16.905  -3.521  10.897  1.00  0.00           C  \\nATOM    140  O   ARG     8      16.109  -4.452  10.597  1.00  0.00           O  \\nATOM    141  N   ARG     9      17.076  -3.031  12.180  1.00  0.00           N  \\nATOM    142  H   ARG     9      17.641  -2.130  12.125  1.00  0.00           H  \\nATOM    143  CA  ARG     9      16.452  -3.634  13.373  1.00  0.00           C  \\nATOM    144  HA  ARG     9      16.624  -4.748  13.288  1.00  0.00           H  \\nATOM    145  CB  ARG     9      17.284  -3.302  14.656  1.00  0.00           C  \\nATOM    146  HB2 ARG     9      17.393  -2.232  14.720  1.00  0.00           H  \\nATOM    147  HB3 ARG     9      16.832  -3.802  15.550  1.00  0.00           H  \\nATOM    148  CG  ARG     9      18.704  -3.839  14.626  1.00  0.00           C  \\nATOM    149  HG2 ARG     9      18.779  -4.957  14.583  1.00  0.00           H  \\nATOM    150  HG3 ARG     9      19.212  -3.434  13.804  1.00  0.00           H  \\nATOM    151  CD  ARG     9      19.377  -3.422  15.999  1.00  0.00           C  \\nATOM    152  HD2 ARG     9      20.434  -3.777  15.988  1.00  0.00           H  \\nATOM    153  HD3 ARG     9      18.981  -3.887  16.876  1.00  0.00           H  \\nATOM    154  NE  ARG     9      19.587  -1.961  16.078  1.00  0.00           N  \\nATOM    155  HE  ARG     9      19.564  -1.482  15.174  1.00  0.00           H  \\nATOM    156  CZ  ARG     9      20.192  -1.322  17.115  1.00  0.00           C  \\nATOM    157  NH1 ARG     9      20.382  -1.892  18.334  1.00  0.00           N  \\nATOM    158 HH11 ARG     9      20.000  -2.760  18.507  1.00  0.00           H  \\nATOM    159 HH12 ARG     9      20.855  -1.332  19.090  1.00  0.00           H  \\nATOM    160  NH2 ARG     9      20.614  -0.090  16.993  1.00  0.00           N  \\nATOM    161 HH21 ARG     9      20.460   0.389  16.141  1.00  0.00           H  \\nATOM    162 HH22 ARG     9      20.990   0.423  17.758  1.00  0.00           H  \\nATOM    163  C   ARG     9      14.960  -3.362  13.497  1.00  0.00           C  \\nATOM    164  O   ARG     9      14.254  -4.150  14.007  1.00  0.00           O  \\nATOM    165  N   AIB    10      14.419  -2.295  12.879  1.00  0.00           N  \\nATOM    166  H7  AIB    10      15.092  -1.680  12.556  1.00  0.00           H  \\nATOM    167  CA  AIB    10      13.054  -1.732  13.118  1.00  0.00           C  \\nATOM    168  C1  AIB    10      13.128  -0.675  14.244  1.00  0.00           C  \\nATOM    169  H1  AIB    10      13.815   0.029  13.897  1.00  0.00           H  \\nATOM    170  H2  AIB    10      13.552  -1.230  15.046  1.00  0.00           H  \\nATOM    171  H3  AIB    10      12.218  -0.296  14.663  1.00  0.00           H  \\nATOM    172  C3  AIB    10      12.564  -1.049  11.886  1.00  0.00           C  \\nATOM    173  H4  AIB    10      12.432  -1.786  11.069  1.00  0.00           H  \\nATOM    174  H5  AIB    10      13.317  -0.366  11.560  1.00  0.00           H  \\nATOM    175  H6  AIB    10      11.684  -0.468  12.181  1.00  0.00           H  \\nATOM    176  C   AIB    10      11.922  -2.756  13.392  1.00  0.00           C  \\nATOM    177  O   AIB    10      11.393  -2.787  14.546  1.00  0.00           O  \\nATOM    178  N   NLE    11      11.570  -3.681  12.485  1.00  0.00           N  \\nATOM    179  H13 NLE    11      12.172  -3.646  11.689  1.00  0.00           H  \\nATOM    180  CA  NLE    11      10.484  -4.721  12.513  1.00  0.00           C  \\nATOM    181  C4  NLE    11      10.856  -5.714  11.384  1.00  0.00           C  \\nATOM    182  C3  NLE    11      10.695  -5.112   9.981  1.00  0.00           C  \\nATOM    183  C2  NLE    11      10.925  -6.187   8.958  1.00  0.00           C  \\nATOM    184  C1  NLE    11      10.702  -5.548   7.560  1.00  0.00           C  \\nATOM    185  H1  NLE    11      11.524  -4.979   7.201  1.00  0.00           H  \\nATOM    186  H2  NLE    11       9.808  -4.891   7.529  1.00  0.00           H  \\nATOM    187  H3  NLE    11      10.523  -6.336   6.806  1.00  0.00           H  \\nATOM    188  H4  NLE    11      11.942  -6.628   9.103  1.00  0.00           H  \\nATOM    189  H5  NLE    11      10.159  -6.950   9.108  1.00  0.00           H  \\nATOM    190  H6  NLE    11       9.666  -4.795   9.964  1.00  0.00           H  \\nATOM    191  H7  NLE    11      11.432  -4.240   9.870  1.00  0.00           H  \\nATOM    192  H8  NLE    11      11.870  -5.983  11.575  1.00  0.00           H  \\nATOM    193  H9  NLE    11      10.343  -6.703  11.405  1.00  0.00           H  \\nATOM    194  H10 NLE    11       9.535  -4.278  12.336  1.00  0.00           H  \\nATOM    195  C   NLE    11      10.414  -5.675  13.792  1.00  0.00           C  \\nATOM    196  O   NLE    11       9.316  -6.110  14.201  1.00  0.00           O  \\nATOM    197  N   LEU    12      11.506  -5.781  14.497  1.00  0.00           N  \\nATOM    198  H   LEU    12      12.248  -5.162  14.184  1.00  0.00           H  \\nATOM    199  CA  LEU    12      11.711  -6.564  15.735  1.00  0.00           C  \\nATOM    200  HA  LEU    12      10.965  -7.376  15.654  1.00  0.00           H  \\nATOM    201  CB  LEU    12      13.085  -7.270  15.742  1.00  0.00           C  \\nATOM    202  HB2 LEU    12      13.845  -6.605  16.107  1.00  0.00           H  \\nATOM    203  HB3 LEU    12      13.018  -8.087  16.406  1.00  0.00           H  \\nATOM    204  CG  LEU    12      13.464  -8.033  14.431  1.00  0.00           C  \\nATOM    205  HG  LEU    12      13.483  -7.349  13.604  1.00  0.00           H  \\nATOM    206  CD1 LEU    12      14.841  -8.682  14.586  1.00  0.00           C  \\nATOM    207 HD11 LEU    12      15.548  -7.951  14.871  1.00  0.00           H  \\nATOM    208 HD12 LEU    12      14.774  -9.507  15.356  1.00  0.00           H  \\nATOM    209 HD13 LEU    12      15.053  -9.027  13.599  1.00  0.00           H  \\nATOM    210  CD2 LEU    12      12.479  -9.174  14.141  1.00  0.00           C  \\nATOM    211 HD21 LEU    12      11.507  -8.869  13.799  1.00  0.00           H  \\nATOM    212 HD22 LEU    12      12.853  -9.905  13.427  1.00  0.00           H  \\nATOM    213 HD23 LEU    12      12.450  -9.824  14.950  1.00  0.00           H  \\nATOM    214  C   LEU    12      11.399  -5.768  17.012  1.00  0.00           C  \\nATOM    215  O   LEU    12      11.264  -6.375  18.076  1.00  0.00           O  \\nATOM    216  N   NLE    13      11.332  -4.428  16.989  1.00  0.00           N  \\nATOM    217  H13 NLE    13      11.466  -3.977  16.155  1.00  0.00           H  \\nATOM    218  CA  NLE    13      11.294  -3.558  18.170  1.00  0.00           C  \\nATOM    219  C4  NLE    13      12.485  -2.551  18.219  1.00  0.00           C  \\nATOM    220  C3  NLE    13      13.886  -3.151  18.452  1.00  0.00           C  \\nATOM    221  C2  NLE    13      14.894  -2.060  18.180  1.00  0.00           C  \\nATOM    222  C1  NLE    13      16.321  -2.488  18.579  1.00  0.00           C  \\nATOM    223  H1  NLE    13      16.826  -3.080  17.859  1.00  0.00           H  \\nATOM    224  H2  NLE    13      17.044  -1.602  18.662  1.00  0.00           H  \\nATOM    225  H3  NLE    13      16.260  -2.956  19.515  1.00  0.00           H  \\nATOM    226  H4  NLE    13      14.864  -1.742  17.160  1.00  0.00           H  \\nATOM    227  H5  NLE    13      14.534  -1.279  18.831  1.00  0.00           H  \\nATOM    228  H6  NLE    13      13.748  -3.423  19.463  1.00  0.00           H  \\nATOM    229  H7  NLE    13      13.975  -4.075  17.819  1.00  0.00           H  \\nATOM    230  H8  NLE    13      12.528  -1.949  17.434  1.00  0.00           H  \\nATOM    231  H9  NLE    13      12.245  -1.834  19.029  1.00  0.00           H  \\nATOM    232  H10 NLE    13      11.301  -4.240  19.081  1.00  0.00           H  \\nATOM    233  C   NLE    13       9.902  -2.867  18.227  1.00  0.00           C  \\nATOM    234  O   NLE    13       9.135  -2.899  17.205  1.00  0.00           O  \\nATOM    235  N   ARG    14       9.542  -2.208  19.361  1.00  0.00           N  \\nATOM    236  H   ARG    14      10.266  -2.144  20.074  1.00  0.00           H  \\nATOM    237  CA  ARG    14       8.226  -1.653  19.506  1.00  0.00           C  \\nATOM    238  HA  ARG    14       7.469  -2.428  19.282  1.00  0.00           H  \\nATOM    239  CB  ARG    14       7.919  -1.134  20.956  1.00  0.00           C  \\nATOM    240  HB2 ARG    14       8.214  -1.899  21.755  1.00  0.00           H  \\nATOM    241  HB3 ARG    14       8.411  -0.181  21.059  1.00  0.00           H  \\nATOM    242  CG  ARG    14       6.485  -0.875  21.383  1.00  0.00           C  \\nATOM    243  HG2 ARG    14       5.885  -0.860  20.447  1.00  0.00           H  \\nATOM    244  HG3 ARG    14       6.172  -1.717  22.052  1.00  0.00           H  \\nATOM    245  CD  ARG    14       6.197   0.395  22.182  1.00  0.00           C  \\nATOM    246  HD2 ARG    14       6.398   1.174  21.462  1.00  0.00           H  \\nATOM    247  HD3 ARG    14       5.059   0.427  22.400  1.00  0.00           H  \\nATOM    248  NE  ARG    14       6.973   0.501  23.451  1.00  0.00           N  \\nATOM    249  HE  ARG    14       7.995   0.240  23.437  1.00  0.00           H  \\nATOM    250  CZ  ARG    14       6.645   1.107  24.600  1.00  0.00           C  \\nATOM    251  NH1 ARG    14       5.545   1.827  24.756  1.00  0.00           N  \\nATOM    252 HH11 ARG    14       4.875   1.880  23.963  1.00  0.00           H  \\nATOM    253 HH12 ARG    14       5.450   2.352  25.683  1.00  0.00           H  \\nATOM    254  NH2 ARG    14       7.381   0.982  25.668  1.00  0.00           N  \\nATOM    255 HH21 ARG    14       8.255   0.570  25.545  1.00  0.00           H  \\nATOM    256 HH22 ARG    14       7.237   1.716  26.316  1.00  0.00           H  \\nATOM    257  C   ARG    14       7.870  -0.643  18.389  1.00  0.00           C  \\nATOM    258  O   ARG    14       6.723  -0.563  18.012  1.00  0.00           O  \\nATOM    259  N   PR3    15       8.768   0.253  17.876  1.00  0.00           N  \\nATOM    260  H10 PR3    15       9.569   0.482  18.411  1.00  0.00           H  \\nATOM    261  CA  PR3    15       8.653   1.187  16.764  1.00  0.00           C  \\nATOM    262  C5  PR3    15       8.122   2.484  17.435  1.00  0.00           C  \\nATOM    263  H6  PR3    15       8.808   2.772  18.248  1.00  0.00           H  \\nATOM    264  H7  PR3    15       7.047   2.397  17.638  1.00  0.00           H  \\nATOM    265  H8  PR3    15       8.150   3.276  16.618  1.00  0.00           H  \\nATOM    266  C3  PR3    15      10.067   1.361  16.124  1.00  0.00           C  \\nATOM    267  C2  PR3    15      10.905   2.281  17.015  1.00  0.00           C  \\nATOM    268  H3  PR3    15      10.705   3.326  16.862  1.00  0.00           H  \\nATOM    269  H4  PR3    15      10.483   0.402  15.979  1.00  0.00           H  \\nATOM    270  H5  PR3    15      10.001   1.796  15.103  1.00  0.00           H  \\nATOM    271  C   PR3    15       7.699   0.645  15.636  1.00  0.00           C  \\nATOM    272  O   PR3    15       6.760   1.348  15.198  1.00  0.00           O  \\nATOM    273  N   PHE    16       7.888  -0.652  15.247  1.00  0.00           N  \\nATOM    274  H   PHE    16       8.671  -1.196  15.676  1.00  0.00           H  \\nATOM    275  CA  PHE    16       7.228  -1.226  14.045  1.00  0.00           C  \\nATOM    276  HA  PHE    16       7.000  -0.501  13.222  1.00  0.00           H  \\nATOM    277  CB  PHE    16       8.057  -2.400  13.485  1.00  0.00           C  \\nATOM    278  HB2 PHE    16       8.936  -2.062  13.054  1.00  0.00           H  \\nATOM    279  HB3 PHE    16       8.320  -3.175  14.328  1.00  0.00           H  \\nATOM    280  CG  PHE    16       7.258  -3.219  12.494  1.00  0.00           C  \\nATOM    281  CD1 PHE    16       6.707  -4.454  12.865  1.00  0.00           C  \\nATOM    282  HD1 PHE    16       6.878  -4.802  13.860  1.00  0.00           H  \\nATOM    283  CE1 PHE    16       5.923  -5.190  12.010  1.00  0.00           C  \\nATOM    284  HE1 PHE    16       5.520  -6.240  12.196  1.00  0.00           H  \\nATOM    285  CZ  PHE    16       5.639  -4.592  10.735  1.00  0.00           C  \\nATOM    286  HZ  PHE    16       4.916  -5.148  10.073  1.00  0.00           H  \\nATOM    287  CE2 PHE    16       6.161  -3.353  10.313  1.00  0.00           C  \\nATOM    288  HE2 PHE    16       5.964  -2.969   9.365  1.00  0.00           H  \\nATOM    289  CD2 PHE    16       6.949  -2.663  11.230  1.00  0.00           C  \\nATOM    290  HD2 PHE    16       7.277  -1.634  10.991  1.00  0.00           H  \\nATOM    291  C   PHE    16       5.796  -1.629  14.479  1.00  0.00           C  \\nATOM    292  O   PHE    16       4.806  -1.383  13.742  1.00  0.00           O  \\nATOM    293  N   LYS    17       5.618  -2.184  15.668  1.00  0.00           N  \\nATOM    294  H   LYS    17       6.465  -2.522  16.041  1.00  0.00           H  \\nATOM    295  CA  LYS    17       4.362  -2.622  16.249  1.00  0.00           C  \\nATOM    296  HA  LYS    17       4.052  -3.425  15.610  1.00  0.00           H  \\nATOM    297  CB  LYS    17       4.794  -3.373  17.573  1.00  0.00           C  \\nATOM    298  HB2 LYS    17       5.508  -4.109  17.263  1.00  0.00           H  \\nATOM    299  HB3 LYS    17       5.324  -2.692  18.284  1.00  0.00           H  \\nATOM    300  CG  LYS    17       3.661  -4.139  18.302  1.00  0.00           C  \\nATOM    301  HG2 LYS    17       4.196  -4.405  19.247  1.00  0.00           H  \\nATOM    302  HG3 LYS    17       2.914  -3.361  18.541  1.00  0.00           H  \\nATOM    303  CD  LYS    17       3.055  -5.270  17.445  1.00  0.00           C  \\nATOM    304  HD2 LYS    17       2.180  -4.908  16.870  1.00  0.00           H  \\nATOM    305  HD3 LYS    17       3.786  -5.675  16.804  1.00  0.00           H  \\nATOM    306  CE  LYS    17       2.518  -6.355  18.359  1.00  0.00           C  \\nATOM    307  HE2 LYS    17       3.240  -6.690  18.912  1.00  0.00           H  \\nATOM    308  HE3 LYS    17       1.694  -5.866  18.953  1.00  0.00           H  \\nATOM    309  NZ  LYS    17       1.948  -7.537  17.661  1.00  0.00           N  \\nATOM    310  HZ1 LYS    17       1.419  -8.092  18.377  1.00  0.00           H  \\nATOM    311  HZ2 LYS    17       1.240  -7.312  17.058  1.00  0.00           H  \\nATOM    312  HZ3 LYS    17       2.616  -8.208  17.253  1.00  0.00           H  \\nATOM    313  C   LYS    17       3.299  -1.564  16.466  1.00  0.00           C  \\nATOM    314  O   LYS    17       2.071  -1.880  16.394  1.00  0.00           O  \\nATOM    315  N   ARG    18       3.689  -0.283  16.646  1.00  0.00           N  \\nATOM    316  H   ARG    18       4.659  -0.174  16.721  1.00  0.00           H  \\nATOM    317  CA  ARG    18       2.809   0.890  16.767  1.00  0.00           C  \\nATOM    318  HA  ARG    18       1.941   0.726  17.421  1.00  0.00           H  \\nATOM    319  CB  ARG    18       3.640   2.087  17.231  1.00  0.00           C  \\nATOM    320  HB2 ARG    18       4.013   1.874  18.271  1.00  0.00           H  \\nATOM    321  HB3 ARG    18       4.584   2.162  16.671  1.00  0.00           H  \\nATOM    322  CG  ARG    18       2.946   3.478  17.026  1.00  0.00           C  \\nATOM    323  HG2 ARG    18       2.978   3.732  15.946  1.00  0.00           H  \\nATOM    324  HG3 ARG    18       1.906   3.368  17.432  1.00  0.00           H  \\nATOM    325  CD  ARG    18       3.709   4.579  17.758  1.00  0.00           C  \\nATOM    326  HD2 ARG    18       3.524   4.399  18.803  1.00  0.00           H  \\nATOM    327  HD3 ARG    18       4.874   4.426  17.704  1.00  0.00           H  \\nATOM    328  NE  ARG    18       3.319   5.924  17.407  1.00  0.00           N  \\nATOM    329  HE  ARG    18       3.910   6.350  16.719  1.00  0.00           H  \\nATOM    330  CZ  ARG    18       2.397   6.659  17.982  1.00  0.00           C  \\nATOM    331  NH1 ARG    18       1.692   6.179  19.007  1.00  0.00           N  \\nATOM    332 HH11 ARG    18       1.628   5.212  19.186  1.00  0.00           H  \\nATOM    333 HH12 ARG    18       1.001   6.815  19.275  1.00  0.00           H  \\nATOM    334  NH2 ARG    18       2.136   7.902  17.692  1.00  0.00           N  \\nATOM    335 HH21 ARG    18       2.658   8.303  16.929  1.00  0.00           H  \\nATOM    336 HH22 ARG    18       1.474   8.395  18.263  1.00  0.00           H  \\nATOM    337  C   ARG    18       2.119   1.195  15.415  1.00  0.00           C  \\nATOM    338  O   ARG    18       1.140   1.917  15.356  1.00  0.00           O  \\nATOM    339  N   LEU    19       2.555   0.651  14.282  1.00  0.00           N  \\nATOM    340  H   LEU    19       3.268  -0.049  14.375  1.00  0.00           H  \\nATOM    341  CA  LEU    19       2.021   0.911  12.975  1.00  0.00           C  \\nATOM    342  HA  LEU    19       1.495   1.826  13.129  1.00  0.00           H  \\nATOM    343  CB  LEU    19       3.209   1.161  12.010  1.00  0.00           C  \\nATOM    344  HB2 LEU    19       3.775   0.246  11.780  1.00  0.00           H  \\nATOM    345  HB3 LEU    19       2.705   1.506  11.100  1.00  0.00           H  \\nATOM    346  CG  LEU    19       4.417   2.178  12.434  1.00  0.00           C  \\nATOM    347  HG  LEU    19       4.764   1.947  13.423  1.00  0.00           H  \\nATOM    348  CD1 LEU    19       5.608   2.054  11.424  1.00  0.00           C  \\nATOM    349 HD11 LEU    19       6.344   2.741  11.843  1.00  0.00           H  \\nATOM    350 HD12 LEU    19       5.947   1.009  11.508  1.00  0.00           H  \\nATOM    351 HD13 LEU    19       5.288   2.245  10.433  1.00  0.00           H  \\nATOM    352  CD2 LEU    19       3.882   3.665  12.405  1.00  0.00           C  \\nATOM    353 HD21 LEU    19       3.816   4.040  11.398  1.00  0.00           H  \\nATOM    354 HD22 LEU    19       2.798   3.619  12.775  1.00  0.00           H  \\nATOM    355 HD23 LEU    19       4.513   4.354  13.079  1.00  0.00           H  \\nATOM    356  C   LEU    19       1.034  -0.047  12.338  1.00  0.00           C  \\nATOM    357  O   LEU    19       0.380   0.372  11.391  1.00  0.00           O  \\nATOM    358  N   GLN    20       0.827  -1.180  12.986  1.00  0.00           N  \\nATOM    359  H   GLN    20       1.382  -1.353  13.752  1.00  0.00           H  \\nATOM    360  CA  GLN    20      -0.242  -2.121  12.675  1.00  0.00           C  \\nATOM    361  HA  GLN    20      -0.494  -2.137  11.636  1.00  0.00           H  \\nATOM    362  CB  GLN    20       0.354  -3.555  13.057  1.00  0.00           C  \\nATOM    363  HB2 GLN    20       0.455  -3.690  14.106  1.00  0.00           H  \\nATOM    364  HB3 GLN    20      -0.438  -4.311  12.796  1.00  0.00           H  \\nATOM    365  CG  GLN    20       1.658  -4.028  12.324  1.00  0.00           C  \\nATOM    366  HG2 GLN    20       1.533  -3.719  11.271  1.00  0.00           H  \\nATOM    367  HG3 GLN    20       2.590  -3.471  12.675  1.00  0.00           H  \\nATOM    368  CD  GLN    20       1.975  -5.498  12.598  1.00  0.00           C  \\nATOM    369  OE1 GLN    20       2.032  -5.993  13.692  1.00  0.00           O  \\nATOM    370  NE2 GLN    20       2.118  -6.273  11.543  1.00  0.00           N  \\nATOM    371 HE21 GLN    20       2.528  -7.091  11.727  1.00  0.00           H  \\nATOM    372 HE22 GLN    20       1.978  -5.822  10.593  1.00  0.00           H  \\nATOM    373  C   GLN    20      -1.493  -1.846  13.452  1.00  0.00           C  \\nATOM    374  O   GLN    20      -2.524  -2.364  13.064  1.00  0.00           O  \\nATOM    375  OXT GLN    20      -1.470  -1.161  14.497  1.00  0.00           O  \\nTER     376      GLN    20 \\nEND   \\n\");\n\tviewer_16950270124179935.setStyle({\"cartoon\": {\"color\": \"spectrum\"}});\n\tviewer_16950270124179935.addModel(\"HEADER                                            18-OCT-22                     \\nTITLE     ESMFOLD V1 PREDICTION FOR INPUT\\nREMARK   1                                                                      \\nREMARK   1 REFERENCE 1                                                          \\nREMARK   1  AUTH   ZEMING LIN, HALIL AKIN, ROSHAN RAO, BRIAN HIE, ZHONGKAI ZHU,\\nREMARK   1  AUTH 2 WENTING LU, NIKITA SMETANIN, ROBERT VERKUIL, ORI KABELI,\\nREMARK   1  AUTH 3 YANIV SHMUELI, ALLAN DOS SANTOS COSTA,\\nREMARK   1  AUTH 4 MARYAM FAZEL-ZARANDI, TOM SERCU, SALVATORE CANDIDO,\\nREMARK   1  AUTH 5 ALEXANDER RIVES\\nREMARK   1  TITL   EVOLUTIONARY-SCALE PREDICTION OF ATOMIC LEVEL PROTEIN\\nREMARK   1  TITL 2 STRUCTURE WITH A LANGUAGE MODEL\\nREMARK   1  REF                                                                 \\nREMARK   1  REFN                                                                \\nREMARK   1  PMID                                                                \\nREMARK   1  DOI    10.1101/2022.07.20.500902                                    \\nREMARK   1                                                                      \\nREMARK   1 LICENSE AND DISCLAIMERS                \\nREMARK   1 ESM METAGENOMIC ATLAS DATA IS AVAILABLE UNDER\\nREMARK   1 A CC-BY-4.0 LICENSE FOR ACADEMIC AND COMMERCIAL USE.\\nREMARK   1 COPYRIGHT (C) META PLATFORMS, INC. ALL RIGHTS RESERVED.\\nREMARK   1 USE OF THE ESM METAGENOMIC ATLAS DATA IS SUBJECT\\nREMARK   1 TO THE META OPEN SOURCE TERMS OF USE AND PRIVACY POLICY.\\nATOM    168  C   ACE A   0       9.976   8.120   7.984  1.00  0.00           C  \\nATOM      1  N   NLE A   1       8.593   7.504   7.797  1.00  0.74           N  \\nATOM      2  CA  NLE A   1       8.649   6.048   7.899  1.00  0.74           C  \\nATOM      3  C   NLE A   1       7.266   5.432   7.712  1.00  0.76           C  \\nATOM      5  O   NLE A   1       6.254   6.043   8.064  1.00  0.67           O  \\nATOM      6  N   ALA A   2       7.248   4.631   6.869  1.00  0.79           N  \\nATOM      7  CA  ALA A   2       5.925   4.058   6.634  1.00  0.80           C  \\nATOM      8  C   ALA A   2       5.507   3.151   7.787  1.00  0.79           C  \\nATOM      9  CB  ALA A   2       5.907   3.284   5.318  1.00  0.74           C  \\nATOM     10  O   ALA A   2       6.253   2.249   8.177  1.00  0.74           O  \\nATOM     11  N   THR A   3       4.679   3.521   8.593  1.00  0.87           N  \\nATOM     12  CA  THR A   3       4.083   2.690   9.633  1.00  0.89           C  \\nATOM     13  C   THR A   3       3.488   1.419   9.035  1.00  0.88           C  \\nATOM     14  CB  THR A   3       2.992   3.457  10.404  1.00  0.83           C  \\nATOM     15  O   THR A   3       3.190   1.369   7.840  1.00  0.85           O  \\nATOM     16  CG2 THR A   3       3.528   4.778  10.946  1.00  0.63           C  \\nATOM     17  OG1 THR A   3       1.893   3.724   9.525  1.00  0.65           O  \\nATOM     18  N   PRO A   4       3.551   0.253   9.836  1.00  0.87           N  \\nATOM     19  CA  PRO A   4       2.940  -0.979   9.331  1.00  0.88           C  \\nATOM     20  C   PRO A   4       1.540  -0.754   8.762  1.00  0.88           C  \\nATOM     21  CB  PRO A   4       2.888  -1.880  10.567  1.00  0.86           C  \\nATOM     22  O   PRO A   4       1.173  -1.365   7.756  1.00  0.87           O  \\nATOM     23  CG  PRO A   4       3.878  -1.284  11.515  1.00  0.79           C  \\nATOM     24  CD  PRO A   4       3.979   0.188  11.237  1.00  0.78           C  \\nATOM     25  N   ALA A   5       0.703   0.219   9.323  1.00  0.90           N  \\nATOM     26  CA  ALA A   5      -0.652   0.461   8.832  1.00  0.90           C  \\nATOM     27  C   ALA A   5      -0.627   1.087   7.441  1.00  0.90           C  \\nATOM     28  CB  ALA A   5      -1.416   1.358   9.802  1.00  0.88           C  \\nATOM     29  O   ALA A   5      -1.409   0.703   6.567  1.00  0.88           O  \\nATOM     30  N   ARG A   6       0.273   2.127   7.209  1.00  0.91           N  \\nATOM     31  CA  ARG A   6       0.365   2.756   5.896  1.00  0.91           C  \\nATOM     32  C   ARG A   6       0.820   1.754   4.839  1.00  0.90           C  \\nATOM     33  CB  ARG A   6       1.325   3.947   5.936  1.00  0.89           C  \\nATOM     34  O   ARG A   6       0.310   1.753   3.717  1.00  0.89           O  \\nATOM     35  CG  ARG A   6       0.713   5.212   6.515  1.00  0.82           C  \\nATOM     36  CD  ARG A   6       1.687   6.381   6.472  1.00  0.80           C  \\nATOM     37  NE  ARG A   6       1.118   7.580   7.080  1.00  0.75           N  \\nATOM     38  NH1 ARG A   6       3.089   8.785   7.044  1.00  0.64           N  \\nATOM     39  NH2 ARG A   6       1.179   9.727   7.895  1.00  0.62           N  \\nATOM     40  CZ  ARG A   6       1.797   8.695   7.338  1.00  0.72           C  \\nATOM     41  N   ARG A   7       1.896   0.856   5.232  1.00  0.91           N  \\nATOM     42  CA  ARG A   7       2.370  -0.125   4.261  1.00  0.91           C  \\nATOM     43  C   ARG A   7       1.259  -1.096   3.878  1.00  0.90           C  \\nATOM     44  CB  ARG A   7       3.571  -0.895   4.817  1.00  0.89           C  \\nATOM     45  O   ARG A   7       1.124  -1.462   2.708  1.00  0.89           O  \\nATOM     46  CG  ARG A   7       4.846  -0.071   4.900  1.00  0.82           C  \\nATOM     47  CD  ARG A   7       6.041  -0.921   5.308  1.00  0.80           C  \\nATOM     48  NE  ARG A   7       7.219  -0.102   5.575  1.00  0.74           N  \\nATOM     49  NH1 ARG A   7       8.464  -1.821   6.487  1.00  0.62           N  \\nATOM     50  NH2 ARG A   7       9.355   0.285   6.327  1.00  0.61           N  \\nATOM     51  CZ  ARG A   7       8.343  -0.548   6.129  1.00  0.70           C  \\nATOM     52  N   ARG A   8       0.495  -1.511   4.920  1.00  0.91           N  \\nATOM     53  CA  ARG A   8      -0.614  -2.428   4.677  1.00  0.91           C  \\nATOM     54  C   ARG A   8      -1.666  -1.788   3.777  1.00  0.90           C  \\nATOM     55  CB  ARG A   8      -1.250  -2.865   5.998  1.00  0.89           C  \\nATOM     56  O   ARG A   8      -2.203  -2.440   2.879  1.00  0.89           O  \\nATOM     57  CG  ARG A   8      -2.207  -4.039   5.863  1.00  0.80           C  \\nATOM     58  CD  ARG A   8      -2.725  -4.503   7.217  1.00  0.77           C  \\nATOM     59  NE  ARG A   8      -1.660  -5.076   8.034  1.00  0.70           N  \\nATOM     60  NH1 ARG A   8      -1.997  -7.247   7.318  1.00  0.56           N  \\nATOM     61  NH2 ARG A   8      -0.351  -6.781   8.844  1.00  0.54           N  \\nATOM     62  CZ  ARG A   8      -1.339  -6.367   8.064  1.00  0.69           C  \\nATOM     63  N   AIB A   9      -2.012  -0.574   4.062  1.00  0.91           N  \\nATOM     64  CA  AIB A   9      -3.006   0.120   3.246  1.00  0.91           C  \\nATOM     65  C   AIB A   9      -2.528   0.264   1.804  1.00  0.90           C  \\nATOM     67  O   AIB A   9      -3.312   0.109   0.865  1.00  0.89           O  \\nATOM     68  N   NLE A  10      -1.218   0.603   1.593  1.00  0.91           N  \\nATOM     69  CA  NLE A  10      -0.664   0.744   0.249  1.00  0.91           C  \\nATOM     70  C   NLE A  10      -0.717  -0.580  -0.507  1.00  0.91           C  \\nATOM     72  O   NLE A  10      -1.010  -0.607  -1.705  1.00  0.89           O  \\nATOM     73  N   LEU A  11      -0.379  -1.687   0.212  1.00  0.91           N  \\nATOM     74  CA  LEU A  11      -0.441  -3.010  -0.400  1.00  0.91           C  \\nATOM     75  C   LEU A  11      -1.874  -3.367  -0.779  1.00  0.91           C  \\nATOM     76  CB  LEU A  11       0.126  -4.068   0.551  1.00  0.90           C  \\nATOM     77  O   LEU A  11      -2.116  -3.933  -1.847  1.00  0.90           O  \\nATOM     78  CG  LEU A  11       1.633  -4.318   0.466  1.00  0.86           C  \\nATOM     79  CD1 LEU A  11       2.349  -3.641   1.629  1.00  0.82           C  \\nATOM     80  CD2 LEU A  11       1.926  -5.814   0.448  1.00  0.83           C  \\nATOM     81  N   NLE A  12      -2.810  -3.043   0.091  1.00  0.91           N  \\nATOM     82  CA  NLE A  12      -4.214  -3.337  -0.183  1.00  0.91           C  \\nATOM     83  C   NLE A  12      -4.696  -2.598  -1.428  1.00  0.91           C  \\nATOM     85  O   NLE A  12      -5.406  -3.167  -2.260  1.00  0.89           O  \\nATOM     86  N   ARG A  13      -4.335  -1.332  -1.531  1.00  0.91           N  \\nATOM     87  CA  ARG A  13      -4.730  -0.538  -2.690  1.00  0.91           C  \\nATOM     88  C   ARG A  13      -4.139  -1.111  -3.974  1.00  0.91           C  \\nATOM     89  CB  ARG A  13      -4.296   0.919  -2.517  1.00  0.90           C  \\nATOM     90  O   ARG A  13      -4.794  -1.112  -5.018  1.00  0.90           O  \\nATOM     91  CG  ARG A  13      -5.205   1.729  -1.607  1.00  0.81           C  \\nATOM     92  CD  ARG A  13      -4.910   3.220  -1.697  1.00  0.79           C  \\nATOM     93  NE  ARG A  13      -3.550   3.527  -1.263  1.00  0.72           N  \\nATOM     94  NH1 ARG A  13      -4.087   3.722   0.976  1.00  0.57           N  \\nATOM     95  NH2 ARG A  13      -1.925   4.029   0.280  1.00  0.55           N  \\nATOM     96  CZ  ARG A  13      -3.191   3.759  -0.003  1.00  0.71           C  \\nATOM     97  N   ALA A  14      -2.817  -1.477  -3.891  1.00  0.90           N  \\nATOM     98  CA  ALA A  14      -2.168  -2.050  -5.067  1.00  0.90           C  \\nATOM     99  C   ALA A  14      -2.889  -3.311  -5.532  1.00  0.90           C  \\nATOM    100  CB  ALA A  14      -0.702  -2.358  -4.768  1.00  0.89           C  \\nATOM    101  O   ALA A  14      -3.099  -3.507  -6.732  1.00  0.89           O  \\nATOM    102  N   PHE A  15      -3.209  -4.231  -4.470  1.00  0.90           N  \\nATOM    103  CA  PHE A  15      -3.896  -5.474  -4.804  1.00  0.90           C  \\nATOM    104  C   PHE A  15      -5.243  -5.189  -5.457  1.00  0.90           C  \\nATOM    105  CB  PHE A  15      -4.090  -6.335  -3.552  1.00  0.88           C  \\nATOM    106  O   PHE A  15      -5.665  -5.913  -6.362  1.00  0.89           O  \\nATOM    107  CG  PHE A  15      -3.011  -7.363  -3.349  1.00  0.77           C  \\nATOM    108  CD1 PHE A  15      -2.972  -8.514  -4.127  1.00  0.68           C  \\nATOM    109  CD2 PHE A  15      -2.034  -7.179  -2.379  1.00  0.69           C  \\nATOM    110  CE1 PHE A  15      -1.974  -9.468  -3.941  1.00  0.68           C  \\nATOM    111  CE2 PHE A  15      -1.033  -8.128  -2.188  1.00  0.67           C  \\nATOM    112  CZ  PHE A  15      -1.005  -9.272  -2.969  1.00  0.67           C  \\nATOM    113  N   LYS A  16      -6.034  -4.138  -5.051  1.00  0.91           N  \\nATOM    114  CA  LYS A  16      -7.334  -3.775  -5.608  1.00  0.91           C  \\nATOM    115  C   LYS A  16      -7.204  -3.330  -7.062  1.00  0.91           C  \\nATOM    116  CB  LYS A  16      -7.983  -2.667  -4.777  1.00  0.87           C  \\nATOM    117  O   LYS A  16      -8.066  -3.635  -7.889  1.00  0.89           O  \\nATOM    118  CG  LYS A  16      -8.592  -3.150  -3.469  1.00  0.74           C  \\nATOM    119  CD  LYS A  16      -9.473  -2.082  -2.834  1.00  0.70           C  \\nATOM    120  CE  LYS A  16     -10.054  -2.551  -1.508  1.00  0.64           C  \\nATOM    121  NZ  LYS A  16     -10.908  -1.502  -0.875  1.00  0.58           N  \\nATOM    122  N   ARG A  17      -6.071  -2.648  -7.354  1.00  0.89           N  \\nATOM    123  CA  ARG A  17      -5.835  -2.201  -8.723  1.00  0.89           C  \\nATOM    124  C   ARG A  17      -5.580  -3.386  -9.649  1.00  0.90           C  \\nATOM    125  CB  ARG A  17      -4.652  -1.231  -8.776  1.00  0.87           C  \\nATOM    126  O   ARG A  17      -5.888  -3.327 -10.841  1.00  0.89           O  \\nATOM    127  CG  ARG A  17      -4.956   0.142  -8.199  1.00  0.77           C  \\nATOM    128  CD  ARG A  17      -3.757   1.074  -8.299  1.00  0.74           C  \\nATOM    129  NE  ARG A  17      -4.010   2.352  -7.640  1.00  0.69           N  \\nATOM    130  NH1 ARG A  17      -1.806   3.042  -7.673  1.00  0.59           N  \\nATOM    131  NH2 ARG A  17      -3.419   4.387  -6.756  1.00  0.55           N  \\nATOM    132  CZ  ARG A  17      -3.078   3.258  -7.358  1.00  0.67           C  \\nATOM    133  N   LEU A  18      -4.984  -4.400  -9.121  1.00  0.89           N  \\nATOM    134  CA  LEU A  18      -4.686  -5.571  -9.938  1.00  0.89           C  \\nATOM    135  C   LEU A  18      -5.960  -6.338 -10.274  1.00  0.87           C  \\nATOM    136  CB  LEU A  18      -3.697  -6.490  -9.216  1.00  0.87           C  \\nATOM    137  O   LEU A  18      -5.977  -7.141 -11.209  1.00  0.84           O  \\nATOM    138  CG  LEU A  18      -2.287  -5.937  -9.003  1.00  0.79           C  \\nATOM    139  CD1 LEU A  18      -1.488  -6.859  -8.089  1.00  0.72           C  \\nATOM    140  CD2 LEU A  18      -1.577  -5.754 -10.340  1.00  0.73           C  \\nATOM    141  N   GLN A  19      -7.053  -6.109  -9.720  1.00  0.85           N  \\nATOM    142  CA  GLN A  19      -8.328  -6.753 -10.016  1.00  0.85           C  \\nATOM    143  C   GLN A  19      -9.083  -6.001 -11.108  1.00  0.83           C  \\nATOM    144  CB  GLN A  19      -9.187  -6.849  -8.754  1.00  0.81           C  \\nATOM    145  O   GLN A  19      -9.759  -6.614 -11.937  1.00  0.78           O  \\nATOM    146  CG  GLN A  19      -8.616  -7.777  -7.690  1.00  0.74           C  \\nATOM    147  CD  GLN A  19      -9.479  -7.842  -6.444  1.00  0.70           C  \\nATOM    148  NE2 GLN A  19      -9.482  -8.996  -5.785  1.00  0.63           N  \\nATOM    149  OE1 GLN A  19     -10.137  -6.864  -6.076  1.00  0.69           O  \");\n\tviewer_16950270124179935.setStyle({\"model\": 1},{\"cartoon\": {\"color\": \"gray\"}});\n\tviewer_16950270124179935.zoomTo();\nviewer_16950270124179935.render();\n});\n</script>",
      "text/html": [
       "<div id=\"3dmolviewer_16950270124179935\"  style=\"position: relative; width: 640px; height: 480px\">\n",
       "        <p id=\"3dmolwarning_16950270124179935\" style=\"background-color:#ffcccc;color:black\">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>\n",
       "        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>\n",
       "        </div>\n",
       "<script>\n",
       "\n",
       "var loadScriptAsync = function(uri){\n",
       "  return new Promise((resolve, reject) => {\n",
       "    //this is to ignore the existence of requirejs amd\n",
       "    var savedexports, savedmodule;\n",
       "    if (typeof exports !== 'undefined') savedexports = exports;\n",
       "    else exports = {}\n",
       "    if (typeof module !== 'undefined') savedmodule = module;\n",
       "    else module = {}\n",
       "\n",
       "    var tag = document.createElement('script');\n",
       "    tag.src = uri;\n",
       "    tag.async = true;\n",
       "    tag.onload = () => {\n",
       "        exports = savedexports;\n",
       "        module = savedmodule;\n",
       "        resolve();\n",
       "    };\n",
       "  var firstScriptTag = document.getElementsByTagName('script')[0];\n",
       "  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n",
       "});\n",
       "};\n",
       "\n",
       "if(typeof $3Dmolpromise === 'undefined') {\n",
       "$3Dmolpromise = null;\n",
       "  $3Dmolpromise = loadScriptAsync('https://cdnjs.cloudflare.com/ajax/libs/3Dmol/2.0.1/3Dmol-min.js');\n",
       "}\n",
       "\n",
       "var viewer_16950270124179935 = null;\n",
       "var warn = document.getElementById(\"3dmolwarning_16950270124179935\");\n",
       "if(warn) {\n",
       "    warn.parentNode.removeChild(warn);\n",
       "}\n",
       "$3Dmolpromise.then(function() {\n",
       "viewer_16950270124179935 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_16950270124179935\"),{backgroundColor:\"white\"});\n",
       "viewer_16950270124179935.zoomTo();\n",
       "\tviewer_16950270124179935.addModel(\"ATOM      1  H1  ACE     1      21.501   0.153  -0.049  1.00  0.00           H  \\nATOM      2  CH3 ACE     1      21.104  -0.394   0.815  1.00  0.00           C  \\nATOM      3  H2  ACE     1      20.853  -1.414   0.526  1.00  0.00           H  \\nATOM      4  H3  ACE     1      20.160   0.074   1.123  1.00  0.00           H  \\nATOM      5  C   ACE     1      22.222  -0.247   1.853  1.00  0.00           C  \\nATOM      6  O   ACE     1      22.760  -1.215   2.324  1.00  0.00           O  \\nATOM      7  N   NLE     2      22.539   0.971   2.160  1.00  0.00           N  \\nATOM      8  H13 NLE     2      22.010   1.683   1.683  1.00  0.00           H  \\nATOM      9  CA  NLE     2      23.564   1.463   3.103  1.00  0.00           C  \\nATOM     10  C4  NLE     2      24.601   2.320   2.381  1.00  0.00           C  \\nATOM     11  C3  NLE     2      25.411   1.553   1.269  1.00  0.00           C  \\nATOM     12  C2  NLE     2      26.402   2.452   0.455  1.00  0.00           C  \\nATOM     13  C1  NLE     2      27.755   2.263   1.089  1.00  0.00           C  \\nATOM     14  H1  NLE     2      27.969   1.186   1.126  1.00  0.00           H  \\nATOM     15  H2  NLE     2      27.714   2.666   2.040  1.00  0.00           H  \\nATOM     16  H3  NLE     2      28.517   2.803   0.438  1.00  0.00           H  \\nATOM     17  H4  NLE     2      26.607   2.074  -0.498  1.00  0.00           H  \\nATOM     18  H5  NLE     2      26.292   3.519   0.444  1.00  0.00           H  \\nATOM     19  H6  NLE     2      25.934   0.721   1.645  1.00  0.00           H  \\nATOM     20  H7  NLE     2      24.748   1.005   0.662  1.00  0.00           H  \\nATOM     21  H8  NLE     2      24.040   3.129   1.912  1.00  0.00           H  \\nATOM     22  H9  NLE     2      25.343   2.598   3.229  1.00  0.00           H  \\nATOM     23  H10 NLE     2      24.080   0.540   3.335  1.00  0.00           H  \\nATOM     24  C   NLE     2      23.015   2.009   4.421  1.00  0.00           C  \\nATOM     25  O   NLE     2      23.377   3.095   4.864  1.00  0.00           O  \\nATOM     26  N   ALA     3      21.947   1.331   4.839  1.00  0.00           N  \\nATOM     27  H   ALA     3      21.726   0.427   4.400  1.00  0.00           H  \\nATOM     28  CA  ALA     3      21.127   1.828   5.909  1.00  0.00           C  \\nATOM     29  HA  ALA     3      20.946   2.873   5.691  1.00  0.00           H  \\nATOM     30  CB  ALA     3      19.763   1.016   5.818  1.00  0.00           C  \\nATOM     31  HB1 ALA     3      19.904  -0.068   5.758  1.00  0.00           H  \\nATOM     32  HB2 ALA     3      19.034   1.106   6.716  1.00  0.00           H  \\nATOM     33  HB3 ALA     3      19.276   1.398   4.886  1.00  0.00           H  \\nATOM     34  C   ALA     3      21.730   1.855   7.340  1.00  0.00           C  \\nATOM     35  O   ALA     3      22.759   1.298   7.567  1.00  0.00           O  \\nATOM     36  N   THR     4      21.138   2.596   8.288  1.00  0.00           N  \\nATOM     37  H   THR     4      20.269   3.012   8.003  1.00  0.00           H  \\nATOM     38  CA  THR     4      21.573   2.754   9.672  1.00  0.00           C  \\nATOM     39  HA  THR     4      22.671   2.717   9.788  1.00  0.00           H  \\nATOM     40  CB  THR     4      21.110   4.144  10.110  1.00  0.00           C  \\nATOM     41  HB  THR     4      21.397   4.280  11.191  1.00  0.00           H  \\nATOM     42  CG2 THR     4      21.801   5.306   9.382  1.00  0.00           C  \\nATOM     43 HG21 THR     4      21.539   5.305   8.357  1.00  0.00           H  \\nATOM     44 HG22 THR     4      21.372   6.219   9.852  1.00  0.00           H  \\nATOM     45 HG23 THR     4      22.898   5.134   9.436  1.00  0.00           H  \\nATOM     46  OG1 THR     4      19.786   4.361  10.087  1.00  0.00           O  \\nATOM     47  HG1 THR     4      19.724   5.267  10.036  1.00  0.00           H  \\nATOM     48  C   THR     4      21.029   1.610  10.527  1.00  0.00           C  \\nATOM     49  O   THR     4      19.894   1.109  10.244  1.00  0.00           O  \\nATOM     50  N   PRO     5      21.705   1.123  11.580  1.00  0.00           N  \\nATOM     51  CD  PRO     5      23.086   1.438  11.860  1.00  0.00           C  \\nATOM     52  HD2 PRO     5      23.014   2.515  12.268  1.00  0.00           H  \\nATOM     53  HD3 PRO     5      23.700   1.509  10.939  1.00  0.00           H  \\nATOM     54  CG  PRO     5      23.598   0.447  12.977  1.00  0.00           C  \\nATOM     55  HG2 PRO     5      24.127   1.045  13.804  1.00  0.00           H  \\nATOM     56  HG3 PRO     5      24.215  -0.360  12.621  1.00  0.00           H  \\nATOM     57  CB  PRO     5      22.269  -0.056  13.545  1.00  0.00           C  \\nATOM     58  HB2 PRO     5      21.919   0.707  14.340  1.00  0.00           H  \\nATOM     59  HB3 PRO     5      22.355  -1.047  13.811  1.00  0.00           H  \\nATOM     60  CA  PRO     5      21.266  -0.007  12.394  1.00  0.00           C  \\nATOM     61  HA  PRO     5      21.355  -0.968  11.912  1.00  0.00           H  \\nATOM     62  C   PRO     5      19.811   0.086  12.967  1.00  0.00           C  \\nATOM     63  O   PRO     5      19.137  -0.914  12.973  1.00  0.00           O  \\nATOM     64  N   PR8     6      19.409   1.284  13.298  1.00  0.00           N  \\nATOM     65  H19 PR8     6      20.082   2.012  13.260  1.00  0.00           H  \\nATOM     66  CA  PR8     6      18.032   1.740  13.505  1.00  0.00           C  \\nATOM     67  C8  PR8     6      17.617   1.287  14.907  1.00  0.00           C  \\nATOM     68  C7  PR8     6      16.356   1.819  15.622  1.00  0.00           C  \\nATOM     69  C6  PR8     6      16.294   1.282  17.077  1.00  0.00           C  \\nATOM     70  C5  PR8     6      14.979   1.439  17.834  1.00  0.00           C  \\nATOM     71  C4  PR8     6      14.161   2.695  17.680  1.00  0.00           C  \\nATOM     72  C3  PR8     6      12.851   2.796  18.438  1.00  0.00           C  \\nATOM     73  C2  PR8     6      11.885   1.845  17.835  1.00  0.00           C  \\nATOM     74  H3  PR8     6      11.952   0.748  17.987  1.00  0.00           H  \\nATOM     75  H4  PR8     6      12.959   2.540  19.471  1.00  0.00           H  \\nATOM     76  H5  PR8     6      12.513   3.854  18.290  1.00  0.00           H  \\nATOM     77  H6  PR8     6      14.762   3.543  18.005  1.00  0.00           H  \\nATOM     78  H7  PR8     6      14.028   2.813  16.537  1.00  0.00           H  \\nATOM     79  H8  PR8     6      14.343   0.530  17.665  1.00  0.00           H  \\nATOM     80  H9  PR8     6      15.174   1.316  18.854  1.00  0.00           H  \\nATOM     81  H10 PR8     6      17.068   1.784  17.598  1.00  0.00           H  \\nATOM     82  H11 PR8     6      16.592   0.243  17.072  1.00  0.00           H  \\nATOM     83  H12 PR8     6      16.339   2.878  15.551  1.00  0.00           H  \\nATOM     84  H13 PR8     6      15.496   1.424  15.111  1.00  0.00           H  \\nATOM     85  H14 PR8     6      17.705   0.216  14.939  1.00  0.00           H  \\nATOM     86  H15 PR8     6      18.336   1.568  15.619  1.00  0.00           H  \\nATOM     87  C10 PR8     6      17.970   3.280  13.431  1.00  0.00           C  \\nATOM     88  H16 PR8     6      18.548   3.703  14.218  1.00  0.00           H  \\nATOM     89  H17 PR8     6      16.994   3.697  13.583  1.00  0.00           H  \\nATOM     90  H18 PR8     6      18.341   3.607  12.465  1.00  0.00           H  \\nATOM     91  C   PR8     6      17.035   1.168  12.464  1.00  0.00           C  \\nATOM     92  O   PR8     6      15.937   0.656  12.783  1.00  0.00           O  \\nATOM     93  N   ARG     7      17.346   1.196  11.162  1.00  0.00           N  \\nATOM     94  H   ARG     7      18.242   1.494  10.929  1.00  0.00           H  \\nATOM     95  CA  ARG     7      16.398   0.839  10.056  1.00  0.00           C  \\nATOM     96  HA  ARG     7      15.344   0.948  10.307  1.00  0.00           H  \\nATOM     97  CB  ARG     7      16.738   1.717   8.888  1.00  0.00           C  \\nATOM     98  HB2 ARG     7      17.847   1.751   8.748  1.00  0.00           H  \\nATOM     99  HB3 ARG     7      16.299   1.343   7.859  1.00  0.00           H  \\nATOM    100  CG  ARG     7      16.447   3.208   9.031  1.00  0.00           C  \\nATOM    101  HG2 ARG     7      17.145   3.590   9.792  1.00  0.00           H  \\nATOM    102  HG3 ARG     7      16.746   3.651   8.019  1.00  0.00           H  \\nATOM    103  CD  ARG     7      14.987   3.550   9.323  1.00  0.00           C  \\nATOM    104  HD2 ARG     7      14.754   3.218  10.312  1.00  0.00           H  \\nATOM    105  HD3 ARG     7      14.838   4.648   9.307  1.00  0.00           H  \\nATOM    106  NE  ARG     7      14.119   3.068   8.228  1.00  0.00           N  \\nATOM    107  HE  ARG     7      14.522   3.144   7.300  1.00  0.00           H  \\nATOM    108  CZ  ARG     7      12.965   2.516   8.270  1.00  0.00           C  \\nATOM    109  NH1 ARG     7      12.436   2.157   9.392  1.00  0.00           N  \\nATOM    110 HH11 ARG     7      12.910   2.516  10.257  1.00  0.00           H  \\nATOM    111 HH12 ARG     7      11.475   1.792   9.317  1.00  0.00           H  \\nATOM    112  NH2 ARG     7      12.346   2.166   7.177  1.00  0.00           N  \\nATOM    113 HH21 ARG     7      12.680   2.479   6.233  1.00  0.00           H  \\nATOM    114 HH22 ARG     7      11.449   1.666   7.270  1.00  0.00           H  \\nATOM    115  C   ARG     7      16.519  -0.593   9.635  1.00  0.00           C  \\nATOM    116  O   ARG     7      15.531  -1.146   9.096  1.00  0.00           O  \\nATOM    117  N   ARG     8      17.631  -1.317   9.775  1.00  0.00           N  \\nATOM    118  H   ARG     8      18.392  -0.772  10.116  1.00  0.00           H  \\nATOM    119  CA  ARG     8      17.670  -2.815   9.744  1.00  0.00           C  \\nATOM    120  HA  ARG     8      17.129  -3.134   8.859  1.00  0.00           H  \\nATOM    121  CB  ARG     8      19.139  -3.246   9.584  1.00  0.00           C  \\nATOM    122  HB2 ARG     8      19.683  -3.029  10.517  1.00  0.00           H  \\nATOM    123  HB3 ARG     8      19.178  -4.265   9.412  1.00  0.00           H  \\nATOM    124  CG  ARG     8      19.940  -2.610   8.399  1.00  0.00           C  \\nATOM    125  HG2 ARG     8      19.380  -2.728   7.524  1.00  0.00           H  \\nATOM    126  HG3 ARG     8      20.216  -1.540   8.559  1.00  0.00           H  \\nATOM    127  CD  ARG     8      21.268  -3.351   8.255  1.00  0.00           C  \\nATOM    128  HD2 ARG     8      21.758  -3.456   9.130  1.00  0.00           H  \\nATOM    129  HD3 ARG     8      21.007  -4.437   7.948  1.00  0.00           H  \\nATOM    130  NE  ARG     8      22.105  -2.670   7.242  1.00  0.00           N  \\nATOM    131  HE  ARG     8      22.658  -1.893   7.618  1.00  0.00           H  \\nATOM    132  CZ  ARG     8      21.931  -2.602   5.881  1.00  0.00           C  \\nATOM    133  NH1 ARG     8      21.127  -3.296   5.113  1.00  0.00           N  \\nATOM    134 HH11 ARG     8      20.746  -4.163   5.485  1.00  0.00           H  \\nATOM    135 HH12 ARG     8      21.114  -3.200   4.072  1.00  0.00           H  \\nATOM    136  NH2 ARG     8      22.666  -1.667   5.249  1.00  0.00           N  \\nATOM    137 HH21 ARG     8      23.311  -1.076   5.775  1.00  0.00           H  \\nATOM    138 HH22 ARG     8      22.906  -1.834   4.263  1.00  0.00           H  \\nATOM    139  C   ARG     8      16.905  -3.521  10.897  1.00  0.00           C  \\nATOM    140  O   ARG     8      16.109  -4.452  10.597  1.00  0.00           O  \\nATOM    141  N   ARG     9      17.076  -3.031  12.180  1.00  0.00           N  \\nATOM    142  H   ARG     9      17.641  -2.130  12.125  1.00  0.00           H  \\nATOM    143  CA  ARG     9      16.452  -3.634  13.373  1.00  0.00           C  \\nATOM    144  HA  ARG     9      16.624  -4.748  13.288  1.00  0.00           H  \\nATOM    145  CB  ARG     9      17.284  -3.302  14.656  1.00  0.00           C  \\nATOM    146  HB2 ARG     9      17.393  -2.232  14.720  1.00  0.00           H  \\nATOM    147  HB3 ARG     9      16.832  -3.802  15.550  1.00  0.00           H  \\nATOM    148  CG  ARG     9      18.704  -3.839  14.626  1.00  0.00           C  \\nATOM    149  HG2 ARG     9      18.779  -4.957  14.583  1.00  0.00           H  \\nATOM    150  HG3 ARG     9      19.212  -3.434  13.804  1.00  0.00           H  \\nATOM    151  CD  ARG     9      19.377  -3.422  15.999  1.00  0.00           C  \\nATOM    152  HD2 ARG     9      20.434  -3.777  15.988  1.00  0.00           H  \\nATOM    153  HD3 ARG     9      18.981  -3.887  16.876  1.00  0.00           H  \\nATOM    154  NE  ARG     9      19.587  -1.961  16.078  1.00  0.00           N  \\nATOM    155  HE  ARG     9      19.564  -1.482  15.174  1.00  0.00           H  \\nATOM    156  CZ  ARG     9      20.192  -1.322  17.115  1.00  0.00           C  \\nATOM    157  NH1 ARG     9      20.382  -1.892  18.334  1.00  0.00           N  \\nATOM    158 HH11 ARG     9      20.000  -2.760  18.507  1.00  0.00           H  \\nATOM    159 HH12 ARG     9      20.855  -1.332  19.090  1.00  0.00           H  \\nATOM    160  NH2 ARG     9      20.614  -0.090  16.993  1.00  0.00           N  \\nATOM    161 HH21 ARG     9      20.460   0.389  16.141  1.00  0.00           H  \\nATOM    162 HH22 ARG     9      20.990   0.423  17.758  1.00  0.00           H  \\nATOM    163  C   ARG     9      14.960  -3.362  13.497  1.00  0.00           C  \\nATOM    164  O   ARG     9      14.254  -4.150  14.007  1.00  0.00           O  \\nATOM    165  N   AIB    10      14.419  -2.295  12.879  1.00  0.00           N  \\nATOM    166  H7  AIB    10      15.092  -1.680  12.556  1.00  0.00           H  \\nATOM    167  CA  AIB    10      13.054  -1.732  13.118  1.00  0.00           C  \\nATOM    168  C1  AIB    10      13.128  -0.675  14.244  1.00  0.00           C  \\nATOM    169  H1  AIB    10      13.815   0.029  13.897  1.00  0.00           H  \\nATOM    170  H2  AIB    10      13.552  -1.230  15.046  1.00  0.00           H  \\nATOM    171  H3  AIB    10      12.218  -0.296  14.663  1.00  0.00           H  \\nATOM    172  C3  AIB    10      12.564  -1.049  11.886  1.00  0.00           C  \\nATOM    173  H4  AIB    10      12.432  -1.786  11.069  1.00  0.00           H  \\nATOM    174  H5  AIB    10      13.317  -0.366  11.560  1.00  0.00           H  \\nATOM    175  H6  AIB    10      11.684  -0.468  12.181  1.00  0.00           H  \\nATOM    176  C   AIB    10      11.922  -2.756  13.392  1.00  0.00           C  \\nATOM    177  O   AIB    10      11.393  -2.787  14.546  1.00  0.00           O  \\nATOM    178  N   NLE    11      11.570  -3.681  12.485  1.00  0.00           N  \\nATOM    179  H13 NLE    11      12.172  -3.646  11.689  1.00  0.00           H  \\nATOM    180  CA  NLE    11      10.484  -4.721  12.513  1.00  0.00           C  \\nATOM    181  C4  NLE    11      10.856  -5.714  11.384  1.00  0.00           C  \\nATOM    182  C3  NLE    11      10.695  -5.112   9.981  1.00  0.00           C  \\nATOM    183  C2  NLE    11      10.925  -6.187   8.958  1.00  0.00           C  \\nATOM    184  C1  NLE    11      10.702  -5.548   7.560  1.00  0.00           C  \\nATOM    185  H1  NLE    11      11.524  -4.979   7.201  1.00  0.00           H  \\nATOM    186  H2  NLE    11       9.808  -4.891   7.529  1.00  0.00           H  \\nATOM    187  H3  NLE    11      10.523  -6.336   6.806  1.00  0.00           H  \\nATOM    188  H4  NLE    11      11.942  -6.628   9.103  1.00  0.00           H  \\nATOM    189  H5  NLE    11      10.159  -6.950   9.108  1.00  0.00           H  \\nATOM    190  H6  NLE    11       9.666  -4.795   9.964  1.00  0.00           H  \\nATOM    191  H7  NLE    11      11.432  -4.240   9.870  1.00  0.00           H  \\nATOM    192  H8  NLE    11      11.870  -5.983  11.575  1.00  0.00           H  \\nATOM    193  H9  NLE    11      10.343  -6.703  11.405  1.00  0.00           H  \\nATOM    194  H10 NLE    11       9.535  -4.278  12.336  1.00  0.00           H  \\nATOM    195  C   NLE    11      10.414  -5.675  13.792  1.00  0.00           C  \\nATOM    196  O   NLE    11       9.316  -6.110  14.201  1.00  0.00           O  \\nATOM    197  N   LEU    12      11.506  -5.781  14.497  1.00  0.00           N  \\nATOM    198  H   LEU    12      12.248  -5.162  14.184  1.00  0.00           H  \\nATOM    199  CA  LEU    12      11.711  -6.564  15.735  1.00  0.00           C  \\nATOM    200  HA  LEU    12      10.965  -7.376  15.654  1.00  0.00           H  \\nATOM    201  CB  LEU    12      13.085  -7.270  15.742  1.00  0.00           C  \\nATOM    202  HB2 LEU    12      13.845  -6.605  16.107  1.00  0.00           H  \\nATOM    203  HB3 LEU    12      13.018  -8.087  16.406  1.00  0.00           H  \\nATOM    204  CG  LEU    12      13.464  -8.033  14.431  1.00  0.00           C  \\nATOM    205  HG  LEU    12      13.483  -7.349  13.604  1.00  0.00           H  \\nATOM    206  CD1 LEU    12      14.841  -8.682  14.586  1.00  0.00           C  \\nATOM    207 HD11 LEU    12      15.548  -7.951  14.871  1.00  0.00           H  \\nATOM    208 HD12 LEU    12      14.774  -9.507  15.356  1.00  0.00           H  \\nATOM    209 HD13 LEU    12      15.053  -9.027  13.599  1.00  0.00           H  \\nATOM    210  CD2 LEU    12      12.479  -9.174  14.141  1.00  0.00           C  \\nATOM    211 HD21 LEU    12      11.507  -8.869  13.799  1.00  0.00           H  \\nATOM    212 HD22 LEU    12      12.853  -9.905  13.427  1.00  0.00           H  \\nATOM    213 HD23 LEU    12      12.450  -9.824  14.950  1.00  0.00           H  \\nATOM    214  C   LEU    12      11.399  -5.768  17.012  1.00  0.00           C  \\nATOM    215  O   LEU    12      11.264  -6.375  18.076  1.00  0.00           O  \\nATOM    216  N   NLE    13      11.332  -4.428  16.989  1.00  0.00           N  \\nATOM    217  H13 NLE    13      11.466  -3.977  16.155  1.00  0.00           H  \\nATOM    218  CA  NLE    13      11.294  -3.558  18.170  1.00  0.00           C  \\nATOM    219  C4  NLE    13      12.485  -2.551  18.219  1.00  0.00           C  \\nATOM    220  C3  NLE    13      13.886  -3.151  18.452  1.00  0.00           C  \\nATOM    221  C2  NLE    13      14.894  -2.060  18.180  1.00  0.00           C  \\nATOM    222  C1  NLE    13      16.321  -2.488  18.579  1.00  0.00           C  \\nATOM    223  H1  NLE    13      16.826  -3.080  17.859  1.00  0.00           H  \\nATOM    224  H2  NLE    13      17.044  -1.602  18.662  1.00  0.00           H  \\nATOM    225  H3  NLE    13      16.260  -2.956  19.515  1.00  0.00           H  \\nATOM    226  H4  NLE    13      14.864  -1.742  17.160  1.00  0.00           H  \\nATOM    227  H5  NLE    13      14.534  -1.279  18.831  1.00  0.00           H  \\nATOM    228  H6  NLE    13      13.748  -3.423  19.463  1.00  0.00           H  \\nATOM    229  H7  NLE    13      13.975  -4.075  17.819  1.00  0.00           H  \\nATOM    230  H8  NLE    13      12.528  -1.949  17.434  1.00  0.00           H  \\nATOM    231  H9  NLE    13      12.245  -1.834  19.029  1.00  0.00           H  \\nATOM    232  H10 NLE    13      11.301  -4.240  19.081  1.00  0.00           H  \\nATOM    233  C   NLE    13       9.902  -2.867  18.227  1.00  0.00           C  \\nATOM    234  O   NLE    13       9.135  -2.899  17.205  1.00  0.00           O  \\nATOM    235  N   ARG    14       9.542  -2.208  19.361  1.00  0.00           N  \\nATOM    236  H   ARG    14      10.266  -2.144  20.074  1.00  0.00           H  \\nATOM    237  CA  ARG    14       8.226  -1.653  19.506  1.00  0.00           C  \\nATOM    238  HA  ARG    14       7.469  -2.428  19.282  1.00  0.00           H  \\nATOM    239  CB  ARG    14       7.919  -1.134  20.956  1.00  0.00           C  \\nATOM    240  HB2 ARG    14       8.214  -1.899  21.755  1.00  0.00           H  \\nATOM    241  HB3 ARG    14       8.411  -0.181  21.059  1.00  0.00           H  \\nATOM    242  CG  ARG    14       6.485  -0.875  21.383  1.00  0.00           C  \\nATOM    243  HG2 ARG    14       5.885  -0.860  20.447  1.00  0.00           H  \\nATOM    244  HG3 ARG    14       6.172  -1.717  22.052  1.00  0.00           H  \\nATOM    245  CD  ARG    14       6.197   0.395  22.182  1.00  0.00           C  \\nATOM    246  HD2 ARG    14       6.398   1.174  21.462  1.00  0.00           H  \\nATOM    247  HD3 ARG    14       5.059   0.427  22.400  1.00  0.00           H  \\nATOM    248  NE  ARG    14       6.973   0.501  23.451  1.00  0.00           N  \\nATOM    249  HE  ARG    14       7.995   0.240  23.437  1.00  0.00           H  \\nATOM    250  CZ  ARG    14       6.645   1.107  24.600  1.00  0.00           C  \\nATOM    251  NH1 ARG    14       5.545   1.827  24.756  1.00  0.00           N  \\nATOM    252 HH11 ARG    14       4.875   1.880  23.963  1.00  0.00           H  \\nATOM    253 HH12 ARG    14       5.450   2.352  25.683  1.00  0.00           H  \\nATOM    254  NH2 ARG    14       7.381   0.982  25.668  1.00  0.00           N  \\nATOM    255 HH21 ARG    14       8.255   0.570  25.545  1.00  0.00           H  \\nATOM    256 HH22 ARG    14       7.237   1.716  26.316  1.00  0.00           H  \\nATOM    257  C   ARG    14       7.870  -0.643  18.389  1.00  0.00           C  \\nATOM    258  O   ARG    14       6.723  -0.563  18.012  1.00  0.00           O  \\nATOM    259  N   PR3    15       8.768   0.253  17.876  1.00  0.00           N  \\nATOM    260  H10 PR3    15       9.569   0.482  18.411  1.00  0.00           H  \\nATOM    261  CA  PR3    15       8.653   1.187  16.764  1.00  0.00           C  \\nATOM    262  C5  PR3    15       8.122   2.484  17.435  1.00  0.00           C  \\nATOM    263  H6  PR3    15       8.808   2.772  18.248  1.00  0.00           H  \\nATOM    264  H7  PR3    15       7.047   2.397  17.638  1.00  0.00           H  \\nATOM    265  H8  PR3    15       8.150   3.276  16.618  1.00  0.00           H  \\nATOM    266  C3  PR3    15      10.067   1.361  16.124  1.00  0.00           C  \\nATOM    267  C2  PR3    15      10.905   2.281  17.015  1.00  0.00           C  \\nATOM    268  H3  PR3    15      10.705   3.326  16.862  1.00  0.00           H  \\nATOM    269  H4  PR3    15      10.483   0.402  15.979  1.00  0.00           H  \\nATOM    270  H5  PR3    15      10.001   1.796  15.103  1.00  0.00           H  \\nATOM    271  C   PR3    15       7.699   0.645  15.636  1.00  0.00           C  \\nATOM    272  O   PR3    15       6.760   1.348  15.198  1.00  0.00           O  \\nATOM    273  N   PHE    16       7.888  -0.652  15.247  1.00  0.00           N  \\nATOM    274  H   PHE    16       8.671  -1.196  15.676  1.00  0.00           H  \\nATOM    275  CA  PHE    16       7.228  -1.226  14.045  1.00  0.00           C  \\nATOM    276  HA  PHE    16       7.000  -0.501  13.222  1.00  0.00           H  \\nATOM    277  CB  PHE    16       8.057  -2.400  13.485  1.00  0.00           C  \\nATOM    278  HB2 PHE    16       8.936  -2.062  13.054  1.00  0.00           H  \\nATOM    279  HB3 PHE    16       8.320  -3.175  14.328  1.00  0.00           H  \\nATOM    280  CG  PHE    16       7.258  -3.219  12.494  1.00  0.00           C  \\nATOM    281  CD1 PHE    16       6.707  -4.454  12.865  1.00  0.00           C  \\nATOM    282  HD1 PHE    16       6.878  -4.802  13.860  1.00  0.00           H  \\nATOM    283  CE1 PHE    16       5.923  -5.190  12.010  1.00  0.00           C  \\nATOM    284  HE1 PHE    16       5.520  -6.240  12.196  1.00  0.00           H  \\nATOM    285  CZ  PHE    16       5.639  -4.592  10.735  1.00  0.00           C  \\nATOM    286  HZ  PHE    16       4.916  -5.148  10.073  1.00  0.00           H  \\nATOM    287  CE2 PHE    16       6.161  -3.353  10.313  1.00  0.00           C  \\nATOM    288  HE2 PHE    16       5.964  -2.969   9.365  1.00  0.00           H  \\nATOM    289  CD2 PHE    16       6.949  -2.663  11.230  1.00  0.00           C  \\nATOM    290  HD2 PHE    16       7.277  -1.634  10.991  1.00  0.00           H  \\nATOM    291  C   PHE    16       5.796  -1.629  14.479  1.00  0.00           C  \\nATOM    292  O   PHE    16       4.806  -1.383  13.742  1.00  0.00           O  \\nATOM    293  N   LYS    17       5.618  -2.184  15.668  1.00  0.00           N  \\nATOM    294  H   LYS    17       6.465  -2.522  16.041  1.00  0.00           H  \\nATOM    295  CA  LYS    17       4.362  -2.622  16.249  1.00  0.00           C  \\nATOM    296  HA  LYS    17       4.052  -3.425  15.610  1.00  0.00           H  \\nATOM    297  CB  LYS    17       4.794  -3.373  17.573  1.00  0.00           C  \\nATOM    298  HB2 LYS    17       5.508  -4.109  17.263  1.00  0.00           H  \\nATOM    299  HB3 LYS    17       5.324  -2.692  18.284  1.00  0.00           H  \\nATOM    300  CG  LYS    17       3.661  -4.139  18.302  1.00  0.00           C  \\nATOM    301  HG2 LYS    17       4.196  -4.405  19.247  1.00  0.00           H  \\nATOM    302  HG3 LYS    17       2.914  -3.361  18.541  1.00  0.00           H  \\nATOM    303  CD  LYS    17       3.055  -5.270  17.445  1.00  0.00           C  \\nATOM    304  HD2 LYS    17       2.180  -4.908  16.870  1.00  0.00           H  \\nATOM    305  HD3 LYS    17       3.786  -5.675  16.804  1.00  0.00           H  \\nATOM    306  CE  LYS    17       2.518  -6.355  18.359  1.00  0.00           C  \\nATOM    307  HE2 LYS    17       3.240  -6.690  18.912  1.00  0.00           H  \\nATOM    308  HE3 LYS    17       1.694  -5.866  18.953  1.00  0.00           H  \\nATOM    309  NZ  LYS    17       1.948  -7.537  17.661  1.00  0.00           N  \\nATOM    310  HZ1 LYS    17       1.419  -8.092  18.377  1.00  0.00           H  \\nATOM    311  HZ2 LYS    17       1.240  -7.312  17.058  1.00  0.00           H  \\nATOM    312  HZ3 LYS    17       2.616  -8.208  17.253  1.00  0.00           H  \\nATOM    313  C   LYS    17       3.299  -1.564  16.466  1.00  0.00           C  \\nATOM    314  O   LYS    17       2.071  -1.880  16.394  1.00  0.00           O  \\nATOM    315  N   ARG    18       3.689  -0.283  16.646  1.00  0.00           N  \\nATOM    316  H   ARG    18       4.659  -0.174  16.721  1.00  0.00           H  \\nATOM    317  CA  ARG    18       2.809   0.890  16.767  1.00  0.00           C  \\nATOM    318  HA  ARG    18       1.941   0.726  17.421  1.00  0.00           H  \\nATOM    319  CB  ARG    18       3.640   2.087  17.231  1.00  0.00           C  \\nATOM    320  HB2 ARG    18       4.013   1.874  18.271  1.00  0.00           H  \\nATOM    321  HB3 ARG    18       4.584   2.162  16.671  1.00  0.00           H  \\nATOM    322  CG  ARG    18       2.946   3.478  17.026  1.00  0.00           C  \\nATOM    323  HG2 ARG    18       2.978   3.732  15.946  1.00  0.00           H  \\nATOM    324  HG3 ARG    18       1.906   3.368  17.432  1.00  0.00           H  \\nATOM    325  CD  ARG    18       3.709   4.579  17.758  1.00  0.00           C  \\nATOM    326  HD2 ARG    18       3.524   4.399  18.803  1.00  0.00           H  \\nATOM    327  HD3 ARG    18       4.874   4.426  17.704  1.00  0.00           H  \\nATOM    328  NE  ARG    18       3.319   5.924  17.407  1.00  0.00           N  \\nATOM    329  HE  ARG    18       3.910   6.350  16.719  1.00  0.00           H  \\nATOM    330  CZ  ARG    18       2.397   6.659  17.982  1.00  0.00           C  \\nATOM    331  NH1 ARG    18       1.692   6.179  19.007  1.00  0.00           N  \\nATOM    332 HH11 ARG    18       1.628   5.212  19.186  1.00  0.00           H  \\nATOM    333 HH12 ARG    18       1.001   6.815  19.275  1.00  0.00           H  \\nATOM    334  NH2 ARG    18       2.136   7.902  17.692  1.00  0.00           N  \\nATOM    335 HH21 ARG    18       2.658   8.303  16.929  1.00  0.00           H  \\nATOM    336 HH22 ARG    18       1.474   8.395  18.263  1.00  0.00           H  \\nATOM    337  C   ARG    18       2.119   1.195  15.415  1.00  0.00           C  \\nATOM    338  O   ARG    18       1.140   1.917  15.356  1.00  0.00           O  \\nATOM    339  N   LEU    19       2.555   0.651  14.282  1.00  0.00           N  \\nATOM    340  H   LEU    19       3.268  -0.049  14.375  1.00  0.00           H  \\nATOM    341  CA  LEU    19       2.021   0.911  12.975  1.00  0.00           C  \\nATOM    342  HA  LEU    19       1.495   1.826  13.129  1.00  0.00           H  \\nATOM    343  CB  LEU    19       3.209   1.161  12.010  1.00  0.00           C  \\nATOM    344  HB2 LEU    19       3.775   0.246  11.780  1.00  0.00           H  \\nATOM    345  HB3 LEU    19       2.705   1.506  11.100  1.00  0.00           H  \\nATOM    346  CG  LEU    19       4.417   2.178  12.434  1.00  0.00           C  \\nATOM    347  HG  LEU    19       4.764   1.947  13.423  1.00  0.00           H  \\nATOM    348  CD1 LEU    19       5.608   2.054  11.424  1.00  0.00           C  \\nATOM    349 HD11 LEU    19       6.344   2.741  11.843  1.00  0.00           H  \\nATOM    350 HD12 LEU    19       5.947   1.009  11.508  1.00  0.00           H  \\nATOM    351 HD13 LEU    19       5.288   2.245  10.433  1.00  0.00           H  \\nATOM    352  CD2 LEU    19       3.882   3.665  12.405  1.00  0.00           C  \\nATOM    353 HD21 LEU    19       3.816   4.040  11.398  1.00  0.00           H  \\nATOM    354 HD22 LEU    19       2.798   3.619  12.775  1.00  0.00           H  \\nATOM    355 HD23 LEU    19       4.513   4.354  13.079  1.00  0.00           H  \\nATOM    356  C   LEU    19       1.034  -0.047  12.338  1.00  0.00           C  \\nATOM    357  O   LEU    19       0.380   0.372  11.391  1.00  0.00           O  \\nATOM    358  N   GLN    20       0.827  -1.180  12.986  1.00  0.00           N  \\nATOM    359  H   GLN    20       1.382  -1.353  13.752  1.00  0.00           H  \\nATOM    360  CA  GLN    20      -0.242  -2.121  12.675  1.00  0.00           C  \\nATOM    361  HA  GLN    20      -0.494  -2.137  11.636  1.00  0.00           H  \\nATOM    362  CB  GLN    20       0.354  -3.555  13.057  1.00  0.00           C  \\nATOM    363  HB2 GLN    20       0.455  -3.690  14.106  1.00  0.00           H  \\nATOM    364  HB3 GLN    20      -0.438  -4.311  12.796  1.00  0.00           H  \\nATOM    365  CG  GLN    20       1.658  -4.028  12.324  1.00  0.00           C  \\nATOM    366  HG2 GLN    20       1.533  -3.719  11.271  1.00  0.00           H  \\nATOM    367  HG3 GLN    20       2.590  -3.471  12.675  1.00  0.00           H  \\nATOM    368  CD  GLN    20       1.975  -5.498  12.598  1.00  0.00           C  \\nATOM    369  OE1 GLN    20       2.032  -5.993  13.692  1.00  0.00           O  \\nATOM    370  NE2 GLN    20       2.118  -6.273  11.543  1.00  0.00           N  \\nATOM    371 HE21 GLN    20       2.528  -7.091  11.727  1.00  0.00           H  \\nATOM    372 HE22 GLN    20       1.978  -5.822  10.593  1.00  0.00           H  \\nATOM    373  C   GLN    20      -1.493  -1.846  13.452  1.00  0.00           C  \\nATOM    374  O   GLN    20      -2.524  -2.364  13.064  1.00  0.00           O  \\nATOM    375  OXT GLN    20      -1.470  -1.161  14.497  1.00  0.00           O  \\nTER     376      GLN    20 \\nEND   \\n\");\n",
       "\tviewer_16950270124179935.setStyle({\"cartoon\": {\"color\": \"spectrum\"}});\n",
       "\tviewer_16950270124179935.addModel(\"HEADER                                            18-OCT-22                     \\nTITLE     ESMFOLD V1 PREDICTION FOR INPUT\\nREMARK   1                                                                      \\nREMARK   1 REFERENCE 1                                                          \\nREMARK   1  AUTH   ZEMING LIN, HALIL AKIN, ROSHAN RAO, BRIAN HIE, ZHONGKAI ZHU,\\nREMARK   1  AUTH 2 WENTING LU, NIKITA SMETANIN, ROBERT VERKUIL, ORI KABELI,\\nREMARK   1  AUTH 3 YANIV SHMUELI, ALLAN DOS SANTOS COSTA,\\nREMARK   1  AUTH 4 MARYAM FAZEL-ZARANDI, TOM SERCU, SALVATORE CANDIDO,\\nREMARK   1  AUTH 5 ALEXANDER RIVES\\nREMARK   1  TITL   EVOLUTIONARY-SCALE PREDICTION OF ATOMIC LEVEL PROTEIN\\nREMARK   1  TITL 2 STRUCTURE WITH A LANGUAGE MODEL\\nREMARK   1  REF                                                                 \\nREMARK   1  REFN                                                                \\nREMARK   1  PMID                                                                \\nREMARK   1  DOI    10.1101/2022.07.20.500902                                    \\nREMARK   1                                                                      \\nREMARK   1 LICENSE AND DISCLAIMERS                \\nREMARK   1 ESM METAGENOMIC ATLAS DATA IS AVAILABLE UNDER\\nREMARK   1 A CC-BY-4.0 LICENSE FOR ACADEMIC AND COMMERCIAL USE.\\nREMARK   1 COPYRIGHT (C) META PLATFORMS, INC. ALL RIGHTS RESERVED.\\nREMARK   1 USE OF THE ESM METAGENOMIC ATLAS DATA IS SUBJECT\\nREMARK   1 TO THE META OPEN SOURCE TERMS OF USE AND PRIVACY POLICY.\\nATOM    168  C   ACE A   0       9.976   8.120   7.984  1.00  0.00           C  \\nATOM      1  N   NLE A   1       8.593   7.504   7.797  1.00  0.74           N  \\nATOM      2  CA  NLE A   1       8.649   6.048   7.899  1.00  0.74           C  \\nATOM      3  C   NLE A   1       7.266   5.432   7.712  1.00  0.76           C  \\nATOM      5  O   NLE A   1       6.254   6.043   8.064  1.00  0.67           O  \\nATOM      6  N   ALA A   2       7.248   4.631   6.869  1.00  0.79           N  \\nATOM      7  CA  ALA A   2       5.925   4.058   6.634  1.00  0.80           C  \\nATOM      8  C   ALA A   2       5.507   3.151   7.787  1.00  0.79           C  \\nATOM      9  CB  ALA A   2       5.907   3.284   5.318  1.00  0.74           C  \\nATOM     10  O   ALA A   2       6.253   2.249   8.177  1.00  0.74           O  \\nATOM     11  N   THR A   3       4.679   3.521   8.593  1.00  0.87           N  \\nATOM     12  CA  THR A   3       4.083   2.690   9.633  1.00  0.89           C  \\nATOM     13  C   THR A   3       3.488   1.419   9.035  1.00  0.88           C  \\nATOM     14  CB  THR A   3       2.992   3.457  10.404  1.00  0.83           C  \\nATOM     15  O   THR A   3       3.190   1.369   7.840  1.00  0.85           O  \\nATOM     16  CG2 THR A   3       3.528   4.778  10.946  1.00  0.63           C  \\nATOM     17  OG1 THR A   3       1.893   3.724   9.525  1.00  0.65           O  \\nATOM     18  N   PRO A   4       3.551   0.253   9.836  1.00  0.87           N  \\nATOM     19  CA  PRO A   4       2.940  -0.979   9.331  1.00  0.88           C  \\nATOM     20  C   PRO A   4       1.540  -0.754   8.762  1.00  0.88           C  \\nATOM     21  CB  PRO A   4       2.888  -1.880  10.567  1.00  0.86           C  \\nATOM     22  O   PRO A   4       1.173  -1.365   7.756  1.00  0.87           O  \\nATOM     23  CG  PRO A   4       3.878  -1.284  11.515  1.00  0.79           C  \\nATOM     24  CD  PRO A   4       3.979   0.188  11.237  1.00  0.78           C  \\nATOM     25  N   ALA A   5       0.703   0.219   9.323  1.00  0.90           N  \\nATOM     26  CA  ALA A   5      -0.652   0.461   8.832  1.00  0.90           C  \\nATOM     27  C   ALA A   5      -0.627   1.087   7.441  1.00  0.90           C  \\nATOM     28  CB  ALA A   5      -1.416   1.358   9.802  1.00  0.88           C  \\nATOM     29  O   ALA A   5      -1.409   0.703   6.567  1.00  0.88           O  \\nATOM     30  N   ARG A   6       0.273   2.127   7.209  1.00  0.91           N  \\nATOM     31  CA  ARG A   6       0.365   2.756   5.896  1.00  0.91           C  \\nATOM     32  C   ARG A   6       0.820   1.754   4.839  1.00  0.90           C  \\nATOM     33  CB  ARG A   6       1.325   3.947   5.936  1.00  0.89           C  \\nATOM     34  O   ARG A   6       0.310   1.753   3.717  1.00  0.89           O  \\nATOM     35  CG  ARG A   6       0.713   5.212   6.515  1.00  0.82           C  \\nATOM     36  CD  ARG A   6       1.687   6.381   6.472  1.00  0.80           C  \\nATOM     37  NE  ARG A   6       1.118   7.580   7.080  1.00  0.75           N  \\nATOM     38  NH1 ARG A   6       3.089   8.785   7.044  1.00  0.64           N  \\nATOM     39  NH2 ARG A   6       1.179   9.727   7.895  1.00  0.62           N  \\nATOM     40  CZ  ARG A   6       1.797   8.695   7.338  1.00  0.72           C  \\nATOM     41  N   ARG A   7       1.896   0.856   5.232  1.00  0.91           N  \\nATOM     42  CA  ARG A   7       2.370  -0.125   4.261  1.00  0.91           C  \\nATOM     43  C   ARG A   7       1.259  -1.096   3.878  1.00  0.90           C  \\nATOM     44  CB  ARG A   7       3.571  -0.895   4.817  1.00  0.89           C  \\nATOM     45  O   ARG A   7       1.124  -1.462   2.708  1.00  0.89           O  \\nATOM     46  CG  ARG A   7       4.846  -0.071   4.900  1.00  0.82           C  \\nATOM     47  CD  ARG A   7       6.041  -0.921   5.308  1.00  0.80           C  \\nATOM     48  NE  ARG A   7       7.219  -0.102   5.575  1.00  0.74           N  \\nATOM     49  NH1 ARG A   7       8.464  -1.821   6.487  1.00  0.62           N  \\nATOM     50  NH2 ARG A   7       9.355   0.285   6.327  1.00  0.61           N  \\nATOM     51  CZ  ARG A   7       8.343  -0.548   6.129  1.00  0.70           C  \\nATOM     52  N   ARG A   8       0.495  -1.511   4.920  1.00  0.91           N  \\nATOM     53  CA  ARG A   8      -0.614  -2.428   4.677  1.00  0.91           C  \\nATOM     54  C   ARG A   8      -1.666  -1.788   3.777  1.00  0.90           C  \\nATOM     55  CB  ARG A   8      -1.250  -2.865   5.998  1.00  0.89           C  \\nATOM     56  O   ARG A   8      -2.203  -2.440   2.879  1.00  0.89           O  \\nATOM     57  CG  ARG A   8      -2.207  -4.039   5.863  1.00  0.80           C  \\nATOM     58  CD  ARG A   8      -2.725  -4.503   7.217  1.00  0.77           C  \\nATOM     59  NE  ARG A   8      -1.660  -5.076   8.034  1.00  0.70           N  \\nATOM     60  NH1 ARG A   8      -1.997  -7.247   7.318  1.00  0.56           N  \\nATOM     61  NH2 ARG A   8      -0.351  -6.781   8.844  1.00  0.54           N  \\nATOM     62  CZ  ARG A   8      -1.339  -6.367   8.064  1.00  0.69           C  \\nATOM     63  N   AIB A   9      -2.012  -0.574   4.062  1.00  0.91           N  \\nATOM     64  CA  AIB A   9      -3.006   0.120   3.246  1.00  0.91           C  \\nATOM     65  C   AIB A   9      -2.528   0.264   1.804  1.00  0.90           C  \\nATOM     67  O   AIB A   9      -3.312   0.109   0.865  1.00  0.89           O  \\nATOM     68  N   NLE A  10      -1.218   0.603   1.593  1.00  0.91           N  \\nATOM     69  CA  NLE A  10      -0.664   0.744   0.249  1.00  0.91           C  \\nATOM     70  C   NLE A  10      -0.717  -0.580  -0.507  1.00  0.91           C  \\nATOM     72  O   NLE A  10      -1.010  -0.607  -1.705  1.00  0.89           O  \\nATOM     73  N   LEU A  11      -0.379  -1.687   0.212  1.00  0.91           N  \\nATOM     74  CA  LEU A  11      -0.441  -3.010  -0.400  1.00  0.91           C  \\nATOM     75  C   LEU A  11      -1.874  -3.367  -0.779  1.00  0.91           C  \\nATOM     76  CB  LEU A  11       0.126  -4.068   0.551  1.00  0.90           C  \\nATOM     77  O   LEU A  11      -2.116  -3.933  -1.847  1.00  0.90           O  \\nATOM     78  CG  LEU A  11       1.633  -4.318   0.466  1.00  0.86           C  \\nATOM     79  CD1 LEU A  11       2.349  -3.641   1.629  1.00  0.82           C  \\nATOM     80  CD2 LEU A  11       1.926  -5.814   0.448  1.00  0.83           C  \\nATOM     81  N   NLE A  12      -2.810  -3.043   0.091  1.00  0.91           N  \\nATOM     82  CA  NLE A  12      -4.214  -3.337  -0.183  1.00  0.91           C  \\nATOM     83  C   NLE A  12      -4.696  -2.598  -1.428  1.00  0.91           C  \\nATOM     85  O   NLE A  12      -5.406  -3.167  -2.260  1.00  0.89           O  \\nATOM     86  N   ARG A  13      -4.335  -1.332  -1.531  1.00  0.91           N  \\nATOM     87  CA  ARG A  13      -4.730  -0.538  -2.690  1.00  0.91           C  \\nATOM     88  C   ARG A  13      -4.139  -1.111  -3.974  1.00  0.91           C  \\nATOM     89  CB  ARG A  13      -4.296   0.919  -2.517  1.00  0.90           C  \\nATOM     90  O   ARG A  13      -4.794  -1.112  -5.018  1.00  0.90           O  \\nATOM     91  CG  ARG A  13      -5.205   1.729  -1.607  1.00  0.81           C  \\nATOM     92  CD  ARG A  13      -4.910   3.220  -1.697  1.00  0.79           C  \\nATOM     93  NE  ARG A  13      -3.550   3.527  -1.263  1.00  0.72           N  \\nATOM     94  NH1 ARG A  13      -4.087   3.722   0.976  1.00  0.57           N  \\nATOM     95  NH2 ARG A  13      -1.925   4.029   0.280  1.00  0.55           N  \\nATOM     96  CZ  ARG A  13      -3.191   3.759  -0.003  1.00  0.71           C  \\nATOM     97  N   ALA A  14      -2.817  -1.477  -3.891  1.00  0.90           N  \\nATOM     98  CA  ALA A  14      -2.168  -2.050  -5.067  1.00  0.90           C  \\nATOM     99  C   ALA A  14      -2.889  -3.311  -5.532  1.00  0.90           C  \\nATOM    100  CB  ALA A  14      -0.702  -2.358  -4.768  1.00  0.89           C  \\nATOM    101  O   ALA A  14      -3.099  -3.507  -6.732  1.00  0.89           O  \\nATOM    102  N   PHE A  15      -3.209  -4.231  -4.470  1.00  0.90           N  \\nATOM    103  CA  PHE A  15      -3.896  -5.474  -4.804  1.00  0.90           C  \\nATOM    104  C   PHE A  15      -5.243  -5.189  -5.457  1.00  0.90           C  \\nATOM    105  CB  PHE A  15      -4.090  -6.335  -3.552  1.00  0.88           C  \\nATOM    106  O   PHE A  15      -5.665  -5.913  -6.362  1.00  0.89           O  \\nATOM    107  CG  PHE A  15      -3.011  -7.363  -3.349  1.00  0.77           C  \\nATOM    108  CD1 PHE A  15      -2.972  -8.514  -4.127  1.00  0.68           C  \\nATOM    109  CD2 PHE A  15      -2.034  -7.179  -2.379  1.00  0.69           C  \\nATOM    110  CE1 PHE A  15      -1.974  -9.468  -3.941  1.00  0.68           C  \\nATOM    111  CE2 PHE A  15      -1.033  -8.128  -2.188  1.00  0.67           C  \\nATOM    112  CZ  PHE A  15      -1.005  -9.272  -2.969  1.00  0.67           C  \\nATOM    113  N   LYS A  16      -6.034  -4.138  -5.051  1.00  0.91           N  \\nATOM    114  CA  LYS A  16      -7.334  -3.775  -5.608  1.00  0.91           C  \\nATOM    115  C   LYS A  16      -7.204  -3.330  -7.062  1.00  0.91           C  \\nATOM    116  CB  LYS A  16      -7.983  -2.667  -4.777  1.00  0.87           C  \\nATOM    117  O   LYS A  16      -8.066  -3.635  -7.889  1.00  0.89           O  \\nATOM    118  CG  LYS A  16      -8.592  -3.150  -3.469  1.00  0.74           C  \\nATOM    119  CD  LYS A  16      -9.473  -2.082  -2.834  1.00  0.70           C  \\nATOM    120  CE  LYS A  16     -10.054  -2.551  -1.508  1.00  0.64           C  \\nATOM    121  NZ  LYS A  16     -10.908  -1.502  -0.875  1.00  0.58           N  \\nATOM    122  N   ARG A  17      -6.071  -2.648  -7.354  1.00  0.89           N  \\nATOM    123  CA  ARG A  17      -5.835  -2.201  -8.723  1.00  0.89           C  \\nATOM    124  C   ARG A  17      -5.580  -3.386  -9.649  1.00  0.90           C  \\nATOM    125  CB  ARG A  17      -4.652  -1.231  -8.776  1.00  0.87           C  \\nATOM    126  O   ARG A  17      -5.888  -3.327 -10.841  1.00  0.89           O  \\nATOM    127  CG  ARG A  17      -4.956   0.142  -8.199  1.00  0.77           C  \\nATOM    128  CD  ARG A  17      -3.757   1.074  -8.299  1.00  0.74           C  \\nATOM    129  NE  ARG A  17      -4.010   2.352  -7.640  1.00  0.69           N  \\nATOM    130  NH1 ARG A  17      -1.806   3.042  -7.673  1.00  0.59           N  \\nATOM    131  NH2 ARG A  17      -3.419   4.387  -6.756  1.00  0.55           N  \\nATOM    132  CZ  ARG A  17      -3.078   3.258  -7.358  1.00  0.67           C  \\nATOM    133  N   LEU A  18      -4.984  -4.400  -9.121  1.00  0.89           N  \\nATOM    134  CA  LEU A  18      -4.686  -5.571  -9.938  1.00  0.89           C  \\nATOM    135  C   LEU A  18      -5.960  -6.338 -10.274  1.00  0.87           C  \\nATOM    136  CB  LEU A  18      -3.697  -6.490  -9.216  1.00  0.87           C  \\nATOM    137  O   LEU A  18      -5.977  -7.141 -11.209  1.00  0.84           O  \\nATOM    138  CG  LEU A  18      -2.287  -5.937  -9.003  1.00  0.79           C  \\nATOM    139  CD1 LEU A  18      -1.488  -6.859  -8.089  1.00  0.72           C  \\nATOM    140  CD2 LEU A  18      -1.577  -5.754 -10.340  1.00  0.73           C  \\nATOM    141  N   GLN A  19      -7.053  -6.109  -9.720  1.00  0.85           N  \\nATOM    142  CA  GLN A  19      -8.328  -6.753 -10.016  1.00  0.85           C  \\nATOM    143  C   GLN A  19      -9.083  -6.001 -11.108  1.00  0.83           C  \\nATOM    144  CB  GLN A  19      -9.187  -6.849  -8.754  1.00  0.81           C  \\nATOM    145  O   GLN A  19      -9.759  -6.614 -11.937  1.00  0.78           O  \\nATOM    146  CG  GLN A  19      -8.616  -7.777  -7.690  1.00  0.74           C  \\nATOM    147  CD  GLN A  19      -9.479  -7.842  -6.444  1.00  0.70           C  \\nATOM    148  NE2 GLN A  19      -9.482  -8.996  -5.785  1.00  0.63           N  \\nATOM    149  OE1 GLN A  19     -10.137  -6.864  -6.076  1.00  0.69           O  \");\n",
       "\tviewer_16950270124179935.setStyle({\"model\": 1},{\"cartoon\": {\"color\": \"gray\"}});\n",
       "\tviewer_16950270124179935.zoomTo();\n",
       "viewer_16950270124179935.render();\n",
       "});\n",
       "</script>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<py3Dmol.view at 0x7f59fd6522e0>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 首先，读取两个PDB文件的内容\n",
    "with open(\"data/template.pdb\", \"r\") as f:\n",
    "    template_data = f.read()\n",
    "\n",
    "with open(\"data/homology_model.pdb\", \"r\") as f:\n",
    "    homology_model_data = f.read()\n",
    "\n",
    "p = py3Dmol.view()\n",
    "p.addModel(homology_model_data, format=\"pdb\")\n",
    "p.setStyle({'cartoon': {'color':'spectrum'}})\n",
    "p.addModel(template_data, format=\"pdb\")\n",
    "p.setStyle({'model': 1}, {'cartoon': {'color':'gray'}}) \n",
    "p.zoomTo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'data/aligned.pdb'"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# align structure to template\n",
    "AlignStructure.align(ref_pdb='data/template.pdb',\n",
    "                     pdb='data/homology_model.pdb',\n",
    "                     output_pdb='data/aligned.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/3dmoljs_load.v0": "<div id=\"3dmolviewer_16950270345760543\"  style=\"position: relative; width: 640px; height: 480px\">\n        <p id=\"3dmolwarning_16950270345760543\" style=\"background-color:#ffcccc;color:black\">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>\n        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>\n        </div>\n<script>\n\nvar loadScriptAsync = function(uri){\n  return new Promise((resolve, reject) => {\n    //this is to ignore the existence of requirejs amd\n    var savedexports, savedmodule;\n    if (typeof exports !== 'undefined') savedexports = exports;\n    else exports = {}\n    if (typeof module !== 'undefined') savedmodule = module;\n    else module = {}\n\n    var tag = document.createElement('script');\n    tag.src = uri;\n    tag.async = true;\n    tag.onload = () => {\n        exports = savedexports;\n        module = savedmodule;\n        resolve();\n    };\n  var firstScriptTag = document.getElementsByTagName('script')[0];\n  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n});\n};\n\nif(typeof $3Dmolpromise === 'undefined') {\n$3Dmolpromise = null;\n  $3Dmolpromise = loadScriptAsync('https://cdnjs.cloudflare.com/ajax/libs/3Dmol/2.0.1/3Dmol-min.js');\n}\n\nvar viewer_16950270345760543 = null;\nvar warn = document.getElementById(\"3dmolwarning_16950270345760543\");\nif(warn) {\n    warn.parentNode.removeChild(warn);\n}\n$3Dmolpromise.then(function() {\nviewer_16950270345760543 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_16950270345760543\"),{backgroundColor:\"white\"});\nviewer_16950270345760543.zoomTo();\n\tviewer_16950270345760543.addModel(\"ATOM      1  H1  ACE     1      13.135   3.604   5.823  1.00  0.00           H  \\nATOM      2  CH3 ACE     1      12.290   2.907   5.776  1.00  0.00           C  \\nATOM      3  H2  ACE     1      12.655   1.882   5.711  1.00  0.00           H  \\nATOM      4  H3  ACE     1      11.736   3.090   4.846  1.00  0.00           H  \\nATOM      5  C   ACE     1      11.492   3.255   7.037  1.00  0.00           C  \\nATOM      6  O   ACE     1      11.274   2.425   7.881  1.00  0.00           O  \\nATOM      7  N   NLE     2      11.096   4.486   7.122  1.00  0.00           N  \\nATOM      8  H13 NLE     2      11.355   5.072   6.346  1.00  0.00           H  \\nATOM      9  CA  NLE     2      10.325   5.150   8.193  1.00  0.00           C  \\nATOM     10  C4  NLE     2      11.122   6.300   8.805  1.00  0.00           C  \\nATOM     11  C3  NLE     2      12.468   5.857   9.492  1.00  0.00           C  \\nATOM     12  C2  NLE     2      13.340   7.043  10.029  1.00  0.00           C  \\nATOM     13  C1  NLE     2      13.027   7.160  11.498  1.00  0.00           C  \\nATOM     14  H1  NLE     2      13.178   6.175  11.961  1.00  0.00           H  \\nATOM     15  H2  NLE     2      12.042   7.463  11.584  1.00  0.00           H  \\nATOM     16  H3  NLE     2      13.741   7.928  11.940  1.00  0.00           H  \\nATOM     17  H4  NLE     2      14.357   6.806  10.091  1.00  0.00           H  \\nATOM     18  H5  NLE     2      13.186   8.042   9.673  1.00  0.00           H  \\nATOM     19  H6  NLE     2      12.322   5.162  10.268  1.00  0.00           H  \\nATOM     20  H7  NLE     2      12.995   5.206   8.854  1.00  0.00           H  \\nATOM     21  H8  NLE     2      11.354   6.971   7.977  1.00  0.00           H  \\nATOM     22  H9  NLE     2      10.414   6.692   9.636  1.00  0.00           H  \\nATOM     23  H10 NLE     2      10.329   4.376   8.950  1.00  0.00           H  \\nATOM     24  C   NLE     2       8.862   5.431   7.851  1.00  0.00           C  \\nATOM     25  O   NLE     2       8.361   6.535   8.041  1.00  0.00           O  \\nATOM     26  N   ALA     3       8.329   4.470   7.098  1.00  0.00           N  \\nATOM     27  H   ALA     3       8.830   3.577   7.000  1.00  0.00           H  \\nATOM     28  CA  ALA     3       7.057   4.653   6.455  1.00  0.00           C  \\nATOM     29  HA  ALA     3       7.092   5.629   5.989  1.00  0.00           H  \\nATOM     30  CB  ALA     3       6.976   3.526   5.336  1.00  0.00           C  \\nATOM     31  HB1 ALA     3       7.207   2.523   5.709  1.00  0.00           H  \\nATOM     32  HB2 ALA     3       5.944   3.354   4.834  1.00  0.00           H  \\nATOM     33  HB3 ALA     3       7.729   3.841   4.571  1.00  0.00           H  \\nATOM     34  C   ALA     3       5.789   4.722   7.349  1.00  0.00           C  \\nATOM     35  O   ALA     3       5.853   4.434   8.503  1.00  0.00           O  \\nATOM     36  N   THR     4       4.650   5.209   6.835  1.00  0.00           N  \\nATOM     37  H   THR     4       4.693   5.407   5.851  1.00  0.00           H  \\nATOM     38  CA  THR     4       3.375   5.365   7.529  1.00  0.00           C  \\nATOM     39  HA  THR     4       3.492   5.603   8.601  1.00  0.00           H  \\nATOM     40  CB  THR     4       2.670   6.549   6.867  1.00  0.00           C  \\nATOM     41  HB  THR     4       1.662   6.669   7.356  1.00  0.00           H  \\nATOM     42  CG2 THR     4       3.362   7.904   7.079  1.00  0.00           C  \\nATOM     43 HG21 THR     4       4.303   7.915   6.595  1.00  0.00           H  \\nATOM     44 HG22 THR     4       2.696   8.635   6.569  1.00  0.00           H  \\nATOM     45 HG23 THR     4       3.557   8.017   8.168  1.00  0.00           H  \\nATOM     46  OG1 THR     4       2.393   6.418   5.561  1.00  0.00           O  \\nATOM     47  HG1 THR     4       2.309   7.279   5.278  1.00  0.00           H  \\nATOM     48  C   THR     4       2.588   4.055   7.481  1.00  0.00           C  \\nATOM     49  O   THR     4       2.698   3.302   6.461  1.00  0.00           O  \\nATOM     50  N   PRO     5       1.770   3.678   8.476  1.00  0.00           N  \\nATOM     51  CD  PRO     5       1.738   4.316   9.771  1.00  0.00           C  \\nATOM     52  HD2 PRO     5       1.184   5.303   9.545  1.00  0.00           H  \\nATOM     53  HD3 PRO     5       2.747   4.615  10.122  1.00  0.00           H  \\nATOM     54  CG  PRO     5       0.892   3.406  10.745  1.00  0.00           C  \\nATOM     55  HG2 PRO     5       0.118   4.054  11.296  1.00  0.00           H  \\nATOM     56  HG3 PRO     5       1.471   2.816  11.435  1.00  0.00           H  \\nATOM     57  CB  PRO     5       0.137   2.535   9.739  1.00  0.00           C  \\nATOM     58  HB2 PRO     5      -0.807   3.117   9.414  1.00  0.00           H  \\nATOM     59  HB3 PRO     5       0.029   1.582  10.114  1.00  0.00           H  \\nATOM     60  CA  PRO     5       1.042   2.413   8.515  1.00  0.00           C  \\nATOM     61  HA  PRO     5       1.656   1.549   8.712  1.00  0.00           H  \\nATOM     62  C   PRO     5       0.177   2.082   7.251  1.00  0.00           C  \\nATOM     63  O   PRO     5       0.167   0.945   6.850  1.00  0.00           O  \\nATOM     64  N   PR8     6      -0.386   3.106   6.668  1.00  0.00           N  \\nATOM     65  H19 PR8     6      -0.309   3.984   7.124  1.00  0.00           H  \\nATOM     66  CA  PR8     6      -0.929   3.174   5.308  1.00  0.00           C  \\nATOM     67  C8  PR8     6      -2.313   2.521   5.347  1.00  0.00           C  \\nATOM     68  C7  PR8     6      -3.335   2.651   4.196  1.00  0.00           C  \\nATOM     69  C6  PR8     6      -4.687   2.004   4.600  1.00  0.00           C  \\nATOM     70  C5  PR8     6      -5.710   1.757   3.496  1.00  0.00           C  \\nATOM     71  C4  PR8     6      -5.896   2.767   2.394  1.00  0.00           C  \\nATOM     72  C3  PR8     6      -6.912   2.467   1.309  1.00  0.00           C  \\nATOM     73  C2  PR8     6      -6.397   1.349   0.480  1.00  0.00           C  \\nATOM     74  H3  PR8     6      -6.385   0.298   0.835  1.00  0.00           H  \\nATOM     75  H4  PR8     6      -7.858   2.167   1.709  1.00  0.00           H  \\nATOM     76  H5  PR8     6      -6.979   3.410   0.707  1.00  0.00           H  \\nATOM     77  H6  PR8     6      -6.202   3.713   2.838  1.00  0.00           H  \\nATOM     78  H7  PR8     6      -4.831   2.936   1.977  1.00  0.00           H  \\nATOM     79  H8  PR8     6      -5.555   0.730   3.070  1.00  0.00           H  \\nATOM     80  H9  PR8     6      -6.643   1.609   3.945  1.00  0.00           H  \\nATOM     81  H10 PR8     6      -5.101   2.646   5.334  1.00  0.00           H  \\nATOM     82  H11 PR8     6      -4.482   1.081   5.124  1.00  0.00           H  \\nATOM     83  H12 PR8     6      -3.411   3.672   3.916  1.00  0.00           H  \\nATOM     84  H13 PR8     6      -2.962   2.089   3.359  1.00  0.00           H  \\nATOM     85  H14 PR8     6      -2.183   1.509   5.688  1.00  0.00           H  \\nATOM     86  H15 PR8     6      -2.894   2.921   6.124  1.00  0.00           H  \\nATOM     87  C10 PR8     6      -1.076   4.647   4.872  1.00  0.00           C  \\nATOM     88  H16 PR8     6      -1.778   5.141   5.501  1.00  0.00           H  \\nATOM     89  H17 PR8     6      -1.478   4.784   3.888  1.00  0.00           H  \\nATOM     90  H18 PR8     6      -0.107   5.133   4.922  1.00  0.00           H  \\nATOM     91  C   PR8     6      -0.047   2.447   4.260  1.00  0.00           C  \\nATOM     92  O   PR8     6      -0.512   1.646   3.417  1.00  0.00           O  \\nATOM     93  N   ARG     7       1.276   2.657   4.246  1.00  0.00           N  \\nATOM     94  H   ARG     7       1.645   3.193   4.969  1.00  0.00           H  \\nATOM     95  CA  ARG     7       2.202   2.155   3.179  1.00  0.00           C  \\nATOM     96  HA  ARG     7       1.729   1.968   2.216  1.00  0.00           H  \\nATOM     97  CB  ARG     7       3.287   3.180   3.025  1.00  0.00           C  \\nATOM     98  HB2 ARG     7       3.645   3.510   4.032  1.00  0.00           H  \\nATOM     99  HB3 ARG     7       4.245   2.787   2.460  1.00  0.00           H  \\nATOM    100  CG  ARG     7       2.890   4.529   2.434  1.00  0.00           C  \\nATOM    101  HG2 ARG     7       2.243   5.017   3.179  1.00  0.00           H  \\nATOM    102  HG3 ARG     7       3.873   5.112   2.378  1.00  0.00           H  \\nATOM    103  CD  ARG     7       2.263   4.459   1.042  1.00  0.00           C  \\nATOM    104  HD2 ARG     7       1.300   4.002   1.129  1.00  0.00           H  \\nATOM    105  HD3 ARG     7       2.101   5.479   0.641  1.00  0.00           H  \\nATOM    106  NE  ARG     7       3.212   3.857   0.083  1.00  0.00           N  \\nATOM    107  HE  ARG     7       4.184   4.107   0.231  1.00  0.00           H  \\nATOM    108  CZ  ARG     7       3.009   3.024  -0.868  1.00  0.00           C  \\nATOM    109  NH1 ARG     7       1.861   2.454  -1.024  1.00  0.00           N  \\nATOM    110 HH11 ARG     7       1.071   2.854  -0.461  1.00  0.00           H  \\nATOM    111 HH12 ARG     7       1.786   1.860  -1.864  1.00  0.00           H  \\nATOM    112  NH2 ARG     7       3.989   2.613  -1.623  1.00  0.00           N  \\nATOM    113 HH21 ARG     7       4.931   3.075  -1.599  1.00  0.00           H  \\nATOM    114 HH22 ARG     7       3.783   1.893  -2.332  1.00  0.00           H  \\nATOM    115  C   ARG     7       2.827   0.840   3.532  1.00  0.00           C  \\nATOM    116  O   ARG     7       3.222   0.095   2.604  1.00  0.00           O  \\nATOM    117  N   ARG     8       3.015   0.419   4.784  1.00  0.00           N  \\nATOM    118  H   ARG     8       2.767   1.114   5.453  1.00  0.00           H  \\nATOM    119  CA  ARG     8       3.253  -1.011   5.164  1.00  0.00           C  \\nATOM    120  HA  ARG     8       4.043  -1.388   4.524  1.00  0.00           H  \\nATOM    121  CB  ARG     8       3.766  -1.034   6.615  1.00  0.00           C  \\nATOM    122  HB2 ARG     8       2.943  -0.758   7.294  1.00  0.00           H  \\nATOM    123  HB3 ARG     8       4.077  -1.992   6.851  1.00  0.00           H  \\nATOM    124  CG  ARG     8       4.993  -0.122   6.949  1.00  0.00           C  \\nATOM    125  HG2 ARG     8       5.743  -0.312   6.246  1.00  0.00           H  \\nATOM    126  HG3 ARG     8       4.752   0.967   6.996  1.00  0.00           H  \\nATOM    127  CD  ARG     8       5.503  -0.482   8.343  1.00  0.00           C  \\nATOM    128  HD2 ARG     8       4.769  -0.525   9.033  1.00  0.00           H  \\nATOM    129  HD3 ARG     8       5.893  -1.571   8.280  1.00  0.00           H  \\nATOM    130  NE  ARG     8       6.566   0.469   8.740  1.00  0.00           N  \\nATOM    131  HE  ARG     8       6.210   1.331   9.167  1.00  0.00           H  \\nATOM    132  CZ  ARG     8       7.841   0.597   8.245  1.00  0.00           C  \\nATOM    133  NH1 ARG     8       8.514  -0.219   7.471  1.00  0.00           N  \\nATOM    134 HH11 ARG     8       8.192  -1.181   7.400  1.00  0.00           H  \\nATOM    135 HH12 ARG     8       9.508  -0.048   7.196  1.00  0.00           H  \\nATOM    136  NH2 ARG     8       8.479   1.736   8.574  1.00  0.00           N  \\nATOM    137 HH21 ARG     8       8.021   2.431   9.166  1.00  0.00           H  \\nATOM    138 HH22 ARG     8       9.506   1.715   8.612  1.00  0.00           H  \\nATOM    139  C   ARG     8       2.073  -1.979   4.874  1.00  0.00           C  \\nATOM    140  O   ARG     8       2.326  -3.057   4.271  1.00  0.00           O  \\nATOM    141  N   ARG     9       0.798  -1.563   5.217  1.00  0.00           N  \\nATOM    142  H   ARG     9       0.847  -0.545   5.527  1.00  0.00           H  \\nATOM    143  CA  ARG     9      -0.405  -2.398   5.045  1.00  0.00           C  \\nATOM    144  HA  ARG     9      -0.139  -3.420   5.448  1.00  0.00           H  \\nATOM    145  CB  ARG     9      -1.524  -1.965   6.049  1.00  0.00           C  \\nATOM    146  HB2 ARG     9      -1.707  -0.911   5.916  1.00  0.00           H  \\nATOM    147  HB3 ARG     9      -2.417  -2.633   5.946  1.00  0.00           H  \\nATOM    148  CG  ARG     9      -1.133  -2.113   7.509  1.00  0.00           C  \\nATOM    149  HG2 ARG     9      -0.927  -3.167   7.832  1.00  0.00           H  \\nATOM    150  HG3 ARG     9      -0.287  -1.527   7.704  1.00  0.00           H  \\nATOM    151  CD  ARG     9      -2.383  -1.645   8.364  1.00  0.00           C  \\nATOM    152  HD2 ARG     9      -2.109  -1.714   9.443  1.00  0.00           H  \\nATOM    153  HD3 ARG     9      -3.252  -2.264   8.302  1.00  0.00           H  \\nATOM    154  NE  ARG     9      -2.612  -0.191   8.238  1.00  0.00           N  \\nATOM    155  HE  ARG     9      -1.804   0.336   7.895  1.00  0.00           H  \\nATOM    156  CZ  ARG     9      -3.580   0.500   8.900  1.00  0.00           C  \\nATOM    157  NH1 ARG     9      -4.647  -0.096   9.494  1.00  0.00           N  \\nATOM    158 HH11 ARG     9      -4.777  -1.044   9.377  1.00  0.00           H  \\nATOM    159 HH12 ARG     9      -5.358   0.506   9.984  1.00  0.00           H  \\nATOM    160  NH2 ARG     9      -3.540   1.804   8.981  1.00  0.00           N  \\nATOM    161 HH21 ARG     9      -2.809   2.293   8.527  1.00  0.00           H  \\nATOM    162 HH22 ARG     9      -4.274   2.335   9.393  1.00  0.00           H  \\nATOM    163  C   ARG     9      -0.867  -2.531   3.601  1.00  0.00           C  \\nATOM    164  O   ARG     9      -1.400  -3.512   3.236  1.00  0.00           O  \\nATOM    165  N   AIB    10      -0.520  -1.595   2.697  1.00  0.00           N  \\nATOM    166  H7  AIB    10      -0.152  -0.804   3.114  1.00  0.00           H  \\nATOM    167  CA  AIB    10      -1.106  -1.424   1.331  1.00  0.00           C  \\nATOM    168  C1  AIB    10      -2.324  -0.475   1.414  1.00  0.00           C  \\nATOM    169  H1  AIB    10      -1.942   0.407   1.818  1.00  0.00           H  \\nATOM    170  H2  AIB    10      -2.941  -0.963   2.130  1.00  0.00           H  \\nATOM    171  H3  AIB    10      -2.967  -0.378   0.563  1.00  0.00           H  \\nATOM    172  C3  AIB    10      -0.103  -0.796   0.424  1.00  0.00           C  \\nATOM    173  H4  AIB    10       0.761  -1.476   0.283  1.00  0.00           H  \\nATOM    174  H5  AIB    10       0.276   0.082   0.900  1.00  0.00           H  \\nATOM    175  H6  AIB    10      -0.646  -0.487  -0.475  1.00  0.00           H  \\nATOM    176  C   AIB    10      -1.466  -2.723   0.566  1.00  0.00           C  \\nATOM    177  O   AIB    10      -2.689  -2.980   0.340  1.00  0.00           O  \\nATOM    178  N   NLE    11      -0.535  -3.634   0.240  1.00  0.00           N  \\nATOM    179  H13 NLE    11       0.355  -3.382   0.617  1.00  0.00           H  \\nATOM    180  CA  NLE    11      -0.645  -4.918  -0.535  1.00  0.00           C  \\nATOM    181  C4  NLE    11       0.659  -5.689  -0.212  1.00  0.00           C  \\nATOM    182  C3  NLE    11       1.906  -5.041  -0.830  1.00  0.00           C  \\nATOM    183  C2  NLE    11       3.089  -5.937  -0.597  1.00  0.00           C  \\nATOM    184  C1  NLE    11       4.314  -5.269  -1.280  1.00  0.00           C  \\nATOM    185  H1  NLE    11       4.754  -4.481  -0.720  1.00  0.00           H  \\nATOM    186  H2  NLE    11       4.073  -4.865  -2.286  1.00  0.00           H  \\nATOM    187  H3  NLE    11       5.114  -6.015  -1.439  1.00  0.00           H  \\nATOM    188  H4  NLE    11       3.215  -6.110   0.500  1.00  0.00           H  \\nATOM    189  H5  NLE    11       2.889  -6.881  -1.107  1.00  0.00           H  \\nATOM    190  H6  NLE    11       1.670  -5.000  -1.879  1.00  0.00           H  \\nATOM    191  H7  NLE    11       2.048  -4.003  -0.363  1.00  0.00           H  \\nATOM    192  H8  NLE    11       0.717  -5.702   0.852  1.00  0.00           H  \\nATOM    193  H9  NLE    11       0.666  -6.776  -0.460  1.00  0.00           H  \\nATOM    194  H10 NLE    11      -0.726  -4.722  -1.576  1.00  0.00           H  \\nATOM    195  C   NLE    11      -1.772  -5.955  -0.082  1.00  0.00           C  \\nATOM    196  O   NLE    11      -2.334  -6.689  -0.923  1.00  0.00           O  \\nATOM    197  N   LEU    12      -2.218  -5.830   1.137  1.00  0.00           N  \\nATOM    198  H   LEU    12      -1.846  -5.018   1.620  1.00  0.00           H  \\nATOM    199  CA  LEU    12      -3.272  -6.628   1.800  1.00  0.00           C  \\nATOM    200  HA  LEU    12      -3.238  -7.597   1.267  1.00  0.00           H  \\nATOM    201  CB  LEU    12      -2.904  -6.954   3.264  1.00  0.00           C  \\nATOM    202  HB2 LEU    12      -3.192  -6.146   3.911  1.00  0.00           H  \\nATOM    203  HB3 LEU    12      -3.453  -7.810   3.546  1.00  0.00           H  \\nATOM    204  CG  LEU    12      -1.453  -7.488   3.498  1.00  0.00           C  \\nATOM    205  HG  LEU    12      -0.739  -6.760   3.165  1.00  0.00           H  \\nATOM    206  CD1 LEU    12      -1.236  -7.770   4.986  1.00  0.00           C  \\nATOM    207 HD11 LEU    12      -1.466  -6.906   5.548  1.00  0.00           H  \\nATOM    208 HD12 LEU    12      -1.886  -8.642   5.294  1.00  0.00           H  \\nATOM    209 HD13 LEU    12      -0.189  -7.970   5.039  1.00  0.00           H  \\nATOM    210  CD2 LEU    12      -1.221  -8.819   2.768  1.00  0.00           C  \\nATOM    211 HD21 LEU    12      -1.128  -8.749   1.700  1.00  0.00           H  \\nATOM    212 HD22 LEU    12      -0.354  -9.370   3.127  1.00  0.00           H  \\nATOM    213 HD23 LEU    12      -1.924  -9.515   3.080  1.00  0.00           H  \\nATOM    214  C   LEU    12      -4.681  -6.043   1.614  1.00  0.00           C  \\nATOM    215  O   LEU    12      -5.659  -6.745   1.874  1.00  0.00           O  \\nATOM    216  N   NLE    13      -4.851  -4.768   1.231  1.00  0.00           N  \\nATOM    217  H13 NLE    13      -4.076  -4.234   1.060  1.00  0.00           H  \\nATOM    218  CA  NLE    13      -6.121  -4.032   1.265  1.00  0.00           C  \\nATOM    219  C4  NLE    13      -6.060  -2.759   2.165  1.00  0.00           C  \\nATOM    220  C3  NLE    13      -5.919  -2.994   3.682  1.00  0.00           C  \\nATOM    221  C2  NLE    13      -5.596  -1.662   4.316  1.00  0.00           C  \\nATOM    222  C1  NLE    13      -5.634  -1.737   5.856  1.00  0.00           C  \\nATOM    223  H1  NLE    13      -4.753  -2.120   6.305  1.00  0.00           H  \\nATOM    224  H2  NLE    13      -5.685  -0.704   6.350  1.00  0.00           H  \\nATOM    225  H3  NLE    13      -6.492  -2.277   6.124  1.00  0.00           H  \\nATOM    226  H4  NLE    13      -4.655  -1.283   3.977  1.00  0.00           H  \\nATOM    227  H5  NLE    13      -6.405  -1.054   3.943  1.00  0.00           H  \\nATOM    228  H6  NLE    13      -6.892  -3.371   3.849  1.00  0.00           H  \\nATOM    229  H7  NLE    13      -5.164  -3.811   3.836  1.00  0.00           H  \\nATOM    230  H8  NLE    13      -5.370  -2.107   1.883  1.00  0.00           H  \\nATOM    231  H9  NLE    13      -6.990  -2.194   1.957  1.00  0.00           H  \\nATOM    232  H10 NLE    13      -6.912  -4.759   1.642  1.00  0.00           H  \\nATOM    233  C   NLE    13      -6.553  -3.731  -0.198  1.00  0.00           C  \\nATOM    234  O   NLE    13      -5.714  -3.879  -1.151  1.00  0.00           O  \\nATOM    235  N   ARG    14      -7.814  -3.278  -0.431  1.00  0.00           N  \\nATOM    236  H   ARG    14      -8.366  -3.085   0.403  1.00  0.00           H  \\nATOM    237  CA  ARG    14      -8.297  -3.094  -1.770  1.00  0.00           C  \\nATOM    238  HA  ARG    14      -8.131  -4.019  -2.355  1.00  0.00           H  \\nATOM    239  CB  ARG    14      -9.836  -2.788  -1.846  1.00  0.00           C  \\nATOM    240  HB2 ARG    14     -10.448  -3.511  -1.204  1.00  0.00           H  \\nATOM    241  HB3 ARG    14      -9.962  -1.751  -1.581  1.00  0.00           H  \\nATOM    242  CG  ARG    14     -10.577  -2.942  -3.162  1.00  0.00           C  \\nATOM    243  HG2 ARG    14      -9.794  -3.009  -3.949  1.00  0.00           H  \\nATOM    244  HG3 ARG    14     -11.177  -3.886  -3.106  1.00  0.00           H  \\nATOM    245  CD  ARG    14     -11.581  -1.856  -3.547  1.00  0.00           C  \\nATOM    246  HD2 ARG    14     -10.945  -0.997  -3.706  1.00  0.00           H  \\nATOM    247  HD3 ARG    14     -12.029  -2.136  -4.579  1.00  0.00           H  \\nATOM    248  NE  ARG    14     -12.667  -1.653  -2.546  1.00  0.00           N  \\nATOM    249  HE  ARG    14     -12.410  -1.639  -1.523  1.00  0.00           H  \\nATOM    250  CZ  ARG    14     -13.929  -1.244  -2.732  1.00  0.00           C  \\nATOM    251  NH1 ARG    14     -14.401  -0.847  -3.903  1.00  0.00           N  \\nATOM    252 HH11 ARG    14     -13.776  -0.907  -4.732  1.00  0.00           H  \\nATOM    253 HH12 ARG    14     -15.390  -0.439  -3.902  1.00  0.00           H  \\nATOM    254  NH2 ARG    14     -14.798  -1.258  -1.761  1.00  0.00           N  \\nATOM    255 HH21 ARG    14     -14.445  -1.420  -0.868  1.00  0.00           H  \\nATOM    256 HH22 ARG    14     -15.554  -0.640  -1.919  1.00  0.00           H  \\nATOM    257  C   ARG    14      -7.422  -2.126  -2.601  1.00  0.00           C  \\nATOM    258  O   ARG    14      -7.301  -2.316  -3.790  1.00  0.00           O  \\nATOM    259  N   PR3    15      -6.861  -0.991  -2.081  1.00  0.00           N  \\nATOM    260  H10 PR3    15      -7.247  -0.606  -1.255  1.00  0.00           H  \\nATOM    261  CA  PR3    15      -5.931  -0.035  -2.666  1.00  0.00           C  \\nATOM    262  C5  PR3    15      -6.863   1.025  -3.316  1.00  0.00           C  \\nATOM    263  H6  PR3    15      -7.550   1.415  -2.548  1.00  0.00           H  \\nATOM    264  H7  PR3    15      -7.268   0.648  -4.264  1.00  0.00           H  \\nATOM    265  H8  PR3    15      -6.171   1.859  -3.664  1.00  0.00           H  \\nATOM    266  C3  PR3    15      -5.045   0.548  -1.519  1.00  0.00           C  \\nATOM    267  C2  PR3    15      -5.861   1.580  -0.737  1.00  0.00           C  \\nATOM    268  H3  PR3    15      -5.893   2.547  -1.206  1.00  0.00           H  \\nATOM    269  H4  PR3    15      -4.691  -0.257  -0.935  1.00  0.00           H  \\nATOM    270  H5  PR3    15      -4.126   1.030  -1.919  1.00  0.00           H  \\nATOM    271  C   PR3    15      -4.959  -0.714  -3.700  1.00  0.00           C  \\nATOM    272  O   PR3    15      -4.820  -0.245  -4.853  1.00  0.00           O  \\nATOM    273  N   PHE    16      -4.370  -1.884  -3.308  1.00  0.00           N  \\nATOM    274  H   PHE    16      -4.553  -2.239  -2.342  1.00  0.00           H  \\nATOM    275  CA  PHE    16      -3.262  -2.513  -4.074  1.00  0.00           C  \\nATOM    276  HA  PHE    16      -2.608  -1.809  -4.650  1.00  0.00           H  \\nATOM    277  CB  PHE    16      -2.393  -3.385  -3.146  1.00  0.00           C  \\nATOM    278  HB2 PHE    16      -1.841  -2.799  -2.495  1.00  0.00           H  \\nATOM    279  HB3 PHE    16      -3.053  -4.130  -2.521  1.00  0.00           H  \\nATOM    280  CG  PHE    16      -1.486  -4.302  -3.938  1.00  0.00           C  \\nATOM    281  CD1 PHE    16      -1.793  -5.663  -4.083  1.00  0.00           C  \\nATOM    282  HD1 PHE    16      -2.676  -6.032  -3.610  1.00  0.00           H  \\nATOM    283  CE1 PHE    16      -1.025  -6.507  -4.849  1.00  0.00           C  \\nATOM    284  HE1 PHE    16      -1.148  -7.637  -4.940  1.00  0.00           H  \\nATOM    285  CZ  PHE    16       0.073  -5.904  -5.552  1.00  0.00           C  \\nATOM    286  HZ  PHE    16       0.642  -6.574  -6.257  1.00  0.00           H  \\nATOM    287  CE2 PHE    16       0.423  -4.544  -5.446  1.00  0.00           C  \\nATOM    288  HE2 PHE    16       1.251  -4.150  -5.941  1.00  0.00           H  \\nATOM    289  CD2 PHE    16      -0.397  -3.748  -4.652  1.00  0.00           C  \\nATOM    290  HD2 PHE    16      -0.236  -2.653  -4.638  1.00  0.00           H  \\nATOM    291  C   PHE    16      -3.922  -3.304  -5.232  1.00  0.00           C  \\nATOM    292  O   PHE    16      -3.442  -3.265  -6.395  1.00  0.00           O  \\nATOM    293  N   LYS    17      -5.037  -3.978  -4.995  1.00  0.00           N  \\nATOM    294  H   LYS    17      -5.181  -4.114  -4.030  1.00  0.00           H  \\nATOM    295  CA  LYS    17      -5.798  -4.769  -5.944  1.00  0.00           C  \\nATOM    296  HA  LYS    17      -5.135  -5.572  -6.197  1.00  0.00           H  \\nATOM    297  CB  LYS    17      -6.894  -5.485  -5.055  1.00  0.00           C  \\nATOM    298  HB2 LYS    17      -6.349  -5.985  -4.280  1.00  0.00           H  \\nATOM    299  HB3 LYS    17      -7.566  -4.748  -4.549  1.00  0.00           H  \\nATOM    300  CG  LYS    17      -7.730  -6.572  -5.777  1.00  0.00           C  \\nATOM    301  HG2 LYS    17      -8.501  -6.764  -4.991  1.00  0.00           H  \\nATOM    302  HG3 LYS    17      -8.218  -6.034  -6.609  1.00  0.00           H  \\nATOM    303  CD  LYS    17      -6.871  -7.750  -6.282  1.00  0.00           C  \\nATOM    304  HD2 LYS    17      -6.540  -7.582  -7.326  1.00  0.00           H  \\nATOM    305  HD3 LYS    17      -6.046  -7.901  -5.645  1.00  0.00           H  \\nATOM    306  CE  LYS    17      -7.722  -9.005  -6.324  1.00  0.00           C  \\nATOM    307  HE2 LYS    17      -8.066  -9.185  -5.436  1.00  0.00           H  \\nATOM    308  HE3 LYS    17      -8.532  -8.794  -7.079  1.00  0.00           H  \\nATOM    309  NZ  LYS    17      -7.003 -10.236  -6.746  1.00  0.00           N  \\nATOM    310  HZ1 LYS    17      -7.732 -10.963  -6.950  1.00  0.00           H  \\nATOM    311  HZ2 LYS    17      -6.593 -10.155  -7.607  1.00  0.00           H  \\nATOM    312  HZ3 LYS    17      -6.382 -10.678  -6.053  1.00  0.00           H  \\nATOM    313  C   LYS    17      -6.367  -4.042  -7.146  1.00  0.00           C  \\nATOM    314  O   LYS    17      -6.506  -4.657  -8.248  1.00  0.00           O  \\nATOM    315  N   ARG    18      -6.633  -2.722  -7.036  1.00  0.00           N  \\nATOM    316  H   ARG    18      -6.523  -2.372  -6.128  1.00  0.00           H  \\nATOM    317  CA  ARG    18      -7.087  -1.829  -8.114  1.00  0.00           C  \\nATOM    318  HA  ARG    18      -7.877  -2.262  -8.744  1.00  0.00           H  \\nATOM    319  CB  ARG    18      -7.528  -0.498  -7.502  1.00  0.00           C  \\nATOM    320  HB2 ARG    18      -8.431  -0.688  -6.860  1.00  0.00           H  \\nATOM    321  HB3 ARG    18      -6.802  -0.138  -6.758  1.00  0.00           H  \\nATOM    322  CG  ARG    18      -7.656   0.678  -8.531  1.00  0.00           C  \\nATOM    323  HG2 ARG    18      -6.637   1.016  -8.810  1.00  0.00           H  \\nATOM    324  HG3 ARG    18      -8.248   0.272  -9.393  1.00  0.00           H  \\nATOM    325  CD  ARG    18      -8.358   1.878  -7.899  1.00  0.00           C  \\nATOM    326  HD2 ARG    18      -9.385   1.575  -7.790  1.00  0.00           H  \\nATOM    327  HD3 ARG    18      -8.047   2.036  -6.776  1.00  0.00           H  \\nATOM    328  NE  ARG    18      -8.277   3.100  -8.664  1.00  0.00           N  \\nATOM    329  HE  ARG    18      -7.546   3.717  -8.364  1.00  0.00           H  \\nATOM    330  CZ  ARG    18      -9.121   3.525  -9.573  1.00  0.00           C  \\nATOM    331  NH1 ARG    18     -10.195   2.800  -9.890  1.00  0.00           N  \\nATOM    332 HH11 ARG    18     -10.252   1.839  -9.682  1.00  0.00           H  \\nATOM    333 HH12 ARG    18     -10.681   3.213 -10.629  1.00  0.00           H  \\nATOM    334  NH2 ARG    18      -9.059   4.677 -10.178  1.00  0.00           N  \\nATOM    335 HH21 ARG    18      -8.266   5.258  -9.955  1.00  0.00           H  \\nATOM    336 HH22 ARG    18      -9.814   4.936 -10.787  1.00  0.00           H  \\nATOM    337  C   ARG    18      -5.958  -1.607  -9.149  1.00  0.00           C  \\nATOM    338  O   ARG    18      -6.197  -1.160 -10.257  1.00  0.00           O  \\nATOM    339  N   LEU    19      -4.697  -1.930  -8.872  1.00  0.00           N  \\nATOM    340  H   LEU    19      -4.548  -2.427  -8.013  1.00  0.00           H  \\nATOM    341  CA  LEU    19      -3.574  -1.715  -9.739  1.00  0.00           C  \\nATOM    342  HA  LEU    19      -3.953  -0.981 -10.415  1.00  0.00           H  \\nATOM    343  CB  LEU    19      -2.429  -1.092  -8.899  1.00  0.00           C  \\nATOM    344  HB2 LEU    19      -1.968  -1.809  -8.203  1.00  0.00           H  \\nATOM    345  HB3 LEU    19      -1.695  -0.818  -9.666  1.00  0.00           H  \\nATOM    346  CG  LEU    19      -2.729   0.166  -7.899  1.00  0.00           C  \\nATOM    347  HG  LEU    19      -3.587  -0.044  -7.288  1.00  0.00           H  \\nATOM    348  CD1 LEU    19      -1.490   0.433  -6.978  1.00  0.00           C  \\nATOM    349 HD11 LEU    19      -1.838   1.252  -6.347  1.00  0.00           H  \\nATOM    350 HD12 LEU    19      -1.362  -0.492  -6.394  1.00  0.00           H  \\nATOM    351 HD13 LEU    19      -0.620   0.612  -7.554  1.00  0.00           H  \\nATOM    352  CD2 LEU    19      -3.009   1.462  -8.759  1.00  0.00           C  \\nATOM    353 HD21 LEU    19      -2.096   1.885  -9.142  1.00  0.00           H  \\nATOM    354 HD22 LEU    19      -3.583   1.109  -9.686  1.00  0.00           H  \\nATOM    355 HD23 LEU    19      -3.626   2.236  -8.169  1.00  0.00           H  \\nATOM    356  C   LEU    19      -3.029  -2.842 -10.594  1.00  0.00           C  \\nATOM    357  O   LEU    19      -2.301  -2.533 -11.529  1.00  0.00           O  \\nATOM    358  N   GLN    20      -3.549  -4.037 -10.374  1.00  0.00           N  \\nATOM    359  H   GLN    20      -4.155  -4.120  -9.633  1.00  0.00           H  \\nATOM    360  CA  GLN    20      -3.339  -5.195 -11.235  1.00  0.00           C  \\nATOM    361  HA  GLN    20      -2.381  -5.194 -11.710  1.00  0.00           H  \\nATOM    362  CB  GLN    20      -3.396  -6.452 -10.248  1.00  0.00           C  \\nATOM    363  HB2 GLN    20      -4.375  -6.638  -9.879  1.00  0.00           H  \\nATOM    364  HB3 GLN    20      -3.204  -7.364 -10.880  1.00  0.00           H  \\nATOM    365  CG  GLN    20      -2.356  -6.513  -9.076  1.00  0.00           C  \\nATOM    366  HG2 GLN    20      -1.402  -6.165  -9.509  1.00  0.00           H  \\nATOM    367  HG3 GLN    20      -2.581  -5.764  -8.245  1.00  0.00           H  \\nATOM    368  CD  GLN    20      -2.361  -7.868  -8.369  1.00  0.00           C  \\nATOM    369  OE1 GLN    20      -3.344  -8.416  -7.946  1.00  0.00           O  \\nATOM    370  NE2 GLN    20      -1.205  -8.495  -8.296  1.00  0.00           N  \\nATOM    371 HE21 GLN    20      -1.190  -9.191  -7.675  1.00  0.00           H  \\nATOM    372 HE22 GLN    20      -0.372  -8.022  -8.753  1.00  0.00           H  \\nATOM    373  C   GLN    20      -4.385  -5.314 -12.302  1.00  0.00           C  \\nATOM    374  O   GLN    20      -4.150  -6.048 -13.244  1.00  0.00           O  \\nATOM    375  OXT GLN    20      -5.485  -4.730 -12.199  1.00  0.00           O  \\nTER     376      GLN    20                                                       \\nEND   \\n\");\n\tviewer_16950270345760543.setStyle({\"cartoon\": {\"color\": \"spectrum\"}});\n\tviewer_16950270345760543.addModel(\"HEADER                                            18-OCT-22                     \\nTITLE     ESMFOLD V1 PREDICTION FOR INPUT\\nREMARK   1                                                                      \\nREMARK   1 REFERENCE 1                                                          \\nREMARK   1  AUTH   ZEMING LIN, HALIL AKIN, ROSHAN RAO, BRIAN HIE, ZHONGKAI ZHU,\\nREMARK   1  AUTH 2 WENTING LU, NIKITA SMETANIN, ROBERT VERKUIL, ORI KABELI,\\nREMARK   1  AUTH 3 YANIV SHMUELI, ALLAN DOS SANTOS COSTA,\\nREMARK   1  AUTH 4 MARYAM FAZEL-ZARANDI, TOM SERCU, SALVATORE CANDIDO,\\nREMARK   1  AUTH 5 ALEXANDER RIVES\\nREMARK   1  TITL   EVOLUTIONARY-SCALE PREDICTION OF ATOMIC LEVEL PROTEIN\\nREMARK   1  TITL 2 STRUCTURE WITH A LANGUAGE MODEL\\nREMARK   1  REF                                                                 \\nREMARK   1  REFN                                                                \\nREMARK   1  PMID                                                                \\nREMARK   1  DOI    10.1101/2022.07.20.500902                                    \\nREMARK   1                                                                      \\nREMARK   1 LICENSE AND DISCLAIMERS                \\nREMARK   1 ESM METAGENOMIC ATLAS DATA IS AVAILABLE UNDER\\nREMARK   1 A CC-BY-4.0 LICENSE FOR ACADEMIC AND COMMERCIAL USE.\\nREMARK   1 COPYRIGHT (C) META PLATFORMS, INC. ALL RIGHTS RESERVED.\\nREMARK   1 USE OF THE ESM METAGENOMIC ATLAS DATA IS SUBJECT\\nREMARK   1 TO THE META OPEN SOURCE TERMS OF USE AND PRIVACY POLICY.\\nATOM    168  C   ACE A   0       9.976   8.120   7.984  1.00  0.00           C  \\nATOM      1  N   NLE A   1       8.593   7.504   7.797  1.00  0.74           N  \\nATOM      2  CA  NLE A   1       8.649   6.048   7.899  1.00  0.74           C  \\nATOM      3  C   NLE A   1       7.266   5.432   7.712  1.00  0.76           C  \\nATOM      5  O   NLE A   1       6.254   6.043   8.064  1.00  0.67           O  \\nATOM      6  N   ALA A   2       7.248   4.631   6.869  1.00  0.79           N  \\nATOM      7  CA  ALA A   2       5.925   4.058   6.634  1.00  0.80           C  \\nATOM      8  C   ALA A   2       5.507   3.151   7.787  1.00  0.79           C  \\nATOM      9  CB  ALA A   2       5.907   3.284   5.318  1.00  0.74           C  \\nATOM     10  O   ALA A   2       6.253   2.249   8.177  1.00  0.74           O  \\nATOM     11  N   THR A   3       4.679   3.521   8.593  1.00  0.87           N  \\nATOM     12  CA  THR A   3       4.083   2.690   9.633  1.00  0.89           C  \\nATOM     13  C   THR A   3       3.488   1.419   9.035  1.00  0.88           C  \\nATOM     14  CB  THR A   3       2.992   3.457  10.404  1.00  0.83           C  \\nATOM     15  O   THR A   3       3.190   1.369   7.840  1.00  0.85           O  \\nATOM     16  CG2 THR A   3       3.528   4.778  10.946  1.00  0.63           C  \\nATOM     17  OG1 THR A   3       1.893   3.724   9.525  1.00  0.65           O  \\nATOM     18  N   PRO A   4       3.551   0.253   9.836  1.00  0.87           N  \\nATOM     19  CA  PRO A   4       2.940  -0.979   9.331  1.00  0.88           C  \\nATOM     20  C   PRO A   4       1.540  -0.754   8.762  1.00  0.88           C  \\nATOM     21  CB  PRO A   4       2.888  -1.880  10.567  1.00  0.86           C  \\nATOM     22  O   PRO A   4       1.173  -1.365   7.756  1.00  0.87           O  \\nATOM     23  CG  PRO A   4       3.878  -1.284  11.515  1.00  0.79           C  \\nATOM     24  CD  PRO A   4       3.979   0.188  11.237  1.00  0.78           C  \\nATOM     25  N   ALA A   5       0.703   0.219   9.323  1.00  0.90           N  \\nATOM     26  CA  ALA A   5      -0.652   0.461   8.832  1.00  0.90           C  \\nATOM     27  C   ALA A   5      -0.627   1.087   7.441  1.00  0.90           C  \\nATOM     28  CB  ALA A   5      -1.416   1.358   9.802  1.00  0.88           C  \\nATOM     29  O   ALA A   5      -1.409   0.703   6.567  1.00  0.88           O  \\nATOM     30  N   ARG A   6       0.273   2.127   7.209  1.00  0.91           N  \\nATOM     31  CA  ARG A   6       0.365   2.756   5.896  1.00  0.91           C  \\nATOM     32  C   ARG A   6       0.820   1.754   4.839  1.00  0.90           C  \\nATOM     33  CB  ARG A   6       1.325   3.947   5.936  1.00  0.89           C  \\nATOM     34  O   ARG A   6       0.310   1.753   3.717  1.00  0.89           O  \\nATOM     35  CG  ARG A   6       0.713   5.212   6.515  1.00  0.82           C  \\nATOM     36  CD  ARG A   6       1.687   6.381   6.472  1.00  0.80           C  \\nATOM     37  NE  ARG A   6       1.118   7.580   7.080  1.00  0.75           N  \\nATOM     38  NH1 ARG A   6       3.089   8.785   7.044  1.00  0.64           N  \\nATOM     39  NH2 ARG A   6       1.179   9.727   7.895  1.00  0.62           N  \\nATOM     40  CZ  ARG A   6       1.797   8.695   7.338  1.00  0.72           C  \\nATOM     41  N   ARG A   7       1.896   0.856   5.232  1.00  0.91           N  \\nATOM     42  CA  ARG A   7       2.370  -0.125   4.261  1.00  0.91           C  \\nATOM     43  C   ARG A   7       1.259  -1.096   3.878  1.00  0.90           C  \\nATOM     44  CB  ARG A   7       3.571  -0.895   4.817  1.00  0.89           C  \\nATOM     45  O   ARG A   7       1.124  -1.462   2.708  1.00  0.89           O  \\nATOM     46  CG  ARG A   7       4.846  -0.071   4.900  1.00  0.82           C  \\nATOM     47  CD  ARG A   7       6.041  -0.921   5.308  1.00  0.80           C  \\nATOM     48  NE  ARG A   7       7.219  -0.102   5.575  1.00  0.74           N  \\nATOM     49  NH1 ARG A   7       8.464  -1.821   6.487  1.00  0.62           N  \\nATOM     50  NH2 ARG A   7       9.355   0.285   6.327  1.00  0.61           N  \\nATOM     51  CZ  ARG A   7       8.343  -0.548   6.129  1.00  0.70           C  \\nATOM     52  N   ARG A   8       0.495  -1.511   4.920  1.00  0.91           N  \\nATOM     53  CA  ARG A   8      -0.614  -2.428   4.677  1.00  0.91           C  \\nATOM     54  C   ARG A   8      -1.666  -1.788   3.777  1.00  0.90           C  \\nATOM     55  CB  ARG A   8      -1.250  -2.865   5.998  1.00  0.89           C  \\nATOM     56  O   ARG A   8      -2.203  -2.440   2.879  1.00  0.89           O  \\nATOM     57  CG  ARG A   8      -2.207  -4.039   5.863  1.00  0.80           C  \\nATOM     58  CD  ARG A   8      -2.725  -4.503   7.217  1.00  0.77           C  \\nATOM     59  NE  ARG A   8      -1.660  -5.076   8.034  1.00  0.70           N  \\nATOM     60  NH1 ARG A   8      -1.997  -7.247   7.318  1.00  0.56           N  \\nATOM     61  NH2 ARG A   8      -0.351  -6.781   8.844  1.00  0.54           N  \\nATOM     62  CZ  ARG A   8      -1.339  -6.367   8.064  1.00  0.69           C  \\nATOM     63  N   AIB A   9      -2.012  -0.574   4.062  1.00  0.91           N  \\nATOM     64  CA  AIB A   9      -3.006   0.120   3.246  1.00  0.91           C  \\nATOM     65  C   AIB A   9      -2.528   0.264   1.804  1.00  0.90           C  \\nATOM     67  O   AIB A   9      -3.312   0.109   0.865  1.00  0.89           O  \\nATOM     68  N   NLE A  10      -1.218   0.603   1.593  1.00  0.91           N  \\nATOM     69  CA  NLE A  10      -0.664   0.744   0.249  1.00  0.91           C  \\nATOM     70  C   NLE A  10      -0.717  -0.580  -0.507  1.00  0.91           C  \\nATOM     72  O   NLE A  10      -1.010  -0.607  -1.705  1.00  0.89           O  \\nATOM     73  N   LEU A  11      -0.379  -1.687   0.212  1.00  0.91           N  \\nATOM     74  CA  LEU A  11      -0.441  -3.010  -0.400  1.00  0.91           C  \\nATOM     75  C   LEU A  11      -1.874  -3.367  -0.779  1.00  0.91           C  \\nATOM     76  CB  LEU A  11       0.126  -4.068   0.551  1.00  0.90           C  \\nATOM     77  O   LEU A  11      -2.116  -3.933  -1.847  1.00  0.90           O  \\nATOM     78  CG  LEU A  11       1.633  -4.318   0.466  1.00  0.86           C  \\nATOM     79  CD1 LEU A  11       2.349  -3.641   1.629  1.00  0.82           C  \\nATOM     80  CD2 LEU A  11       1.926  -5.814   0.448  1.00  0.83           C  \\nATOM     81  N   NLE A  12      -2.810  -3.043   0.091  1.00  0.91           N  \\nATOM     82  CA  NLE A  12      -4.214  -3.337  -0.183  1.00  0.91           C  \\nATOM     83  C   NLE A  12      -4.696  -2.598  -1.428  1.00  0.91           C  \\nATOM     85  O   NLE A  12      -5.406  -3.167  -2.260  1.00  0.89           O  \\nATOM     86  N   ARG A  13      -4.335  -1.332  -1.531  1.00  0.91           N  \\nATOM     87  CA  ARG A  13      -4.730  -0.538  -2.690  1.00  0.91           C  \\nATOM     88  C   ARG A  13      -4.139  -1.111  -3.974  1.00  0.91           C  \\nATOM     89  CB  ARG A  13      -4.296   0.919  -2.517  1.00  0.90           C  \\nATOM     90  O   ARG A  13      -4.794  -1.112  -5.018  1.00  0.90           O  \\nATOM     91  CG  ARG A  13      -5.205   1.729  -1.607  1.00  0.81           C  \\nATOM     92  CD  ARG A  13      -4.910   3.220  -1.697  1.00  0.79           C  \\nATOM     93  NE  ARG A  13      -3.550   3.527  -1.263  1.00  0.72           N  \\nATOM     94  NH1 ARG A  13      -4.087   3.722   0.976  1.00  0.57           N  \\nATOM     95  NH2 ARG A  13      -1.925   4.029   0.280  1.00  0.55           N  \\nATOM     96  CZ  ARG A  13      -3.191   3.759  -0.003  1.00  0.71           C  \\nATOM     97  N   ALA A  14      -2.817  -1.477  -3.891  1.00  0.90           N  \\nATOM     98  CA  ALA A  14      -2.168  -2.050  -5.067  1.00  0.90           C  \\nATOM     99  C   ALA A  14      -2.889  -3.311  -5.532  1.00  0.90           C  \\nATOM    100  CB  ALA A  14      -0.702  -2.358  -4.768  1.00  0.89           C  \\nATOM    101  O   ALA A  14      -3.099  -3.507  -6.732  1.00  0.89           O  \\nATOM    102  N   PHE A  15      -3.209  -4.231  -4.470  1.00  0.90           N  \\nATOM    103  CA  PHE A  15      -3.896  -5.474  -4.804  1.00  0.90           C  \\nATOM    104  C   PHE A  15      -5.243  -5.189  -5.457  1.00  0.90           C  \\nATOM    105  CB  PHE A  15      -4.090  -6.335  -3.552  1.00  0.88           C  \\nATOM    106  O   PHE A  15      -5.665  -5.913  -6.362  1.00  0.89           O  \\nATOM    107  CG  PHE A  15      -3.011  -7.363  -3.349  1.00  0.77           C  \\nATOM    108  CD1 PHE A  15      -2.972  -8.514  -4.127  1.00  0.68           C  \\nATOM    109  CD2 PHE A  15      -2.034  -7.179  -2.379  1.00  0.69           C  \\nATOM    110  CE1 PHE A  15      -1.974  -9.468  -3.941  1.00  0.68           C  \\nATOM    111  CE2 PHE A  15      -1.033  -8.128  -2.188  1.00  0.67           C  \\nATOM    112  CZ  PHE A  15      -1.005  -9.272  -2.969  1.00  0.67           C  \\nATOM    113  N   LYS A  16      -6.034  -4.138  -5.051  1.00  0.91           N  \\nATOM    114  CA  LYS A  16      -7.334  -3.775  -5.608  1.00  0.91           C  \\nATOM    115  C   LYS A  16      -7.204  -3.330  -7.062  1.00  0.91           C  \\nATOM    116  CB  LYS A  16      -7.983  -2.667  -4.777  1.00  0.87           C  \\nATOM    117  O   LYS A  16      -8.066  -3.635  -7.889  1.00  0.89           O  \\nATOM    118  CG  LYS A  16      -8.592  -3.150  -3.469  1.00  0.74           C  \\nATOM    119  CD  LYS A  16      -9.473  -2.082  -2.834  1.00  0.70           C  \\nATOM    120  CE  LYS A  16     -10.054  -2.551  -1.508  1.00  0.64           C  \\nATOM    121  NZ  LYS A  16     -10.908  -1.502  -0.875  1.00  0.58           N  \\nATOM    122  N   ARG A  17      -6.071  -2.648  -7.354  1.00  0.89           N  \\nATOM    123  CA  ARG A  17      -5.835  -2.201  -8.723  1.00  0.89           C  \\nATOM    124  C   ARG A  17      -5.580  -3.386  -9.649  1.00  0.90           C  \\nATOM    125  CB  ARG A  17      -4.652  -1.231  -8.776  1.00  0.87           C  \\nATOM    126  O   ARG A  17      -5.888  -3.327 -10.841  1.00  0.89           O  \\nATOM    127  CG  ARG A  17      -4.956   0.142  -8.199  1.00  0.77           C  \\nATOM    128  CD  ARG A  17      -3.757   1.074  -8.299  1.00  0.74           C  \\nATOM    129  NE  ARG A  17      -4.010   2.352  -7.640  1.00  0.69           N  \\nATOM    130  NH1 ARG A  17      -1.806   3.042  -7.673  1.00  0.59           N  \\nATOM    131  NH2 ARG A  17      -3.419   4.387  -6.756  1.00  0.55           N  \\nATOM    132  CZ  ARG A  17      -3.078   3.258  -7.358  1.00  0.67           C  \\nATOM    133  N   LEU A  18      -4.984  -4.400  -9.121  1.00  0.89           N  \\nATOM    134  CA  LEU A  18      -4.686  -5.571  -9.938  1.00  0.89           C  \\nATOM    135  C   LEU A  18      -5.960  -6.338 -10.274  1.00  0.87           C  \\nATOM    136  CB  LEU A  18      -3.697  -6.490  -9.216  1.00  0.87           C  \\nATOM    137  O   LEU A  18      -5.977  -7.141 -11.209  1.00  0.84           O  \\nATOM    138  CG  LEU A  18      -2.287  -5.937  -9.003  1.00  0.79           C  \\nATOM    139  CD1 LEU A  18      -1.488  -6.859  -8.089  1.00  0.72           C  \\nATOM    140  CD2 LEU A  18      -1.577  -5.754 -10.340  1.00  0.73           C  \\nATOM    141  N   GLN A  19      -7.053  -6.109  -9.720  1.00  0.85           N  \\nATOM    142  CA  GLN A  19      -8.328  -6.753 -10.016  1.00  0.85           C  \\nATOM    143  C   GLN A  19      -9.083  -6.001 -11.108  1.00  0.83           C  \\nATOM    144  CB  GLN A  19      -9.187  -6.849  -8.754  1.00  0.81           C  \\nATOM    145  O   GLN A  19      -9.759  -6.614 -11.937  1.00  0.78           O  \\nATOM    146  CG  GLN A  19      -8.616  -7.777  -7.690  1.00  0.74           C  \\nATOM    147  CD  GLN A  19      -9.479  -7.842  -6.444  1.00  0.70           C  \\nATOM    148  NE2 GLN A  19      -9.482  -8.996  -5.785  1.00  0.63           N  \\nATOM    149  OE1 GLN A  19     -10.137  -6.864  -6.076  1.00  0.69           O  \");\n\tviewer_16950270345760543.setStyle({\"model\": 1},{\"cartoon\": {\"color\": \"gray\"}});\n\tviewer_16950270345760543.zoomTo();\nviewer_16950270345760543.render();\n});\n</script>",
      "text/html": [
       "<div id=\"3dmolviewer_16950270345760543\"  style=\"position: relative; width: 640px; height: 480px\">\n",
       "        <p id=\"3dmolwarning_16950270345760543\" style=\"background-color:#ffcccc;color:black\">You appear to be running in JupyterLab (or JavaScript failed to load for some other reason).  You need to install the 3dmol extension: <br>\n",
       "        <tt>jupyter labextension install jupyterlab_3dmol</tt></p>\n",
       "        </div>\n",
       "<script>\n",
       "\n",
       "var loadScriptAsync = function(uri){\n",
       "  return new Promise((resolve, reject) => {\n",
       "    //this is to ignore the existence of requirejs amd\n",
       "    var savedexports, savedmodule;\n",
       "    if (typeof exports !== 'undefined') savedexports = exports;\n",
       "    else exports = {}\n",
       "    if (typeof module !== 'undefined') savedmodule = module;\n",
       "    else module = {}\n",
       "\n",
       "    var tag = document.createElement('script');\n",
       "    tag.src = uri;\n",
       "    tag.async = true;\n",
       "    tag.onload = () => {\n",
       "        exports = savedexports;\n",
       "        module = savedmodule;\n",
       "        resolve();\n",
       "    };\n",
       "  var firstScriptTag = document.getElementsByTagName('script')[0];\n",
       "  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);\n",
       "});\n",
       "};\n",
       "\n",
       "if(typeof $3Dmolpromise === 'undefined') {\n",
       "$3Dmolpromise = null;\n",
       "  $3Dmolpromise = loadScriptAsync('https://cdnjs.cloudflare.com/ajax/libs/3Dmol/2.0.1/3Dmol-min.js');\n",
       "}\n",
       "\n",
       "var viewer_16950270345760543 = null;\n",
       "var warn = document.getElementById(\"3dmolwarning_16950270345760543\");\n",
       "if(warn) {\n",
       "    warn.parentNode.removeChild(warn);\n",
       "}\n",
       "$3Dmolpromise.then(function() {\n",
       "viewer_16950270345760543 = $3Dmol.createViewer(document.getElementById(\"3dmolviewer_16950270345760543\"),{backgroundColor:\"white\"});\n",
       "viewer_16950270345760543.zoomTo();\n",
       "\tviewer_16950270345760543.addModel(\"ATOM      1  H1  ACE     1      13.135   3.604   5.823  1.00  0.00           H  \\nATOM      2  CH3 ACE     1      12.290   2.907   5.776  1.00  0.00           C  \\nATOM      3  H2  ACE     1      12.655   1.882   5.711  1.00  0.00           H  \\nATOM      4  H3  ACE     1      11.736   3.090   4.846  1.00  0.00           H  \\nATOM      5  C   ACE     1      11.492   3.255   7.037  1.00  0.00           C  \\nATOM      6  O   ACE     1      11.274   2.425   7.881  1.00  0.00           O  \\nATOM      7  N   NLE     2      11.096   4.486   7.122  1.00  0.00           N  \\nATOM      8  H13 NLE     2      11.355   5.072   6.346  1.00  0.00           H  \\nATOM      9  CA  NLE     2      10.325   5.150   8.193  1.00  0.00           C  \\nATOM     10  C4  NLE     2      11.122   6.300   8.805  1.00  0.00           C  \\nATOM     11  C3  NLE     2      12.468   5.857   9.492  1.00  0.00           C  \\nATOM     12  C2  NLE     2      13.340   7.043  10.029  1.00  0.00           C  \\nATOM     13  C1  NLE     2      13.027   7.160  11.498  1.00  0.00           C  \\nATOM     14  H1  NLE     2      13.178   6.175  11.961  1.00  0.00           H  \\nATOM     15  H2  NLE     2      12.042   7.463  11.584  1.00  0.00           H  \\nATOM     16  H3  NLE     2      13.741   7.928  11.940  1.00  0.00           H  \\nATOM     17  H4  NLE     2      14.357   6.806  10.091  1.00  0.00           H  \\nATOM     18  H5  NLE     2      13.186   8.042   9.673  1.00  0.00           H  \\nATOM     19  H6  NLE     2      12.322   5.162  10.268  1.00  0.00           H  \\nATOM     20  H7  NLE     2      12.995   5.206   8.854  1.00  0.00           H  \\nATOM     21  H8  NLE     2      11.354   6.971   7.977  1.00  0.00           H  \\nATOM     22  H9  NLE     2      10.414   6.692   9.636  1.00  0.00           H  \\nATOM     23  H10 NLE     2      10.329   4.376   8.950  1.00  0.00           H  \\nATOM     24  C   NLE     2       8.862   5.431   7.851  1.00  0.00           C  \\nATOM     25  O   NLE     2       8.361   6.535   8.041  1.00  0.00           O  \\nATOM     26  N   ALA     3       8.329   4.470   7.098  1.00  0.00           N  \\nATOM     27  H   ALA     3       8.830   3.577   7.000  1.00  0.00           H  \\nATOM     28  CA  ALA     3       7.057   4.653   6.455  1.00  0.00           C  \\nATOM     29  HA  ALA     3       7.092   5.629   5.989  1.00  0.00           H  \\nATOM     30  CB  ALA     3       6.976   3.526   5.336  1.00  0.00           C  \\nATOM     31  HB1 ALA     3       7.207   2.523   5.709  1.00  0.00           H  \\nATOM     32  HB2 ALA     3       5.944   3.354   4.834  1.00  0.00           H  \\nATOM     33  HB3 ALA     3       7.729   3.841   4.571  1.00  0.00           H  \\nATOM     34  C   ALA     3       5.789   4.722   7.349  1.00  0.00           C  \\nATOM     35  O   ALA     3       5.853   4.434   8.503  1.00  0.00           O  \\nATOM     36  N   THR     4       4.650   5.209   6.835  1.00  0.00           N  \\nATOM     37  H   THR     4       4.693   5.407   5.851  1.00  0.00           H  \\nATOM     38  CA  THR     4       3.375   5.365   7.529  1.00  0.00           C  \\nATOM     39  HA  THR     4       3.492   5.603   8.601  1.00  0.00           H  \\nATOM     40  CB  THR     4       2.670   6.549   6.867  1.00  0.00           C  \\nATOM     41  HB  THR     4       1.662   6.669   7.356  1.00  0.00           H  \\nATOM     42  CG2 THR     4       3.362   7.904   7.079  1.00  0.00           C  \\nATOM     43 HG21 THR     4       4.303   7.915   6.595  1.00  0.00           H  \\nATOM     44 HG22 THR     4       2.696   8.635   6.569  1.00  0.00           H  \\nATOM     45 HG23 THR     4       3.557   8.017   8.168  1.00  0.00           H  \\nATOM     46  OG1 THR     4       2.393   6.418   5.561  1.00  0.00           O  \\nATOM     47  HG1 THR     4       2.309   7.279   5.278  1.00  0.00           H  \\nATOM     48  C   THR     4       2.588   4.055   7.481  1.00  0.00           C  \\nATOM     49  O   THR     4       2.698   3.302   6.461  1.00  0.00           O  \\nATOM     50  N   PRO     5       1.770   3.678   8.476  1.00  0.00           N  \\nATOM     51  CD  PRO     5       1.738   4.316   9.771  1.00  0.00           C  \\nATOM     52  HD2 PRO     5       1.184   5.303   9.545  1.00  0.00           H  \\nATOM     53  HD3 PRO     5       2.747   4.615  10.122  1.00  0.00           H  \\nATOM     54  CG  PRO     5       0.892   3.406  10.745  1.00  0.00           C  \\nATOM     55  HG2 PRO     5       0.118   4.054  11.296  1.00  0.00           H  \\nATOM     56  HG3 PRO     5       1.471   2.816  11.435  1.00  0.00           H  \\nATOM     57  CB  PRO     5       0.137   2.535   9.739  1.00  0.00           C  \\nATOM     58  HB2 PRO     5      -0.807   3.117   9.414  1.00  0.00           H  \\nATOM     59  HB3 PRO     5       0.029   1.582  10.114  1.00  0.00           H  \\nATOM     60  CA  PRO     5       1.042   2.413   8.515  1.00  0.00           C  \\nATOM     61  HA  PRO     5       1.656   1.549   8.712  1.00  0.00           H  \\nATOM     62  C   PRO     5       0.177   2.082   7.251  1.00  0.00           C  \\nATOM     63  O   PRO     5       0.167   0.945   6.850  1.00  0.00           O  \\nATOM     64  N   PR8     6      -0.386   3.106   6.668  1.00  0.00           N  \\nATOM     65  H19 PR8     6      -0.309   3.984   7.124  1.00  0.00           H  \\nATOM     66  CA  PR8     6      -0.929   3.174   5.308  1.00  0.00           C  \\nATOM     67  C8  PR8     6      -2.313   2.521   5.347  1.00  0.00           C  \\nATOM     68  C7  PR8     6      -3.335   2.651   4.196  1.00  0.00           C  \\nATOM     69  C6  PR8     6      -4.687   2.004   4.600  1.00  0.00           C  \\nATOM     70  C5  PR8     6      -5.710   1.757   3.496  1.00  0.00           C  \\nATOM     71  C4  PR8     6      -5.896   2.767   2.394  1.00  0.00           C  \\nATOM     72  C3  PR8     6      -6.912   2.467   1.309  1.00  0.00           C  \\nATOM     73  C2  PR8     6      -6.397   1.349   0.480  1.00  0.00           C  \\nATOM     74  H3  PR8     6      -6.385   0.298   0.835  1.00  0.00           H  \\nATOM     75  H4  PR8     6      -7.858   2.167   1.709  1.00  0.00           H  \\nATOM     76  H5  PR8     6      -6.979   3.410   0.707  1.00  0.00           H  \\nATOM     77  H6  PR8     6      -6.202   3.713   2.838  1.00  0.00           H  \\nATOM     78  H7  PR8     6      -4.831   2.936   1.977  1.00  0.00           H  \\nATOM     79  H8  PR8     6      -5.555   0.730   3.070  1.00  0.00           H  \\nATOM     80  H9  PR8     6      -6.643   1.609   3.945  1.00  0.00           H  \\nATOM     81  H10 PR8     6      -5.101   2.646   5.334  1.00  0.00           H  \\nATOM     82  H11 PR8     6      -4.482   1.081   5.124  1.00  0.00           H  \\nATOM     83  H12 PR8     6      -3.411   3.672   3.916  1.00  0.00           H  \\nATOM     84  H13 PR8     6      -2.962   2.089   3.359  1.00  0.00           H  \\nATOM     85  H14 PR8     6      -2.183   1.509   5.688  1.00  0.00           H  \\nATOM     86  H15 PR8     6      -2.894   2.921   6.124  1.00  0.00           H  \\nATOM     87  C10 PR8     6      -1.076   4.647   4.872  1.00  0.00           C  \\nATOM     88  H16 PR8     6      -1.778   5.141   5.501  1.00  0.00           H  \\nATOM     89  H17 PR8     6      -1.478   4.784   3.888  1.00  0.00           H  \\nATOM     90  H18 PR8     6      -0.107   5.133   4.922  1.00  0.00           H  \\nATOM     91  C   PR8     6      -0.047   2.447   4.260  1.00  0.00           C  \\nATOM     92  O   PR8     6      -0.512   1.646   3.417  1.00  0.00           O  \\nATOM     93  N   ARG     7       1.276   2.657   4.246  1.00  0.00           N  \\nATOM     94  H   ARG     7       1.645   3.193   4.969  1.00  0.00           H  \\nATOM     95  CA  ARG     7       2.202   2.155   3.179  1.00  0.00           C  \\nATOM     96  HA  ARG     7       1.729   1.968   2.216  1.00  0.00           H  \\nATOM     97  CB  ARG     7       3.287   3.180   3.025  1.00  0.00           C  \\nATOM     98  HB2 ARG     7       3.645   3.510   4.032  1.00  0.00           H  \\nATOM     99  HB3 ARG     7       4.245   2.787   2.460  1.00  0.00           H  \\nATOM    100  CG  ARG     7       2.890   4.529   2.434  1.00  0.00           C  \\nATOM    101  HG2 ARG     7       2.243   5.017   3.179  1.00  0.00           H  \\nATOM    102  HG3 ARG     7       3.873   5.112   2.378  1.00  0.00           H  \\nATOM    103  CD  ARG     7       2.263   4.459   1.042  1.00  0.00           C  \\nATOM    104  HD2 ARG     7       1.300   4.002   1.129  1.00  0.00           H  \\nATOM    105  HD3 ARG     7       2.101   5.479   0.641  1.00  0.00           H  \\nATOM    106  NE  ARG     7       3.212   3.857   0.083  1.00  0.00           N  \\nATOM    107  HE  ARG     7       4.184   4.107   0.231  1.00  0.00           H  \\nATOM    108  CZ  ARG     7       3.009   3.024  -0.868  1.00  0.00           C  \\nATOM    109  NH1 ARG     7       1.861   2.454  -1.024  1.00  0.00           N  \\nATOM    110 HH11 ARG     7       1.071   2.854  -0.461  1.00  0.00           H  \\nATOM    111 HH12 ARG     7       1.786   1.860  -1.864  1.00  0.00           H  \\nATOM    112  NH2 ARG     7       3.989   2.613  -1.623  1.00  0.00           N  \\nATOM    113 HH21 ARG     7       4.931   3.075  -1.599  1.00  0.00           H  \\nATOM    114 HH22 ARG     7       3.783   1.893  -2.332  1.00  0.00           H  \\nATOM    115  C   ARG     7       2.827   0.840   3.532  1.00  0.00           C  \\nATOM    116  O   ARG     7       3.222   0.095   2.604  1.00  0.00           O  \\nATOM    117  N   ARG     8       3.015   0.419   4.784  1.00  0.00           N  \\nATOM    118  H   ARG     8       2.767   1.114   5.453  1.00  0.00           H  \\nATOM    119  CA  ARG     8       3.253  -1.011   5.164  1.00  0.00           C  \\nATOM    120  HA  ARG     8       4.043  -1.388   4.524  1.00  0.00           H  \\nATOM    121  CB  ARG     8       3.766  -1.034   6.615  1.00  0.00           C  \\nATOM    122  HB2 ARG     8       2.943  -0.758   7.294  1.00  0.00           H  \\nATOM    123  HB3 ARG     8       4.077  -1.992   6.851  1.00  0.00           H  \\nATOM    124  CG  ARG     8       4.993  -0.122   6.949  1.00  0.00           C  \\nATOM    125  HG2 ARG     8       5.743  -0.312   6.246  1.00  0.00           H  \\nATOM    126  HG3 ARG     8       4.752   0.967   6.996  1.00  0.00           H  \\nATOM    127  CD  ARG     8       5.503  -0.482   8.343  1.00  0.00           C  \\nATOM    128  HD2 ARG     8       4.769  -0.525   9.033  1.00  0.00           H  \\nATOM    129  HD3 ARG     8       5.893  -1.571   8.280  1.00  0.00           H  \\nATOM    130  NE  ARG     8       6.566   0.469   8.740  1.00  0.00           N  \\nATOM    131  HE  ARG     8       6.210   1.331   9.167  1.00  0.00           H  \\nATOM    132  CZ  ARG     8       7.841   0.597   8.245  1.00  0.00           C  \\nATOM    133  NH1 ARG     8       8.514  -0.219   7.471  1.00  0.00           N  \\nATOM    134 HH11 ARG     8       8.192  -1.181   7.400  1.00  0.00           H  \\nATOM    135 HH12 ARG     8       9.508  -0.048   7.196  1.00  0.00           H  \\nATOM    136  NH2 ARG     8       8.479   1.736   8.574  1.00  0.00           N  \\nATOM    137 HH21 ARG     8       8.021   2.431   9.166  1.00  0.00           H  \\nATOM    138 HH22 ARG     8       9.506   1.715   8.612  1.00  0.00           H  \\nATOM    139  C   ARG     8       2.073  -1.979   4.874  1.00  0.00           C  \\nATOM    140  O   ARG     8       2.326  -3.057   4.271  1.00  0.00           O  \\nATOM    141  N   ARG     9       0.798  -1.563   5.217  1.00  0.00           N  \\nATOM    142  H   ARG     9       0.847  -0.545   5.527  1.00  0.00           H  \\nATOM    143  CA  ARG     9      -0.405  -2.398   5.045  1.00  0.00           C  \\nATOM    144  HA  ARG     9      -0.139  -3.420   5.448  1.00  0.00           H  \\nATOM    145  CB  ARG     9      -1.524  -1.965   6.049  1.00  0.00           C  \\nATOM    146  HB2 ARG     9      -1.707  -0.911   5.916  1.00  0.00           H  \\nATOM    147  HB3 ARG     9      -2.417  -2.633   5.946  1.00  0.00           H  \\nATOM    148  CG  ARG     9      -1.133  -2.113   7.509  1.00  0.00           C  \\nATOM    149  HG2 ARG     9      -0.927  -3.167   7.832  1.00  0.00           H  \\nATOM    150  HG3 ARG     9      -0.287  -1.527   7.704  1.00  0.00           H  \\nATOM    151  CD  ARG     9      -2.383  -1.645   8.364  1.00  0.00           C  \\nATOM    152  HD2 ARG     9      -2.109  -1.714   9.443  1.00  0.00           H  \\nATOM    153  HD3 ARG     9      -3.252  -2.264   8.302  1.00  0.00           H  \\nATOM    154  NE  ARG     9      -2.612  -0.191   8.238  1.00  0.00           N  \\nATOM    155  HE  ARG     9      -1.804   0.336   7.895  1.00  0.00           H  \\nATOM    156  CZ  ARG     9      -3.580   0.500   8.900  1.00  0.00           C  \\nATOM    157  NH1 ARG     9      -4.647  -0.096   9.494  1.00  0.00           N  \\nATOM    158 HH11 ARG     9      -4.777  -1.044   9.377  1.00  0.00           H  \\nATOM    159 HH12 ARG     9      -5.358   0.506   9.984  1.00  0.00           H  \\nATOM    160  NH2 ARG     9      -3.540   1.804   8.981  1.00  0.00           N  \\nATOM    161 HH21 ARG     9      -2.809   2.293   8.527  1.00  0.00           H  \\nATOM    162 HH22 ARG     9      -4.274   2.335   9.393  1.00  0.00           H  \\nATOM    163  C   ARG     9      -0.867  -2.531   3.601  1.00  0.00           C  \\nATOM    164  O   ARG     9      -1.400  -3.512   3.236  1.00  0.00           O  \\nATOM    165  N   AIB    10      -0.520  -1.595   2.697  1.00  0.00           N  \\nATOM    166  H7  AIB    10      -0.152  -0.804   3.114  1.00  0.00           H  \\nATOM    167  CA  AIB    10      -1.106  -1.424   1.331  1.00  0.00           C  \\nATOM    168  C1  AIB    10      -2.324  -0.475   1.414  1.00  0.00           C  \\nATOM    169  H1  AIB    10      -1.942   0.407   1.818  1.00  0.00           H  \\nATOM    170  H2  AIB    10      -2.941  -0.963   2.130  1.00  0.00           H  \\nATOM    171  H3  AIB    10      -2.967  -0.378   0.563  1.00  0.00           H  \\nATOM    172  C3  AIB    10      -0.103  -0.796   0.424  1.00  0.00           C  \\nATOM    173  H4  AIB    10       0.761  -1.476   0.283  1.00  0.00           H  \\nATOM    174  H5  AIB    10       0.276   0.082   0.900  1.00  0.00           H  \\nATOM    175  H6  AIB    10      -0.646  -0.487  -0.475  1.00  0.00           H  \\nATOM    176  C   AIB    10      -1.466  -2.723   0.566  1.00  0.00           C  \\nATOM    177  O   AIB    10      -2.689  -2.980   0.340  1.00  0.00           O  \\nATOM    178  N   NLE    11      -0.535  -3.634   0.240  1.00  0.00           N  \\nATOM    179  H13 NLE    11       0.355  -3.382   0.617  1.00  0.00           H  \\nATOM    180  CA  NLE    11      -0.645  -4.918  -0.535  1.00  0.00           C  \\nATOM    181  C4  NLE    11       0.659  -5.689  -0.212  1.00  0.00           C  \\nATOM    182  C3  NLE    11       1.906  -5.041  -0.830  1.00  0.00           C  \\nATOM    183  C2  NLE    11       3.089  -5.937  -0.597  1.00  0.00           C  \\nATOM    184  C1  NLE    11       4.314  -5.269  -1.280  1.00  0.00           C  \\nATOM    185  H1  NLE    11       4.754  -4.481  -0.720  1.00  0.00           H  \\nATOM    186  H2  NLE    11       4.073  -4.865  -2.286  1.00  0.00           H  \\nATOM    187  H3  NLE    11       5.114  -6.015  -1.439  1.00  0.00           H  \\nATOM    188  H4  NLE    11       3.215  -6.110   0.500  1.00  0.00           H  \\nATOM    189  H5  NLE    11       2.889  -6.881  -1.107  1.00  0.00           H  \\nATOM    190  H6  NLE    11       1.670  -5.000  -1.879  1.00  0.00           H  \\nATOM    191  H7  NLE    11       2.048  -4.003  -0.363  1.00  0.00           H  \\nATOM    192  H8  NLE    11       0.717  -5.702   0.852  1.00  0.00           H  \\nATOM    193  H9  NLE    11       0.666  -6.776  -0.460  1.00  0.00           H  \\nATOM    194  H10 NLE    11      -0.726  -4.722  -1.576  1.00  0.00           H  \\nATOM    195  C   NLE    11      -1.772  -5.955  -0.082  1.00  0.00           C  \\nATOM    196  O   NLE    11      -2.334  -6.689  -0.923  1.00  0.00           O  \\nATOM    197  N   LEU    12      -2.218  -5.830   1.137  1.00  0.00           N  \\nATOM    198  H   LEU    12      -1.846  -5.018   1.620  1.00  0.00           H  \\nATOM    199  CA  LEU    12      -3.272  -6.628   1.800  1.00  0.00           C  \\nATOM    200  HA  LEU    12      -3.238  -7.597   1.267  1.00  0.00           H  \\nATOM    201  CB  LEU    12      -2.904  -6.954   3.264  1.00  0.00           C  \\nATOM    202  HB2 LEU    12      -3.192  -6.146   3.911  1.00  0.00           H  \\nATOM    203  HB3 LEU    12      -3.453  -7.810   3.546  1.00  0.00           H  \\nATOM    204  CG  LEU    12      -1.453  -7.488   3.498  1.00  0.00           C  \\nATOM    205  HG  LEU    12      -0.739  -6.760   3.165  1.00  0.00           H  \\nATOM    206  CD1 LEU    12      -1.236  -7.770   4.986  1.00  0.00           C  \\nATOM    207 HD11 LEU    12      -1.466  -6.906   5.548  1.00  0.00           H  \\nATOM    208 HD12 LEU    12      -1.886  -8.642   5.294  1.00  0.00           H  \\nATOM    209 HD13 LEU    12      -0.189  -7.970   5.039  1.00  0.00           H  \\nATOM    210  CD2 LEU    12      -1.221  -8.819   2.768  1.00  0.00           C  \\nATOM    211 HD21 LEU    12      -1.128  -8.749   1.700  1.00  0.00           H  \\nATOM    212 HD22 LEU    12      -0.354  -9.370   3.127  1.00  0.00           H  \\nATOM    213 HD23 LEU    12      -1.924  -9.515   3.080  1.00  0.00           H  \\nATOM    214  C   LEU    12      -4.681  -6.043   1.614  1.00  0.00           C  \\nATOM    215  O   LEU    12      -5.659  -6.745   1.874  1.00  0.00           O  \\nATOM    216  N   NLE    13      -4.851  -4.768   1.231  1.00  0.00           N  \\nATOM    217  H13 NLE    13      -4.076  -4.234   1.060  1.00  0.00           H  \\nATOM    218  CA  NLE    13      -6.121  -4.032   1.265  1.00  0.00           C  \\nATOM    219  C4  NLE    13      -6.060  -2.759   2.165  1.00  0.00           C  \\nATOM    220  C3  NLE    13      -5.919  -2.994   3.682  1.00  0.00           C  \\nATOM    221  C2  NLE    13      -5.596  -1.662   4.316  1.00  0.00           C  \\nATOM    222  C1  NLE    13      -5.634  -1.737   5.856  1.00  0.00           C  \\nATOM    223  H1  NLE    13      -4.753  -2.120   6.305  1.00  0.00           H  \\nATOM    224  H2  NLE    13      -5.685  -0.704   6.350  1.00  0.00           H  \\nATOM    225  H3  NLE    13      -6.492  -2.277   6.124  1.00  0.00           H  \\nATOM    226  H4  NLE    13      -4.655  -1.283   3.977  1.00  0.00           H  \\nATOM    227  H5  NLE    13      -6.405  -1.054   3.943  1.00  0.00           H  \\nATOM    228  H6  NLE    13      -6.892  -3.371   3.849  1.00  0.00           H  \\nATOM    229  H7  NLE    13      -5.164  -3.811   3.836  1.00  0.00           H  \\nATOM    230  H8  NLE    13      -5.370  -2.107   1.883  1.00  0.00           H  \\nATOM    231  H9  NLE    13      -6.990  -2.194   1.957  1.00  0.00           H  \\nATOM    232  H10 NLE    13      -6.912  -4.759   1.642  1.00  0.00           H  \\nATOM    233  C   NLE    13      -6.553  -3.731  -0.198  1.00  0.00           C  \\nATOM    234  O   NLE    13      -5.714  -3.879  -1.151  1.00  0.00           O  \\nATOM    235  N   ARG    14      -7.814  -3.278  -0.431  1.00  0.00           N  \\nATOM    236  H   ARG    14      -8.366  -3.085   0.403  1.00  0.00           H  \\nATOM    237  CA  ARG    14      -8.297  -3.094  -1.770  1.00  0.00           C  \\nATOM    238  HA  ARG    14      -8.131  -4.019  -2.355  1.00  0.00           H  \\nATOM    239  CB  ARG    14      -9.836  -2.788  -1.846  1.00  0.00           C  \\nATOM    240  HB2 ARG    14     -10.448  -3.511  -1.204  1.00  0.00           H  \\nATOM    241  HB3 ARG    14      -9.962  -1.751  -1.581  1.00  0.00           H  \\nATOM    242  CG  ARG    14     -10.577  -2.942  -3.162  1.00  0.00           C  \\nATOM    243  HG2 ARG    14      -9.794  -3.009  -3.949  1.00  0.00           H  \\nATOM    244  HG3 ARG    14     -11.177  -3.886  -3.106  1.00  0.00           H  \\nATOM    245  CD  ARG    14     -11.581  -1.856  -3.547  1.00  0.00           C  \\nATOM    246  HD2 ARG    14     -10.945  -0.997  -3.706  1.00  0.00           H  \\nATOM    247  HD3 ARG    14     -12.029  -2.136  -4.579  1.00  0.00           H  \\nATOM    248  NE  ARG    14     -12.667  -1.653  -2.546  1.00  0.00           N  \\nATOM    249  HE  ARG    14     -12.410  -1.639  -1.523  1.00  0.00           H  \\nATOM    250  CZ  ARG    14     -13.929  -1.244  -2.732  1.00  0.00           C  \\nATOM    251  NH1 ARG    14     -14.401  -0.847  -3.903  1.00  0.00           N  \\nATOM    252 HH11 ARG    14     -13.776  -0.907  -4.732  1.00  0.00           H  \\nATOM    253 HH12 ARG    14     -15.390  -0.439  -3.902  1.00  0.00           H  \\nATOM    254  NH2 ARG    14     -14.798  -1.258  -1.761  1.00  0.00           N  \\nATOM    255 HH21 ARG    14     -14.445  -1.420  -0.868  1.00  0.00           H  \\nATOM    256 HH22 ARG    14     -15.554  -0.640  -1.919  1.00  0.00           H  \\nATOM    257  C   ARG    14      -7.422  -2.126  -2.601  1.00  0.00           C  \\nATOM    258  O   ARG    14      -7.301  -2.316  -3.790  1.00  0.00           O  \\nATOM    259  N   PR3    15      -6.861  -0.991  -2.081  1.00  0.00           N  \\nATOM    260  H10 PR3    15      -7.247  -0.606  -1.255  1.00  0.00           H  \\nATOM    261  CA  PR3    15      -5.931  -0.035  -2.666  1.00  0.00           C  \\nATOM    262  C5  PR3    15      -6.863   1.025  -3.316  1.00  0.00           C  \\nATOM    263  H6  PR3    15      -7.550   1.415  -2.548  1.00  0.00           H  \\nATOM    264  H7  PR3    15      -7.268   0.648  -4.264  1.00  0.00           H  \\nATOM    265  H8  PR3    15      -6.171   1.859  -3.664  1.00  0.00           H  \\nATOM    266  C3  PR3    15      -5.045   0.548  -1.519  1.00  0.00           C  \\nATOM    267  C2  PR3    15      -5.861   1.580  -0.737  1.00  0.00           C  \\nATOM    268  H3  PR3    15      -5.893   2.547  -1.206  1.00  0.00           H  \\nATOM    269  H4  PR3    15      -4.691  -0.257  -0.935  1.00  0.00           H  \\nATOM    270  H5  PR3    15      -4.126   1.030  -1.919  1.00  0.00           H  \\nATOM    271  C   PR3    15      -4.959  -0.714  -3.700  1.00  0.00           C  \\nATOM    272  O   PR3    15      -4.820  -0.245  -4.853  1.00  0.00           O  \\nATOM    273  N   PHE    16      -4.370  -1.884  -3.308  1.00  0.00           N  \\nATOM    274  H   PHE    16      -4.553  -2.239  -2.342  1.00  0.00           H  \\nATOM    275  CA  PHE    16      -3.262  -2.513  -4.074  1.00  0.00           C  \\nATOM    276  HA  PHE    16      -2.608  -1.809  -4.650  1.00  0.00           H  \\nATOM    277  CB  PHE    16      -2.393  -3.385  -3.146  1.00  0.00           C  \\nATOM    278  HB2 PHE    16      -1.841  -2.799  -2.495  1.00  0.00           H  \\nATOM    279  HB3 PHE    16      -3.053  -4.130  -2.521  1.00  0.00           H  \\nATOM    280  CG  PHE    16      -1.486  -4.302  -3.938  1.00  0.00           C  \\nATOM    281  CD1 PHE    16      -1.793  -5.663  -4.083  1.00  0.00           C  \\nATOM    282  HD1 PHE    16      -2.676  -6.032  -3.610  1.00  0.00           H  \\nATOM    283  CE1 PHE    16      -1.025  -6.507  -4.849  1.00  0.00           C  \\nATOM    284  HE1 PHE    16      -1.148  -7.637  -4.940  1.00  0.00           H  \\nATOM    285  CZ  PHE    16       0.073  -5.904  -5.552  1.00  0.00           C  \\nATOM    286  HZ  PHE    16       0.642  -6.574  -6.257  1.00  0.00           H  \\nATOM    287  CE2 PHE    16       0.423  -4.544  -5.446  1.00  0.00           C  \\nATOM    288  HE2 PHE    16       1.251  -4.150  -5.941  1.00  0.00           H  \\nATOM    289  CD2 PHE    16      -0.397  -3.748  -4.652  1.00  0.00           C  \\nATOM    290  HD2 PHE    16      -0.236  -2.653  -4.638  1.00  0.00           H  \\nATOM    291  C   PHE    16      -3.922  -3.304  -5.232  1.00  0.00           C  \\nATOM    292  O   PHE    16      -3.442  -3.265  -6.395  1.00  0.00           O  \\nATOM    293  N   LYS    17      -5.037  -3.978  -4.995  1.00  0.00           N  \\nATOM    294  H   LYS    17      -5.181  -4.114  -4.030  1.00  0.00           H  \\nATOM    295  CA  LYS    17      -5.798  -4.769  -5.944  1.00  0.00           C  \\nATOM    296  HA  LYS    17      -5.135  -5.572  -6.197  1.00  0.00           H  \\nATOM    297  CB  LYS    17      -6.894  -5.485  -5.055  1.00  0.00           C  \\nATOM    298  HB2 LYS    17      -6.349  -5.985  -4.280  1.00  0.00           H  \\nATOM    299  HB3 LYS    17      -7.566  -4.748  -4.549  1.00  0.00           H  \\nATOM    300  CG  LYS    17      -7.730  -6.572  -5.777  1.00  0.00           C  \\nATOM    301  HG2 LYS    17      -8.501  -6.764  -4.991  1.00  0.00           H  \\nATOM    302  HG3 LYS    17      -8.218  -6.034  -6.609  1.00  0.00           H  \\nATOM    303  CD  LYS    17      -6.871  -7.750  -6.282  1.00  0.00           C  \\nATOM    304  HD2 LYS    17      -6.540  -7.582  -7.326  1.00  0.00           H  \\nATOM    305  HD3 LYS    17      -6.046  -7.901  -5.645  1.00  0.00           H  \\nATOM    306  CE  LYS    17      -7.722  -9.005  -6.324  1.00  0.00           C  \\nATOM    307  HE2 LYS    17      -8.066  -9.185  -5.436  1.00  0.00           H  \\nATOM    308  HE3 LYS    17      -8.532  -8.794  -7.079  1.00  0.00           H  \\nATOM    309  NZ  LYS    17      -7.003 -10.236  -6.746  1.00  0.00           N  \\nATOM    310  HZ1 LYS    17      -7.732 -10.963  -6.950  1.00  0.00           H  \\nATOM    311  HZ2 LYS    17      -6.593 -10.155  -7.607  1.00  0.00           H  \\nATOM    312  HZ3 LYS    17      -6.382 -10.678  -6.053  1.00  0.00           H  \\nATOM    313  C   LYS    17      -6.367  -4.042  -7.146  1.00  0.00           C  \\nATOM    314  O   LYS    17      -6.506  -4.657  -8.248  1.00  0.00           O  \\nATOM    315  N   ARG    18      -6.633  -2.722  -7.036  1.00  0.00           N  \\nATOM    316  H   ARG    18      -6.523  -2.372  -6.128  1.00  0.00           H  \\nATOM    317  CA  ARG    18      -7.087  -1.829  -8.114  1.00  0.00           C  \\nATOM    318  HA  ARG    18      -7.877  -2.262  -8.744  1.00  0.00           H  \\nATOM    319  CB  ARG    18      -7.528  -0.498  -7.502  1.00  0.00           C  \\nATOM    320  HB2 ARG    18      -8.431  -0.688  -6.860  1.00  0.00           H  \\nATOM    321  HB3 ARG    18      -6.802  -0.138  -6.758  1.00  0.00           H  \\nATOM    322  CG  ARG    18      -7.656   0.678  -8.531  1.00  0.00           C  \\nATOM    323  HG2 ARG    18      -6.637   1.016  -8.810  1.00  0.00           H  \\nATOM    324  HG3 ARG    18      -8.248   0.272  -9.393  1.00  0.00           H  \\nATOM    325  CD  ARG    18      -8.358   1.878  -7.899  1.00  0.00           C  \\nATOM    326  HD2 ARG    18      -9.385   1.575  -7.790  1.00  0.00           H  \\nATOM    327  HD3 ARG    18      -8.047   2.036  -6.776  1.00  0.00           H  \\nATOM    328  NE  ARG    18      -8.277   3.100  -8.664  1.00  0.00           N  \\nATOM    329  HE  ARG    18      -7.546   3.717  -8.364  1.00  0.00           H  \\nATOM    330  CZ  ARG    18      -9.121   3.525  -9.573  1.00  0.00           C  \\nATOM    331  NH1 ARG    18     -10.195   2.800  -9.890  1.00  0.00           N  \\nATOM    332 HH11 ARG    18     -10.252   1.839  -9.682  1.00  0.00           H  \\nATOM    333 HH12 ARG    18     -10.681   3.213 -10.629  1.00  0.00           H  \\nATOM    334  NH2 ARG    18      -9.059   4.677 -10.178  1.00  0.00           N  \\nATOM    335 HH21 ARG    18      -8.266   5.258  -9.955  1.00  0.00           H  \\nATOM    336 HH22 ARG    18      -9.814   4.936 -10.787  1.00  0.00           H  \\nATOM    337  C   ARG    18      -5.958  -1.607  -9.149  1.00  0.00           C  \\nATOM    338  O   ARG    18      -6.197  -1.160 -10.257  1.00  0.00           O  \\nATOM    339  N   LEU    19      -4.697  -1.930  -8.872  1.00  0.00           N  \\nATOM    340  H   LEU    19      -4.548  -2.427  -8.013  1.00  0.00           H  \\nATOM    341  CA  LEU    19      -3.574  -1.715  -9.739  1.00  0.00           C  \\nATOM    342  HA  LEU    19      -3.953  -0.981 -10.415  1.00  0.00           H  \\nATOM    343  CB  LEU    19      -2.429  -1.092  -8.899  1.00  0.00           C  \\nATOM    344  HB2 LEU    19      -1.968  -1.809  -8.203  1.00  0.00           H  \\nATOM    345  HB3 LEU    19      -1.695  -0.818  -9.666  1.00  0.00           H  \\nATOM    346  CG  LEU    19      -2.729   0.166  -7.899  1.00  0.00           C  \\nATOM    347  HG  LEU    19      -3.587  -0.044  -7.288  1.00  0.00           H  \\nATOM    348  CD1 LEU    19      -1.490   0.433  -6.978  1.00  0.00           C  \\nATOM    349 HD11 LEU    19      -1.838   1.252  -6.347  1.00  0.00           H  \\nATOM    350 HD12 LEU    19      -1.362  -0.492  -6.394  1.00  0.00           H  \\nATOM    351 HD13 LEU    19      -0.620   0.612  -7.554  1.00  0.00           H  \\nATOM    352  CD2 LEU    19      -3.009   1.462  -8.759  1.00  0.00           C  \\nATOM    353 HD21 LEU    19      -2.096   1.885  -9.142  1.00  0.00           H  \\nATOM    354 HD22 LEU    19      -3.583   1.109  -9.686  1.00  0.00           H  \\nATOM    355 HD23 LEU    19      -3.626   2.236  -8.169  1.00  0.00           H  \\nATOM    356  C   LEU    19      -3.029  -2.842 -10.594  1.00  0.00           C  \\nATOM    357  O   LEU    19      -2.301  -2.533 -11.529  1.00  0.00           O  \\nATOM    358  N   GLN    20      -3.549  -4.037 -10.374  1.00  0.00           N  \\nATOM    359  H   GLN    20      -4.155  -4.120  -9.633  1.00  0.00           H  \\nATOM    360  CA  GLN    20      -3.339  -5.195 -11.235  1.00  0.00           C  \\nATOM    361  HA  GLN    20      -2.381  -5.194 -11.710  1.00  0.00           H  \\nATOM    362  CB  GLN    20      -3.396  -6.452 -10.248  1.00  0.00           C  \\nATOM    363  HB2 GLN    20      -4.375  -6.638  -9.879  1.00  0.00           H  \\nATOM    364  HB3 GLN    20      -3.204  -7.364 -10.880  1.00  0.00           H  \\nATOM    365  CG  GLN    20      -2.356  -6.513  -9.076  1.00  0.00           C  \\nATOM    366  HG2 GLN    20      -1.402  -6.165  -9.509  1.00  0.00           H  \\nATOM    367  HG3 GLN    20      -2.581  -5.764  -8.245  1.00  0.00           H  \\nATOM    368  CD  GLN    20      -2.361  -7.868  -8.369  1.00  0.00           C  \\nATOM    369  OE1 GLN    20      -3.344  -8.416  -7.946  1.00  0.00           O  \\nATOM    370  NE2 GLN    20      -1.205  -8.495  -8.296  1.00  0.00           N  \\nATOM    371 HE21 GLN    20      -1.190  -9.191  -7.675  1.00  0.00           H  \\nATOM    372 HE22 GLN    20      -0.372  -8.022  -8.753  1.00  0.00           H  \\nATOM    373  C   GLN    20      -4.385  -5.314 -12.302  1.00  0.00           C  \\nATOM    374  O   GLN    20      -4.150  -6.048 -13.244  1.00  0.00           O  \\nATOM    375  OXT GLN    20      -5.485  -4.730 -12.199  1.00  0.00           O  \\nTER     376      GLN    20                                                       \\nEND   \\n\");\n",
       "\tviewer_16950270345760543.setStyle({\"cartoon\": {\"color\": \"spectrum\"}});\n",
       "\tviewer_16950270345760543.addModel(\"HEADER                                            18-OCT-22                     \\nTITLE     ESMFOLD V1 PREDICTION FOR INPUT\\nREMARK   1                                                                      \\nREMARK   1 REFERENCE 1                                                          \\nREMARK   1  AUTH   ZEMING LIN, HALIL AKIN, ROSHAN RAO, BRIAN HIE, ZHONGKAI ZHU,\\nREMARK   1  AUTH 2 WENTING LU, NIKITA SMETANIN, ROBERT VERKUIL, ORI KABELI,\\nREMARK   1  AUTH 3 YANIV SHMUELI, ALLAN DOS SANTOS COSTA,\\nREMARK   1  AUTH 4 MARYAM FAZEL-ZARANDI, TOM SERCU, SALVATORE CANDIDO,\\nREMARK   1  AUTH 5 ALEXANDER RIVES\\nREMARK   1  TITL   EVOLUTIONARY-SCALE PREDICTION OF ATOMIC LEVEL PROTEIN\\nREMARK   1  TITL 2 STRUCTURE WITH A LANGUAGE MODEL\\nREMARK   1  REF                                                                 \\nREMARK   1  REFN                                                                \\nREMARK   1  PMID                                                                \\nREMARK   1  DOI    10.1101/2022.07.20.500902                                    \\nREMARK   1                                                                      \\nREMARK   1 LICENSE AND DISCLAIMERS                \\nREMARK   1 ESM METAGENOMIC ATLAS DATA IS AVAILABLE UNDER\\nREMARK   1 A CC-BY-4.0 LICENSE FOR ACADEMIC AND COMMERCIAL USE.\\nREMARK   1 COPYRIGHT (C) META PLATFORMS, INC. ALL RIGHTS RESERVED.\\nREMARK   1 USE OF THE ESM METAGENOMIC ATLAS DATA IS SUBJECT\\nREMARK   1 TO THE META OPEN SOURCE TERMS OF USE AND PRIVACY POLICY.\\nATOM    168  C   ACE A   0       9.976   8.120   7.984  1.00  0.00           C  \\nATOM      1  N   NLE A   1       8.593   7.504   7.797  1.00  0.74           N  \\nATOM      2  CA  NLE A   1       8.649   6.048   7.899  1.00  0.74           C  \\nATOM      3  C   NLE A   1       7.266   5.432   7.712  1.00  0.76           C  \\nATOM      5  O   NLE A   1       6.254   6.043   8.064  1.00  0.67           O  \\nATOM      6  N   ALA A   2       7.248   4.631   6.869  1.00  0.79           N  \\nATOM      7  CA  ALA A   2       5.925   4.058   6.634  1.00  0.80           C  \\nATOM      8  C   ALA A   2       5.507   3.151   7.787  1.00  0.79           C  \\nATOM      9  CB  ALA A   2       5.907   3.284   5.318  1.00  0.74           C  \\nATOM     10  O   ALA A   2       6.253   2.249   8.177  1.00  0.74           O  \\nATOM     11  N   THR A   3       4.679   3.521   8.593  1.00  0.87           N  \\nATOM     12  CA  THR A   3       4.083   2.690   9.633  1.00  0.89           C  \\nATOM     13  C   THR A   3       3.488   1.419   9.035  1.00  0.88           C  \\nATOM     14  CB  THR A   3       2.992   3.457  10.404  1.00  0.83           C  \\nATOM     15  O   THR A   3       3.190   1.369   7.840  1.00  0.85           O  \\nATOM     16  CG2 THR A   3       3.528   4.778  10.946  1.00  0.63           C  \\nATOM     17  OG1 THR A   3       1.893   3.724   9.525  1.00  0.65           O  \\nATOM     18  N   PRO A   4       3.551   0.253   9.836  1.00  0.87           N  \\nATOM     19  CA  PRO A   4       2.940  -0.979   9.331  1.00  0.88           C  \\nATOM     20  C   PRO A   4       1.540  -0.754   8.762  1.00  0.88           C  \\nATOM     21  CB  PRO A   4       2.888  -1.880  10.567  1.00  0.86           C  \\nATOM     22  O   PRO A   4       1.173  -1.365   7.756  1.00  0.87           O  \\nATOM     23  CG  PRO A   4       3.878  -1.284  11.515  1.00  0.79           C  \\nATOM     24  CD  PRO A   4       3.979   0.188  11.237  1.00  0.78           C  \\nATOM     25  N   ALA A   5       0.703   0.219   9.323  1.00  0.90           N  \\nATOM     26  CA  ALA A   5      -0.652   0.461   8.832  1.00  0.90           C  \\nATOM     27  C   ALA A   5      -0.627   1.087   7.441  1.00  0.90           C  \\nATOM     28  CB  ALA A   5      -1.416   1.358   9.802  1.00  0.88           C  \\nATOM     29  O   ALA A   5      -1.409   0.703   6.567  1.00  0.88           O  \\nATOM     30  N   ARG A   6       0.273   2.127   7.209  1.00  0.91           N  \\nATOM     31  CA  ARG A   6       0.365   2.756   5.896  1.00  0.91           C  \\nATOM     32  C   ARG A   6       0.820   1.754   4.839  1.00  0.90           C  \\nATOM     33  CB  ARG A   6       1.325   3.947   5.936  1.00  0.89           C  \\nATOM     34  O   ARG A   6       0.310   1.753   3.717  1.00  0.89           O  \\nATOM     35  CG  ARG A   6       0.713   5.212   6.515  1.00  0.82           C  \\nATOM     36  CD  ARG A   6       1.687   6.381   6.472  1.00  0.80           C  \\nATOM     37  NE  ARG A   6       1.118   7.580   7.080  1.00  0.75           N  \\nATOM     38  NH1 ARG A   6       3.089   8.785   7.044  1.00  0.64           N  \\nATOM     39  NH2 ARG A   6       1.179   9.727   7.895  1.00  0.62           N  \\nATOM     40  CZ  ARG A   6       1.797   8.695   7.338  1.00  0.72           C  \\nATOM     41  N   ARG A   7       1.896   0.856   5.232  1.00  0.91           N  \\nATOM     42  CA  ARG A   7       2.370  -0.125   4.261  1.00  0.91           C  \\nATOM     43  C   ARG A   7       1.259  -1.096   3.878  1.00  0.90           C  \\nATOM     44  CB  ARG A   7       3.571  -0.895   4.817  1.00  0.89           C  \\nATOM     45  O   ARG A   7       1.124  -1.462   2.708  1.00  0.89           O  \\nATOM     46  CG  ARG A   7       4.846  -0.071   4.900  1.00  0.82           C  \\nATOM     47  CD  ARG A   7       6.041  -0.921   5.308  1.00  0.80           C  \\nATOM     48  NE  ARG A   7       7.219  -0.102   5.575  1.00  0.74           N  \\nATOM     49  NH1 ARG A   7       8.464  -1.821   6.487  1.00  0.62           N  \\nATOM     50  NH2 ARG A   7       9.355   0.285   6.327  1.00  0.61           N  \\nATOM     51  CZ  ARG A   7       8.343  -0.548   6.129  1.00  0.70           C  \\nATOM     52  N   ARG A   8       0.495  -1.511   4.920  1.00  0.91           N  \\nATOM     53  CA  ARG A   8      -0.614  -2.428   4.677  1.00  0.91           C  \\nATOM     54  C   ARG A   8      -1.666  -1.788   3.777  1.00  0.90           C  \\nATOM     55  CB  ARG A   8      -1.250  -2.865   5.998  1.00  0.89           C  \\nATOM     56  O   ARG A   8      -2.203  -2.440   2.879  1.00  0.89           O  \\nATOM     57  CG  ARG A   8      -2.207  -4.039   5.863  1.00  0.80           C  \\nATOM     58  CD  ARG A   8      -2.725  -4.503   7.217  1.00  0.77           C  \\nATOM     59  NE  ARG A   8      -1.660  -5.076   8.034  1.00  0.70           N  \\nATOM     60  NH1 ARG A   8      -1.997  -7.247   7.318  1.00  0.56           N  \\nATOM     61  NH2 ARG A   8      -0.351  -6.781   8.844  1.00  0.54           N  \\nATOM     62  CZ  ARG A   8      -1.339  -6.367   8.064  1.00  0.69           C  \\nATOM     63  N   AIB A   9      -2.012  -0.574   4.062  1.00  0.91           N  \\nATOM     64  CA  AIB A   9      -3.006   0.120   3.246  1.00  0.91           C  \\nATOM     65  C   AIB A   9      -2.528   0.264   1.804  1.00  0.90           C  \\nATOM     67  O   AIB A   9      -3.312   0.109   0.865  1.00  0.89           O  \\nATOM     68  N   NLE A  10      -1.218   0.603   1.593  1.00  0.91           N  \\nATOM     69  CA  NLE A  10      -0.664   0.744   0.249  1.00  0.91           C  \\nATOM     70  C   NLE A  10      -0.717  -0.580  -0.507  1.00  0.91           C  \\nATOM     72  O   NLE A  10      -1.010  -0.607  -1.705  1.00  0.89           O  \\nATOM     73  N   LEU A  11      -0.379  -1.687   0.212  1.00  0.91           N  \\nATOM     74  CA  LEU A  11      -0.441  -3.010  -0.400  1.00  0.91           C  \\nATOM     75  C   LEU A  11      -1.874  -3.367  -0.779  1.00  0.91           C  \\nATOM     76  CB  LEU A  11       0.126  -4.068   0.551  1.00  0.90           C  \\nATOM     77  O   LEU A  11      -2.116  -3.933  -1.847  1.00  0.90           O  \\nATOM     78  CG  LEU A  11       1.633  -4.318   0.466  1.00  0.86           C  \\nATOM     79  CD1 LEU A  11       2.349  -3.641   1.629  1.00  0.82           C  \\nATOM     80  CD2 LEU A  11       1.926  -5.814   0.448  1.00  0.83           C  \\nATOM     81  N   NLE A  12      -2.810  -3.043   0.091  1.00  0.91           N  \\nATOM     82  CA  NLE A  12      -4.214  -3.337  -0.183  1.00  0.91           C  \\nATOM     83  C   NLE A  12      -4.696  -2.598  -1.428  1.00  0.91           C  \\nATOM     85  O   NLE A  12      -5.406  -3.167  -2.260  1.00  0.89           O  \\nATOM     86  N   ARG A  13      -4.335  -1.332  -1.531  1.00  0.91           N  \\nATOM     87  CA  ARG A  13      -4.730  -0.538  -2.690  1.00  0.91           C  \\nATOM     88  C   ARG A  13      -4.139  -1.111  -3.974  1.00  0.91           C  \\nATOM     89  CB  ARG A  13      -4.296   0.919  -2.517  1.00  0.90           C  \\nATOM     90  O   ARG A  13      -4.794  -1.112  -5.018  1.00  0.90           O  \\nATOM     91  CG  ARG A  13      -5.205   1.729  -1.607  1.00  0.81           C  \\nATOM     92  CD  ARG A  13      -4.910   3.220  -1.697  1.00  0.79           C  \\nATOM     93  NE  ARG A  13      -3.550   3.527  -1.263  1.00  0.72           N  \\nATOM     94  NH1 ARG A  13      -4.087   3.722   0.976  1.00  0.57           N  \\nATOM     95  NH2 ARG A  13      -1.925   4.029   0.280  1.00  0.55           N  \\nATOM     96  CZ  ARG A  13      -3.191   3.759  -0.003  1.00  0.71           C  \\nATOM     97  N   ALA A  14      -2.817  -1.477  -3.891  1.00  0.90           N  \\nATOM     98  CA  ALA A  14      -2.168  -2.050  -5.067  1.00  0.90           C  \\nATOM     99  C   ALA A  14      -2.889  -3.311  -5.532  1.00  0.90           C  \\nATOM    100  CB  ALA A  14      -0.702  -2.358  -4.768  1.00  0.89           C  \\nATOM    101  O   ALA A  14      -3.099  -3.507  -6.732  1.00  0.89           O  \\nATOM    102  N   PHE A  15      -3.209  -4.231  -4.470  1.00  0.90           N  \\nATOM    103  CA  PHE A  15      -3.896  -5.474  -4.804  1.00  0.90           C  \\nATOM    104  C   PHE A  15      -5.243  -5.189  -5.457  1.00  0.90           C  \\nATOM    105  CB  PHE A  15      -4.090  -6.335  -3.552  1.00  0.88           C  \\nATOM    106  O   PHE A  15      -5.665  -5.913  -6.362  1.00  0.89           O  \\nATOM    107  CG  PHE A  15      -3.011  -7.363  -3.349  1.00  0.77           C  \\nATOM    108  CD1 PHE A  15      -2.972  -8.514  -4.127  1.00  0.68           C  \\nATOM    109  CD2 PHE A  15      -2.034  -7.179  -2.379  1.00  0.69           C  \\nATOM    110  CE1 PHE A  15      -1.974  -9.468  -3.941  1.00  0.68           C  \\nATOM    111  CE2 PHE A  15      -1.033  -8.128  -2.188  1.00  0.67           C  \\nATOM    112  CZ  PHE A  15      -1.005  -9.272  -2.969  1.00  0.67           C  \\nATOM    113  N   LYS A  16      -6.034  -4.138  -5.051  1.00  0.91           N  \\nATOM    114  CA  LYS A  16      -7.334  -3.775  -5.608  1.00  0.91           C  \\nATOM    115  C   LYS A  16      -7.204  -3.330  -7.062  1.00  0.91           C  \\nATOM    116  CB  LYS A  16      -7.983  -2.667  -4.777  1.00  0.87           C  \\nATOM    117  O   LYS A  16      -8.066  -3.635  -7.889  1.00  0.89           O  \\nATOM    118  CG  LYS A  16      -8.592  -3.150  -3.469  1.00  0.74           C  \\nATOM    119  CD  LYS A  16      -9.473  -2.082  -2.834  1.00  0.70           C  \\nATOM    120  CE  LYS A  16     -10.054  -2.551  -1.508  1.00  0.64           C  \\nATOM    121  NZ  LYS A  16     -10.908  -1.502  -0.875  1.00  0.58           N  \\nATOM    122  N   ARG A  17      -6.071  -2.648  -7.354  1.00  0.89           N  \\nATOM    123  CA  ARG A  17      -5.835  -2.201  -8.723  1.00  0.89           C  \\nATOM    124  C   ARG A  17      -5.580  -3.386  -9.649  1.00  0.90           C  \\nATOM    125  CB  ARG A  17      -4.652  -1.231  -8.776  1.00  0.87           C  \\nATOM    126  O   ARG A  17      -5.888  -3.327 -10.841  1.00  0.89           O  \\nATOM    127  CG  ARG A  17      -4.956   0.142  -8.199  1.00  0.77           C  \\nATOM    128  CD  ARG A  17      -3.757   1.074  -8.299  1.00  0.74           C  \\nATOM    129  NE  ARG A  17      -4.010   2.352  -7.640  1.00  0.69           N  \\nATOM    130  NH1 ARG A  17      -1.806   3.042  -7.673  1.00  0.59           N  \\nATOM    131  NH2 ARG A  17      -3.419   4.387  -6.756  1.00  0.55           N  \\nATOM    132  CZ  ARG A  17      -3.078   3.258  -7.358  1.00  0.67           C  \\nATOM    133  N   LEU A  18      -4.984  -4.400  -9.121  1.00  0.89           N  \\nATOM    134  CA  LEU A  18      -4.686  -5.571  -9.938  1.00  0.89           C  \\nATOM    135  C   LEU A  18      -5.960  -6.338 -10.274  1.00  0.87           C  \\nATOM    136  CB  LEU A  18      -3.697  -6.490  -9.216  1.00  0.87           C  \\nATOM    137  O   LEU A  18      -5.977  -7.141 -11.209  1.00  0.84           O  \\nATOM    138  CG  LEU A  18      -2.287  -5.937  -9.003  1.00  0.79           C  \\nATOM    139  CD1 LEU A  18      -1.488  -6.859  -8.089  1.00  0.72           C  \\nATOM    140  CD2 LEU A  18      -1.577  -5.754 -10.340  1.00  0.73           C  \\nATOM    141  N   GLN A  19      -7.053  -6.109  -9.720  1.00  0.85           N  \\nATOM    142  CA  GLN A  19      -8.328  -6.753 -10.016  1.00  0.85           C  \\nATOM    143  C   GLN A  19      -9.083  -6.001 -11.108  1.00  0.83           C  \\nATOM    144  CB  GLN A  19      -9.187  -6.849  -8.754  1.00  0.81           C  \\nATOM    145  O   GLN A  19      -9.759  -6.614 -11.937  1.00  0.78           O  \\nATOM    146  CG  GLN A  19      -8.616  -7.777  -7.690  1.00  0.74           C  \\nATOM    147  CD  GLN A  19      -9.479  -7.842  -6.444  1.00  0.70           C  \\nATOM    148  NE2 GLN A  19      -9.482  -8.996  -5.785  1.00  0.63           N  \\nATOM    149  OE1 GLN A  19     -10.137  -6.864  -6.076  1.00  0.69           O  \");\n",
       "\tviewer_16950270345760543.setStyle({\"model\": 1},{\"cartoon\": {\"color\": \"gray\"}});\n",
       "\tviewer_16950270345760543.zoomTo();\n",
       "viewer_16950270345760543.render();\n",
       "});\n",
       "</script>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<py3Dmol.view at 0x7f5a1f5c51f0>"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "with open(\"data/aligned.pdb\", \"r\") as f:\n",
    "    aligned_data = f.read()\n",
    "\n",
    "p = py3Dmol.view()\n",
    "p.addModel(aligned_data, format=\"pdb\")\n",
    "p.setStyle({'cartoon': {'color':'spectrum'}})\n",
    "p.addModel(template_data, format=\"pdb\")\n",
    "p.setStyle({'model': 1}, {'cartoon': {'color':'gray'}}) \n",
    "p.zoomTo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "stapep",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
